<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Introduction | Single Cell Sequencing Analysis</title>
  <meta name="description" content="Training Class of Single Cell Sequencing Analysis." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Introduction | Single Cell Sequencing Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Training Class of Single Cell Sequencing Analysis." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Introduction | Single Cell Sequencing Analysis" />
  
  <meta name="twitter:description" content="Training Class of Single Cell Sequencing Analysis." />
  

<meta name="author" content="Zhilong Jia" />


<meta name="date" content="2021-04-22" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="a-introduction-to-single-cell-sequening.html"/>
<script src="libs/header-attrs-2.7/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Single Cell Sequencing Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Content</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="intro.html"><a href="intro.html#read-data-library-packages"><i class="fa fa-check"></i><b>2.1</b> Read data Library packages</a></li>
<li class="chapter" data-level="2.2" data-path="intro.html"><a href="intro.html#dimplot-cell-or-cluster"><i class="fa fa-check"></i><b>2.2</b> DimPlot : cell or cluster</a></li>
<li class="chapter" data-level="2.3" data-path="intro.html"><a href="intro.html#单细胞可视化"><i class="fa fa-check"></i><b>2.3</b> 单细胞可视化</a></li>
<li class="chapter" data-level="2.4" data-path="intro.html"><a href="intro.html#dimplot1"><i class="fa fa-check"></i><b>2.4</b> dimplot1</a></li>
<li class="chapter" data-level="2.5" data-path="intro.html"><a href="intro.html#比例"><i class="fa fa-check"></i><b>2.5</b> 比例</a></li>
<li class="chapter" data-level="2.6" data-path="intro.html"><a href="intro.html#数据转化"><i class="fa fa-check"></i><b>2.6</b> 数据转化</a></li>
<li class="chapter" data-level="2.7" data-path="intro.html"><a href="intro.html#sessioninfo"><i class="fa fa-check"></i><b>2.7</b> sessionInfo</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="a-introduction-to-single-cell-sequening.html"><a href="a-introduction-to-single-cell-sequening.html"><i class="fa fa-check"></i><b>3</b> A introduction to Single Cell Sequening</a>
<ul>
<li class="chapter" data-level="3.1" data-path="a-introduction-to-single-cell-sequening.html"><a href="a-introduction-to-single-cell-sequening.html#data-format-of-seurat"><i class="fa fa-check"></i><b>3.1</b> data format of Seurat</a></li>
<li class="chapter" data-level="3.2" data-path="a-introduction-to-single-cell-sequening.html"><a href="a-introduction-to-single-cell-sequening.html#data-access"><i class="fa fa-check"></i><b>3.2</b> Data Access</a></li>
<li class="chapter" data-level="3.3" data-path="a-introduction-to-single-cell-sequening.html"><a href="a-introduction-to-single-cell-sequening.html#methods"><i class="fa fa-check"></i><b>3.3</b> Methods</a></li>
<li class="chapter" data-level="3.4" data-path="a-introduction-to-single-cell-sequening.html"><a href="a-introduction-to-single-cell-sequening.html#test-read-h5-file."><i class="fa fa-check"></i><b>3.4</b> test read h5 file.</a></li>
<li class="chapter" data-level="3.5" data-path="a-introduction-to-single-cell-sequening.html"><a href="a-introduction-to-single-cell-sequening.html#reproduce-umap-fig-1a"><i class="fa fa-check"></i><b>3.5</b> reproduce UMAP fig 1a</a></li>
<li class="chapter" data-level="3.6" data-path="a-introduction-to-single-cell-sequening.html"><a href="a-introduction-to-single-cell-sequening.html#sessioninfo-1"><i class="fa fa-check"></i><b>3.6</b> sessionInfo</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="pipeline.html"><a href="pipeline.html"><i class="fa fa-check"></i><b>4</b> Pipeline</a>
<ul>
<li class="chapter" data-level="4.1" data-path="pipeline.html"><a href="pipeline.html#数据准备"><i class="fa fa-check"></i><b>4.1</b> 数据准备</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="pipeline.html"><a href="pipeline.html#下载测试数据"><i class="fa fa-check"></i><b>4.1.1</b> 下载测试数据</a></li>
<li class="chapter" data-level="4.1.2" data-path="pipeline.html"><a href="pipeline.html#载入表达矩阵"><i class="fa fa-check"></i><b>4.1.2</b> 载入表达矩阵</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="pipeline.html"><a href="pipeline.html#数据预处理"><i class="fa fa-check"></i><b>4.2</b> 数据预处理</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="pipeline.html"><a href="pipeline.html#构建seurat数据对象"><i class="fa fa-check"></i><b>4.2.1</b> 构建Seurat数据对象</a></li>
<li class="chapter" data-level="4.2.2" data-path="pipeline.html"><a href="pipeline.html#多个单细胞矩阵合并"><i class="fa fa-check"></i><b>4.2.2</b> 多个单细胞矩阵合并</a></li>
<li class="chapter" data-level="4.2.3" data-path="pipeline.html"><a href="pipeline.html#单细胞表达矩阵质控"><i class="fa fa-check"></i><b>4.2.3</b> 单细胞表达矩阵质控</a></li>
<li class="chapter" data-level="4.2.4" data-path="pipeline.html"><a href="pipeline.html#单细胞数据标准化和均一化"><i class="fa fa-check"></i><b>4.2.4</b> 单细胞数据标准化和均一化</a></li>
<li class="chapter" data-level="4.2.5" data-path="pipeline.html"><a href="pipeline.html#数据降维"><i class="fa fa-check"></i><b>4.2.5</b> 数据降维</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="pipeline.html"><a href="pipeline.html#单细胞聚类"><i class="fa fa-check"></i><b>4.3</b> 单细胞聚类</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="pipeline.html"><a href="pipeline.html#聚类分簇"><i class="fa fa-check"></i><b>4.3.1</b> 聚类分簇</a></li>
<li class="chapter" data-level="4.3.2" data-path="pipeline.html"><a href="pipeline.html#单细胞聚类可视化"><i class="fa fa-check"></i><b>4.3.2</b> 单细胞聚类可视化</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="pipeline.html"><a href="pipeline.html#单细胞聚类簇类型鉴定differential-expression-analysis"><i class="fa fa-check"></i><b>4.4</b> 单细胞聚类簇类型鉴定(Differential expression analysis)</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="pipeline.html"><a href="pipeline.html#细胞簇差异基因分析"><i class="fa fa-check"></i><b>4.4.1</b> 细胞簇差异基因分析</a></li>
<li class="chapter" data-level="4.4.2" data-path="pipeline.html"><a href="pipeline.html#marker基因展示"><i class="fa fa-check"></i><b>4.4.2</b> marker基因展示</a></li>
<li class="chapter" data-level="4.4.3" data-path="pipeline.html"><a href="pipeline.html#细胞类型标签注释"><i class="fa fa-check"></i><b>4.4.3</b> 细胞类型标签注释</a></li>
<li class="chapter" data-level="4.4.4" data-path="pipeline.html"><a href="pipeline.html#细胞类型比例可视化"><i class="fa fa-check"></i><b>4.4.4</b> 细胞类型比例可视化</a></li>
<li class="chapter" data-level="4.4.5" data-path="pipeline.html"><a href="pipeline.html#细胞类型marker基因展示"><i class="fa fa-check"></i><b>4.4.5</b> 细胞类型marker基因展示</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="pipeline.html"><a href="pipeline.html#sessioninfo-2"><i class="fa fa-check"></i><b>4.5</b> sessionInfo</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="富集分析.html"><a href="富集分析.html"><i class="fa fa-check"></i><b>5</b> 富集分析</a>
<ul>
<li class="chapter" data-level="5.1" data-path="富集分析.html"><a href="富集分析.html#获取差异基因"><i class="fa fa-check"></i><b>5.1</b> 获取差异基因</a></li>
<li class="chapter" data-level="5.2" data-path="富集分析.html"><a href="富集分析.html#go富集分析"><i class="fa fa-check"></i><b>5.2</b> GO富集分析</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="富集分析.html"><a href="富集分析.html#go富集分析-1"><i class="fa fa-check"></i><b>5.2.1</b> GO富集分析</a></li>
<li class="chapter" data-level="5.2.2" data-path="富集分析.html"><a href="富集分析.html#go富集结果可视化"><i class="fa fa-check"></i><b>5.2.2</b> GO富集结果可视化</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="富集分析.html"><a href="富集分析.html#kegg富集分析"><i class="fa fa-check"></i><b>5.3</b> KEGG富集分析</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="富集分析.html"><a href="富集分析.html#kegg富集分析代码"><i class="fa fa-check"></i><b>5.3.1</b> KEGG富集分析代码</a></li>
<li class="chapter" data-level="5.3.2" data-path="富集分析.html"><a href="富集分析.html#kegg结果可视化"><i class="fa fa-check"></i><b>5.3.2</b> KEGG结果可视化</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="富集分析.html"><a href="富集分析.html#gsea富集分析"><i class="fa fa-check"></i><b>5.4</b> GSEA富集分析</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="富集分析.html"><a href="富集分析.html#gsea富集"><i class="fa fa-check"></i><b>5.4.1</b> GSEA富集</a></li>
<li class="chapter" data-level="5.4.2" data-path="富集分析.html"><a href="富集分析.html#gsea结果可视化"><i class="fa fa-check"></i><b>5.4.2</b> GSEA结果可视化</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="富集分析.html"><a href="富集分析.html#gsva富集分析"><i class="fa fa-check"></i><b>5.5</b> GSVA富集分析</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="富集分析.html"><a href="富集分析.html#gsva富集示例"><i class="fa fa-check"></i><b>5.5.1</b> GSVA富集示例</a></li>
<li class="chapter" data-level="5.5.2" data-path="富集分析.html"><a href="富集分析.html#gsva结果可视化"><i class="fa fa-check"></i><b>5.5.2</b> GSVA结果可视化</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="富集分析.html"><a href="富集分析.html#sessioninfo-3"><i class="fa fa-check"></i><b>5.6</b> sessionInfo</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>6</b> References</a>
<ul>
<li class="chapter" data-level="6.1" data-path="references.html"><a href="references.html#python"><i class="fa fa-check"></i><b>6.1</b> Python</a></li>
<li class="chapter" data-level="6.2" data-path="references.html"><a href="references.html#analysis-of-single-cell-rna-seq-data-分析流程"><i class="fa fa-check"></i><b>6.2</b> Analysis of single cell RNA-seq data 分析流程</a></li>
<li class="chapter" data-level="6.3" data-path="references.html"><a href="references.html#assigning-cell-types-with-singler-细胞类型注释"><i class="fa fa-check"></i><b>6.3</b> Assigning cell types with SingleR # 细胞类型注释</a></li>
<li class="chapter" data-level="6.4" data-path="references.html"><a href="references.html#museum-of-spatial-transcriptomics-空转简史"><i class="fa fa-check"></i><b>6.4</b> Museum of Spatial Transcriptomics # 空转简史</a></li>
<li class="chapter" data-level="6.5" data-path="references.html"><a href="references.html#atac-分析-archr"><i class="fa fa-check"></i><b>6.5</b> ATAC 分析 Archr</a></li>
<li class="chapter" data-level="6.6" data-path="references.html"><a href="references.html#monocl-and-velocyto"><i class="fa fa-check"></i><b>6.6</b> monocl and velocyto</a></li>
<li class="chapter" data-level="6.7" data-path="references.html"><a href="references.html#integraton"><i class="fa fa-check"></i><b>6.7</b> integraton</a></li>
<li class="chapter" data-level="6.8" data-path="references.html"><a href="references.html#gene-set-enrichment"><i class="fa fa-check"></i><b>6.8</b> Gene set enrichment</a></li>
<li class="chapter" data-level="6.9" data-path="references.html"><a href="references.html#转录因子"><i class="fa fa-check"></i><b>6.9</b> 转录因子</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Single Cell Sequencing Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="intro" class="section level1" number="2">
<h1><span class="header-section-number">Chapter 2</span> Introduction</h1>
<p>Based on <code>singleronSC1.html</code></p>
<div id="read-data-library-packages" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Read data Library packages</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="intro.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb1-2"><a href="intro.html#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SeuratData)</span>
<span id="cb1-3"><a href="intro.html#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-4"><a href="intro.html#cb1-4" aria-hidden="true" tabindex="-1"></a>pbmc <span class="ot">&lt;-</span> pbmc3k.final</span>
<span id="cb1-5"><a href="intro.html#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(pbmc<span class="sc">@</span>meta.data)</span></code></pre></div>
<pre><code>##                orig.ident nCount_RNA nFeature_RNA seurat_annotations percent.mt
## AAACATACAACCAC     pbmc3k       2419          779       Memory CD4 T  3.0177759
## AAACATTGAGCTAC     pbmc3k       4903         1352                  B  3.7935958
## AAACATTGATCAGC     pbmc3k       3147         1129       Memory CD4 T  0.8897363
## AAACCGTGCTTCCG     pbmc3k       2639          960         CD14+ Mono  1.7430845
## AAACCGTGTATGCG     pbmc3k        980          521                 NK  1.2244898
## AAACGCACTGGTAC     pbmc3k       2163          781       Memory CD4 T  1.6643551
##                RNA_snn_res.0.5 seurat_clusters
## AAACATACAACCAC               1               1
## AAACATTGAGCTAC               3               3
## AAACATTGATCAGC               1               1
## AAACCGTGCTTCCG               2               2
## AAACCGTGTATGCG               6               6
## AAACGCACTGGTAC               1               1</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="intro.html#cb3-1" aria-hidden="true" tabindex="-1"></a>pbmc<span class="sc">@</span>commands<span class="sc">$</span>FindClusters</span></code></pre></div>
<pre><code>## Command: FindClusters(pbmc3k.final, resolution = 0.5)
## Time: 2020-04-30 12:54:53
## graph.name : RNA_snn 
## modularity.fxn : 1 
## resolution : 0.5 
## method : matrix 
## algorithm : 1 
## n.start : 10 
## n.iter : 10 
## random.seed : 0 
## group.singletons : TRUE 
## verbose : TRUE</code></pre>
</div>
<div id="dimplot-cell-or-cluster" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> DimPlot : cell or cluster</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="intro.html#cb5-1" aria-hidden="true" tabindex="-1"></a>plot1<span class="ot">&lt;-</span><span class="fu">DimPlot</span>(pbmc, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>)</span>
<span id="cb5-2"><a href="intro.html#cb5-2" aria-hidden="true" tabindex="-1"></a>plot2<span class="ot">&lt;-</span><span class="fu">DimPlot</span>(pbmc, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>,<span class="at">label =</span> T)</span>
<span id="cb5-3"><a href="intro.html#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">CombinePlots</span>(<span class="at">plots =</span> <span class="fu">list</span>(plot1, plot2),<span class="at">legend=</span><span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<pre><code>## Warning: CombinePlots is being deprecated. Plots should now be combined using
## the patchwork system.</code></pre>
<p><img src="Single_cell_training_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
</div>
<div id="单细胞可视化" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> 单细胞可视化</h2>
</div>
<div id="dimplot1" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> dimplot1</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="intro.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb7-2"><a href="intro.html#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(pbmc3k.final)<span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&#39;Zhilong&#39;</span>)</span></code></pre></div>
<p><img src="Single_cell_training_files/figure-html/unnamed-chunk-3-1.png" width="672" />
## umap1</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="intro.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb8-2"><a href="intro.html#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SeuratData)</span>
<span id="cb8-3"><a href="intro.html#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb8-4"><a href="intro.html#cb8-4" aria-hidden="true" tabindex="-1"></a>p1<span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(pbmc3k.final)</span>
<span id="cb8-5"><a href="intro.html#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">DimPlot</span>(pbmc3k.final)<span class="sc">$</span>data)</span></code></pre></div>
<pre><code>##                   UMAP_1    UMAP_2        ident
## AAACATACAACCAC -4.232792 -4.152139 Memory CD4 T
## AAACATTGAGCTAC -4.892886 10.985685            B
## AAACATTGATCAGC -5.508639 -7.211088 Memory CD4 T
## AAACCGTGCTTCCG 11.332233  3.161727   CD14+ Mono
## AAACCGTGTATGCG -7.450703  1.092022           NK
## AAACGCACTGGTAC -3.509504 -6.087042 Memory CD4 T</code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="intro.html#cb10-1" aria-hidden="true" tabindex="-1"></a>p2<span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">DimPlot</span>(pbmc3k.final)<span class="sc">$</span>data, <span class="fu">aes</span>(UMAP_1, UMAP_2, <span class="at">fill=</span>ident)) <span class="sc">+</span> </span>
<span id="cb10-2"><a href="intro.html#cb10-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape=</span><span class="dv">21</span>,<span class="at">colour=</span><span class="st">&quot;black&quot;</span>, <span class="at">stroke=</span><span class="fl">0.25</span>, <span class="at">alpha=</span><span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb10-3"><a href="intro.html#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">DimPlot</span>(pbmc3k.final, <span class="at">label =</span> T)<span class="sc">$</span>theme <span class="sc">+</span></span>
<span id="cb10-4"><a href="intro.html#cb10-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span> <span class="fu">NoLegend</span>()</span>
<span id="cb10-5"><a href="intro.html#cb10-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-6"><a href="intro.html#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="intro.html#cb10-7" aria-hidden="true" tabindex="-1"></a>p2<span class="sc">+</span> p1</span></code></pre></div>
<p><img src="Single_cell_training_files/figure-html/unnamed-chunk-5-1.png" width="672" />
## umap2</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="intro.html#cb11-1" aria-hidden="true" tabindex="-1"></a>pbmc<span class="ot">&lt;-</span> pbmc3k.final</span>
<span id="cb11-2"><a href="intro.html#cb11-2" aria-hidden="true" tabindex="-1"></a>pbmc<span class="ot">&lt;-</span><span class="fu">AddMetaData</span>(pbmc,pbmc<span class="sc">@</span>reductions<span class="sc">$</span>umap<span class="sc">@</span>cell.embeddings,<span class="at">col.name =</span> <span class="fu">colnames</span>(pbmc<span class="sc">@</span>reductions<span class="sc">$</span>umap<span class="sc">@</span>cell.embeddings))</span>
<span id="cb11-3"><a href="intro.html#cb11-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-4"><a href="intro.html#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(pbmc<span class="sc">@</span>meta.data)</span></code></pre></div>
<pre><code>##                orig.ident nCount_RNA nFeature_RNA seurat_annotations percent.mt
## AAACATACAACCAC     pbmc3k       2419          779       Memory CD4 T  3.0177759
## AAACATTGAGCTAC     pbmc3k       4903         1352                  B  3.7935958
## AAACATTGATCAGC     pbmc3k       3147         1129       Memory CD4 T  0.8897363
## AAACCGTGCTTCCG     pbmc3k       2639          960         CD14+ Mono  1.7430845
## AAACCGTGTATGCG     pbmc3k        980          521                 NK  1.2244898
## AAACGCACTGGTAC     pbmc3k       2163          781       Memory CD4 T  1.6643551
##                RNA_snn_res.0.5 seurat_clusters    UMAP_1    UMAP_2
## AAACATACAACCAC               1               1 -4.232792 -4.152139
## AAACATTGAGCTAC               3               3 -4.892886 10.985685
## AAACATTGATCAGC               1               1 -5.508639 -7.211088
## AAACCGTGCTTCCG               2               2 11.332233  3.161727
## AAACCGTGTATGCG               6               6 -7.450703  1.092022
## AAACGCACTGGTAC               1               1 -3.509504 -6.087042</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="intro.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb13-2"><a href="intro.html#cb13-2" aria-hidden="true" tabindex="-1"></a>class_avg <span class="ot">&lt;-</span> pbmc<span class="sc">@</span>meta.data <span class="sc">%&gt;%</span></span>
<span id="cb13-3"><a href="intro.html#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(seurat_annotations) <span class="sc">%&gt;%</span></span>
<span id="cb13-4"><a href="intro.html#cb13-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb13-5"><a href="intro.html#cb13-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">UMAP_1 =</span> <span class="fu">median</span>(UMAP_1),</span>
<span id="cb13-6"><a href="intro.html#cb13-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">UMAP_2 =</span> <span class="fu">median</span>(UMAP_2)</span>
<span id="cb13-7"><a href="intro.html#cb13-7" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb13-8"><a href="intro.html#cb13-8" aria-hidden="true" tabindex="-1"></a>allcolour<span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;#DC143C&quot;</span>,<span class="st">&quot;#0000FF&quot;</span>,<span class="st">&quot;#20B2AA&quot;</span>,<span class="st">&quot;#FFA500&quot;</span>,<span class="st">&quot;#9370DB&quot;</span>,<span class="st">&quot;#98FB98&quot;</span>,<span class="st">&quot;#F08080&quot;</span>,<span class="st">&quot;#1E90FF&quot;</span>,<span class="st">&quot;#7CFC00&quot;</span>,<span class="st">&quot;#FFFF00&quot;</span>,</span>
<span id="cb13-9"><a href="intro.html#cb13-9" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;#808000&quot;</span>,<span class="st">&quot;#FF00FF&quot;</span>,<span class="st">&quot;#FA8072&quot;</span>,<span class="st">&quot;#7B68EE&quot;</span>,<span class="st">&quot;#9400D3&quot;</span>,<span class="st">&quot;#800080&quot;</span>,<span class="st">&quot;#A0522D&quot;</span>,<span class="st">&quot;#D2B48C&quot;</span>,<span class="st">&quot;#D2691E&quot;</span>,<span class="st">&quot;#87CEEB&quot;</span>,<span class="st">&quot;#40E0D0&quot;</span>,<span class="st">&quot;#5F9EA0&quot;</span>,</span>
<span id="cb13-10"><a href="intro.html#cb13-10" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;#FF1493&quot;</span>,<span class="st">&quot;#0000CD&quot;</span>,<span class="st">&quot;#008B8B&quot;</span>,<span class="st">&quot;#FFE4B5&quot;</span>,<span class="st">&quot;#8A2BE2&quot;</span>,<span class="st">&quot;#228B22&quot;</span>,<span class="st">&quot;#E9967A&quot;</span>,<span class="st">&quot;#4682B4&quot;</span>,<span class="st">&quot;#32CD32&quot;</span>,<span class="st">&quot;#F0E68C&quot;</span>,<span class="st">&quot;#FFFFE0&quot;</span>,<span class="st">&quot;#EE82EE&quot;</span>,</span>
<span id="cb13-11"><a href="intro.html#cb13-11" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;#FF6347&quot;</span>,<span class="st">&quot;#6A5ACD&quot;</span>,<span class="st">&quot;#9932CC&quot;</span>,<span class="st">&quot;#8B008B&quot;</span>,<span class="st">&quot;#8B4513&quot;</span>,<span class="st">&quot;#DEB887&quot;</span>)</span>
<span id="cb13-12"><a href="intro.html#cb13-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-13"><a href="intro.html#cb13-13" aria-hidden="true" tabindex="-1"></a>umap <span class="ot">&lt;-</span>  <span class="fu">ggplot</span>(pbmc<span class="sc">@</span>meta.data ,<span class="fu">aes</span>(<span class="at">x=</span>UMAP_1,<span class="at">y=</span>UMAP_2))<span class="sc">+</span></span>
<span id="cb13-14"><a href="intro.html#cb13-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color=</span>seurat_annotations))<span class="sc">+</span></span>
<span id="cb13-15"><a href="intro.html#cb13-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> allcolour)<span class="sc">+</span></span>
<span id="cb13-16"><a href="intro.html#cb13-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> seurat_annotations), <span class="at">data =</span> class_avg)<span class="sc">+</span></span>
<span id="cb13-17"><a href="intro.html#cb13-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text=</span><span class="fu">element_text</span>(<span class="at">family=</span><span class="st">&quot;Arial&quot;</span>,<span class="at">size=</span><span class="dv">18</span>)) <span class="sc">+</span></span>
<span id="cb13-18"><a href="intro.html#cb13-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill=</span><span class="st">&#39;white&#39;</span>, <span class="at">colour=</span><span class="st">&#39;black&#39;</span>), </span>
<span id="cb13-19"><a href="intro.html#cb13-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid=</span><span class="fu">element_blank</span>(), <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">color=</span><span class="st">&#39;black&#39;</span>,</span>
<span id="cb13-20"><a href="intro.html#cb13-20" aria-hidden="true" tabindex="-1"></a>                                                              <span class="at">family=</span><span class="st">&quot;Arial&quot;</span>,<span class="at">size=</span><span class="dv">18</span>),<span class="at">axis.ticks.length =</span> <span class="fu">unit</span>(<span class="fl">0.4</span>,<span class="st">&quot;lines&quot;</span>), </span>
<span id="cb13-21"><a href="intro.html#cb13-21" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks =</span> <span class="fu">element_line</span>(<span class="at">color=</span><span class="st">&#39;black&#39;</span>), </span>
<span id="cb13-22"><a href="intro.html#cb13-22" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.margin =</span> <span class="fu">unit</span>(<span class="fl">0.6</span>,<span class="st">&quot;lines&quot;</span>),</span>
<span id="cb13-23"><a href="intro.html#cb13-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>), </span>
<span id="cb13-24"><a href="intro.html#cb13-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x=</span><span class="fu">element_text</span>(<span class="at">colour=</span><span class="st">&#39;black&#39;</span>, <span class="at">size=</span><span class="dv">18</span>),</span>
<span id="cb13-25"><a href="intro.html#cb13-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y=</span><span class="fu">element_text</span>(<span class="at">colour=</span><span class="st">&#39;black&#39;</span>, <span class="at">size=</span><span class="dv">18</span>),</span>
<span id="cb13-26"><a href="intro.html#cb13-26" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text=</span><span class="fu">element_text</span>(<span class="at">colour=</span><span class="st">&#39;black&#39;</span>,<span class="at">size=</span><span class="dv">18</span>),</span>
<span id="cb13-27"><a href="intro.html#cb13-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title=</span><span class="fu">element_blank</span>(),</span>
<span id="cb13-28"><a href="intro.html#cb13-28" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">family=</span><span class="st">&quot;Arial&quot;</span>, <span class="at">size=</span><span class="dv">18</span>),</span>
<span id="cb13-29"><a href="intro.html#cb13-29" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key=</span><span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb13-30"><a href="intro.html#cb13-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">22</span>,<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>,<span class="at">face =</span> <span class="st">&quot;bold&quot;</span>))  <span class="sc">+</span> </span>
<span id="cb13-31"><a href="intro.html#cb13-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">colour =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">size=</span><span class="dv">5</span>)))</span></code></pre></div>
<pre><code>## Warning: `axis.ticks.margin` is deprecated. Please set `margin` property of
## `axis.text` instead</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="intro.html#cb15-1" aria-hidden="true" tabindex="-1"></a>umap</span></code></pre></div>
<p><img src="Single_cell_training_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="intro.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)<span class="co">#可重复性的</span></span>
<span id="cb16-2"><a href="intro.html#cb16-2" aria-hidden="true" tabindex="-1"></a>pbmc<span class="sc">$</span>replicate <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">c</span>(<span class="st">&quot;rep1&quot;</span>, <span class="st">&quot;rep2&quot;</span>), <span class="at">size =</span> <span class="fu">ncol</span>(pbmc), <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb16-3"><a href="intro.html#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(pbmc<span class="sc">@</span>meta.data)</span></code></pre></div>
<pre><code>##                orig.ident nCount_RNA nFeature_RNA seurat_annotations percent.mt
## AAACATACAACCAC     pbmc3k       2419          779       Memory CD4 T  3.0177759
## AAACATTGAGCTAC     pbmc3k       4903         1352                  B  3.7935958
## AAACATTGATCAGC     pbmc3k       3147         1129       Memory CD4 T  0.8897363
## AAACCGTGCTTCCG     pbmc3k       2639          960         CD14+ Mono  1.7430845
## AAACCGTGTATGCG     pbmc3k        980          521                 NK  1.2244898
## AAACGCACTGGTAC     pbmc3k       2163          781       Memory CD4 T  1.6643551
##                RNA_snn_res.0.5 seurat_clusters    UMAP_1    UMAP_2 replicate
## AAACATACAACCAC               1               1 -4.232792 -4.152139      rep1
## AAACATTGAGCTAC               3               3 -4.892886 10.985685      rep1
## AAACATTGATCAGC               1               1 -5.508639 -7.211088      rep1
## AAACCGTGCTTCCG               2               2 11.332233  3.161727      rep1
## AAACCGTGTATGCG               6               6 -7.450703  1.092022      rep2
## AAACGCACTGGTAC               1               1 -3.509504 -6.087042      rep2</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="intro.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">as.data.frame</span>(<span class="fu">prop.table</span>(<span class="fu">table</span>(<span class="fu">Idents</span>(pbmc), pbmc<span class="sc">@</span>meta.data[,<span class="st">&quot;replicate&quot;</span>]), <span class="at">margin =</span> <span class="dv">2</span>))<span class="ot">-&gt;</span> pdf <span class="ot">-&gt;</span> td</span>
<span id="cb18-2"><a href="intro.html#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb18-3"><a href="intro.html#cb18-3" aria-hidden="true" tabindex="-1"></a>allcolour<span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;#DC143C&quot;</span>,<span class="st">&quot;#0000FF&quot;</span>,<span class="st">&quot;#20B2AA&quot;</span>,<span class="st">&quot;#FFA500&quot;</span>,<span class="st">&quot;#9370DB&quot;</span>,<span class="st">&quot;#98FB98&quot;</span>,<span class="st">&quot;#F08080&quot;</span>,<span class="st">&quot;#1E90FF&quot;</span>,<span class="st">&quot;#7CFC00&quot;</span>,<span class="st">&quot;#FFFF00&quot;</span>,<span class="st">&quot;#808000&quot;</span>,<span class="st">&quot;#FF00FF&quot;</span>,<span class="st">&quot;#FA8072&quot;</span>,<span class="st">&quot;#7B68EE&quot;</span>,<span class="st">&quot;#9400D3&quot;</span>,<span class="st">&quot;#800080&quot;</span>,<span class="st">&quot;#A0522D&quot;</span>,<span class="st">&quot;#D2B48C&quot;</span>,<span class="st">&quot;#D2691E&quot;</span>,<span class="st">&quot;#87CEEB&quot;</span>,<span class="st">&quot;#40E0D0&quot;</span>,<span class="st">&quot;#5F9EA0&quot;</span>,<span class="st">&quot;#FF1493&quot;</span>,<span class="st">&quot;#0000CD&quot;</span>,<span class="st">&quot;#008B8B&quot;</span>,<span class="st">&quot;#FFE4B5&quot;</span>,<span class="st">&quot;#8A2BE2&quot;</span>,<span class="st">&quot;#228B22&quot;</span>,<span class="st">&quot;#E9967A&quot;</span>,<span class="st">&quot;#4682B4&quot;</span>,<span class="st">&quot;#32CD32&quot;</span>,<span class="st">&quot;#F0E68C&quot;</span>,<span class="st">&quot;#FFFFE0&quot;</span>,<span class="st">&quot;#EE82EE&quot;</span>,<span class="st">&quot;#FF6347&quot;</span>,<span class="st">&quot;#6A5ACD&quot;</span>,<span class="st">&quot;#9932CC&quot;</span>,<span class="st">&quot;#8B008B&quot;</span>,<span class="st">&quot;#8B4513&quot;</span>,<span class="st">&quot;#DEB887&quot;</span>)</span>
<span id="cb18-4"><a href="intro.html#cb18-4" aria-hidden="true" tabindex="-1"></a>allcolour <span class="ot">-&gt;</span> colour1</span>
<span id="cb18-5"><a href="intro.html#cb18-5" aria-hidden="true" tabindex="-1"></a>plt<span class="ot">&lt;-</span> <span class="fu">ggplot</span>(td,<span class="fu">aes</span>(<span class="at">x=</span>td[,<span class="dv">2</span>],<span class="at">y=</span>td[,<span class="dv">3</span>],<span class="at">fill=</span>td[,<span class="dv">1</span>]))<span class="sc">+</span></span>
<span id="cb18-6"><a href="intro.html#cb18-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">position =</span> <span class="st">&#39;stack&#39;</span>,<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="sc">+</span></span>
<span id="cb18-7"><a href="intro.html#cb18-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;replicate&quot;</span>,<span class="at">y=</span><span class="st">&quot;Cells Ratio&quot;</span>)<span class="sc">+</span></span>
<span id="cb18-8"><a href="intro.html#cb18-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.background=</span><span class="fu">element_rect</span>(<span class="at">fill=</span><span class="st">&#39;transparent&#39;</span>, <span class="at">color=</span><span class="st">&#39;black&#39;</span>),</span>
<span id="cb18-9"><a href="intro.html#cb18-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key=</span><span class="fu">element_rect</span>(<span class="at">fill=</span><span class="st">&#39;transparent&#39;</span>, <span class="at">color=</span><span class="st">&#39;transparent&#39;</span>),<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">color=</span><span class="st">&quot;black&quot;</span>))<span class="sc">+</span></span>
<span id="cb18-10"><a href="intro.html#cb18-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand=</span><span class="fu">c</span>(<span class="fl">0.001</span>,<span class="fl">0.001</span>))<span class="sc">+</span></span>
<span id="cb18-11"><a href="intro.html#cb18-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>colour1)<span class="sc">+</span></span>
<span id="cb18-12"><a href="intro.html#cb18-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">keywidth =</span> <span class="dv">1</span>, <span class="at">keyheight =</span> <span class="dv">1</span>,<span class="at">ncol=</span><span class="dv">1</span>,<span class="at">title =</span> <span class="st">&#39;Cluster&#39;</span>))</span>
<span id="cb18-13"><a href="intro.html#cb18-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-14"><a href="intro.html#cb18-14" aria-hidden="true" tabindex="-1"></a>plt</span></code></pre></div>
<p><img src="Single_cell_training_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
</div>
<div id="比例" class="section level2" number="2.5">
<h2><span class="header-section-number">2.5</span> 比例</h2>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="intro.html#cb19-1" aria-hidden="true" tabindex="-1"></a>col <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;red&#39;</span>,<span class="st">&#39;blue&#39;</span>,<span class="st">&#39;cyan4&#39;</span>,<span class="st">&#39;yellow&#39;</span>,<span class="st">&#39;skyblue&#39;</span>,<span class="st">&#39;purple&#39;</span>, <span class="st">&#39;olivedrab3&#39;</span>,<span class="st">&#39;springgreen&#39;</span>,<span class="st">&#39;olivedrab1&#39;</span>,<span class="st">&#39;forestgreen&#39;</span>,<span class="st">&#39;darkslateblue&#39;</span>)</span>
<span id="cb19-2"><a href="intro.html#cb19-2" aria-hidden="true" tabindex="-1"></a>p3<span class="ot">&lt;-</span>  <span class="fu">ggplot</span>(td,<span class="fu">aes</span>(<span class="at">x=</span>td[,<span class="dv">2</span>],<span class="at">y=</span>td[,<span class="dv">3</span>],<span class="at">fill=</span>td[,<span class="dv">1</span>])) <span class="sc">+</span></span>
<span id="cb19-3"><a href="intro.html#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="at">position =</span> <span class="st">&quot;fill&quot;</span>)<span class="sc">+</span></span>
<span id="cb19-4"><a href="intro.html#cb19-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>col) <span class="sc">+</span><span class="fu">coord_flip</span>()</span>
<span id="cb19-5"><a href="intro.html#cb19-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-6"><a href="intro.html#cb19-6" aria-hidden="true" tabindex="-1"></a>p4<span class="ot">&lt;-</span> <span class="fu">ggplot</span>(td,<span class="fu">aes</span>(<span class="at">x=</span>td[,<span class="dv">2</span>],<span class="at">y=</span>td[,<span class="dv">3</span>],<span class="at">fill=</span>td[,<span class="dv">1</span>])) <span class="sc">+</span></span>
<span id="cb19-7"><a href="intro.html#cb19-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="at">position =</span> <span class="st">&quot;fill&quot;</span>)<span class="sc">+</span></span>
<span id="cb19-8"><a href="intro.html#cb19-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>col) <span class="sc">+</span><span class="fu">facet_wrap</span>(<span class="sc">~</span> td[,<span class="dv">2</span>], <span class="at">scales =</span> <span class="st">&#39;free&#39;</span>)</span>
<span id="cb19-9"><a href="intro.html#cb19-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-10"><a href="intro.html#cb19-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-11"><a href="intro.html#cb19-11" aria-hidden="true" tabindex="-1"></a>p5<span class="ot">&lt;-</span> <span class="fu">ggplot</span>(td,<span class="fu">aes</span>(<span class="at">x=</span>td[,<span class="dv">2</span>],<span class="at">y=</span>td[,<span class="dv">3</span>],<span class="at">fill=</span>td[,<span class="dv">1</span>])) <span class="sc">+</span></span>
<span id="cb19-12"><a href="intro.html#cb19-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="at">position =</span> <span class="st">&quot;fill&quot;</span>)<span class="sc">+</span></span>
<span id="cb19-13"><a href="intro.html#cb19-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>col) <span class="sc">+</span><span class="fu">coord_polar</span>(<span class="at">theta =</span> <span class="st">&#39;y&#39;</span>)</span>
<span id="cb19-14"><a href="intro.html#cb19-14" aria-hidden="true" tabindex="-1"></a>p5</span></code></pre></div>
<p><img src="Single_cell_training_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
</div>
<div id="数据转化" class="section level2" number="2.6">
<h2><span class="header-section-number">2.6</span> 数据转化</h2>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="intro.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb20-2"><a href="intro.html#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>( pbmc<span class="sc">@</span>meta.data)</span></code></pre></div>
<pre><code>##                orig.ident nCount_RNA nFeature_RNA seurat_annotations percent.mt
## AAACATACAACCAC     pbmc3k       2419          779       Memory CD4 T  3.0177759
## AAACATTGAGCTAC     pbmc3k       4903         1352                  B  3.7935958
## AAACATTGATCAGC     pbmc3k       3147         1129       Memory CD4 T  0.8897363
## AAACCGTGCTTCCG     pbmc3k       2639          960         CD14+ Mono  1.7430845
## AAACCGTGTATGCG     pbmc3k        980          521                 NK  1.2244898
## AAACGCACTGGTAC     pbmc3k       2163          781       Memory CD4 T  1.6643551
##                RNA_snn_res.0.5 seurat_clusters    UMAP_1    UMAP_2 replicate
## AAACATACAACCAC               1               1 -4.232792 -4.152139      rep1
## AAACATTGAGCTAC               3               3 -4.892886 10.985685      rep1
## AAACATTGATCAGC               1               1 -5.508639 -7.211088      rep1
## AAACCGTGCTTCCG               2               2 11.332233  3.161727      rep1
## AAACCGTGTATGCG               6               6 -7.450703  1.092022      rep2
## AAACGCACTGGTAC               1               1 -3.509504 -6.087042      rep2</code></pre>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="intro.html#cb22-1" aria-hidden="true" tabindex="-1"></a>td_tidy<span class="ot">&lt;-</span><span class="fu">spread</span>(<span class="at">data =</span> td, <span class="at">key =</span> <span class="st">&quot;Var1&quot;</span>, <span class="at">value =</span> <span class="st">&quot;Freq&quot;</span>)</span>
<span id="cb22-2"><a href="intro.html#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(td_tidy)</span></code></pre></div>
<pre><code>##   Var2 Naive CD4 T Memory CD4 T CD14+ Mono         B      CD8 T FCGR3A+ Mono
## 1 rep1   0.2626113    0.1847181  0.1721068 0.1283383 0.11424332   0.06008902
## 2 rep2   0.2658915    0.1813953  0.1922481 0.1325581 0.09069767   0.06279070
##           NK         DC    Platelet
## 1 0.06008902 0.01335312 0.004451039
## 2 0.05736434 0.01085271 0.006201550</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="intro.html#cb24-1" aria-hidden="true" tabindex="-1"></a>td_tidy2 <span class="ot">&lt;-</span> <span class="fu">gather</span>(<span class="at">data =</span> td_tidy,<span class="at">key =</span> <span class="st">&quot;Var1&quot;</span>, <span class="at">value =</span> <span class="st">&quot;Freq&quot;</span>, <span class="sc">-</span>Var2)</span>
<span id="cb24-2"><a href="intro.html#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(td_tidy2)</span></code></pre></div>
<pre><code>##   Var2         Var1      Freq
## 1 rep1  Naive CD4 T 0.2626113
## 2 rep2  Naive CD4 T 0.2658915
## 3 rep1 Memory CD4 T 0.1847181
## 4 rep2 Memory CD4 T 0.1813953
## 5 rep1   CD14+ Mono 0.1721068
## 6 rep2   CD14+ Mono 0.1922481</code></pre>
</div>
<div id="sessioninfo" class="section level2" number="2.7">
<h2><span class="header-section-number">2.7</span> sessionInfo</h2>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="intro.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.4 (2021-02-15)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.10
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] forcats_0.5.1           stringr_1.4.0           dplyr_1.0.5            
##  [4] purrr_0.3.4             readr_1.4.0             tidyr_1.1.3            
##  [7] tibble_3.1.1            ggplot2_3.3.3           tidyverse_1.3.1        
## [10] pbmc3k.SeuratData_3.1.4 SeuratData_0.2.1        SeuratObject_4.0.0     
## [13] Seurat_4.0.1           
## 
## loaded via a namespace (and not attached):
##   [1] readxl_1.3.1          backports_1.2.1       plyr_1.8.6           
##   [4] igraph_1.2.6          lazyeval_0.2.2        splines_4.0.4        
##   [7] listenv_0.8.0         scattermore_0.7       digest_0.6.27        
##  [10] htmltools_0.5.1.1     fansi_0.4.2           magrittr_2.0.1       
##  [13] tensor_1.5            cluster_2.1.2         ROCR_1.0-11          
##  [16] globals_0.14.0        modelr_0.1.8          matrixStats_0.58.0   
##  [19] spatstat.sparse_2.0-0 colorspace_2.0-0      rvest_1.0.0          
##  [22] rappdirs_0.3.3        ggrepel_0.9.1         haven_2.4.0          
##  [25] xfun_0.22             crayon_1.4.1          jsonlite_1.7.2       
##  [28] spatstat.data_2.1-0   survival_3.2-10       zoo_1.8-9            
##  [31] glue_1.4.2            polyclip_1.10-0       gtable_0.3.0         
##  [34] leiden_0.3.7          future.apply_1.7.0    abind_1.4-5          
##  [37] scales_1.1.1          DBI_1.1.1             miniUI_0.1.1.1       
##  [40] Rcpp_1.0.6            viridisLite_0.4.0     xtable_1.8-4         
##  [43] reticulate_1.18       spatstat.core_2.1-2   htmlwidgets_1.5.3    
##  [46] httr_1.4.2            RColorBrewer_1.1-2    ellipsis_0.3.1       
##  [49] ica_1.0-2             pkgconfig_2.0.3       farver_2.1.0         
##  [52] sass_0.3.1            uwot_0.1.10           dbplyr_2.1.1         
##  [55] deldir_0.2-10         utf8_1.2.1            tidyselect_1.1.0     
##  [58] labeling_0.4.2        rlang_0.4.10          reshape2_1.4.4       
##  [61] later_1.1.0.1         munsell_0.5.0         cellranger_1.1.0     
##  [64] tools_4.0.4           cli_2.4.0             generics_0.1.0       
##  [67] broom_0.7.6           ggridges_0.5.3        evaluate_0.14        
##  [70] fastmap_1.1.0         yaml_2.2.1            goftest_1.2-2        
##  [73] knitr_1.32            fs_1.5.0              fitdistrplus_1.1-3   
##  [76] RANN_2.6.1            pbapply_1.4-3         future_1.21.0        
##  [79] nlme_3.1-152          mime_0.10             xml2_1.3.2           
##  [82] compiler_4.0.4        rstudioapi_0.13       plotly_4.9.3         
##  [85] png_0.1-7             spatstat.utils_2.1-0  reprex_2.0.0         
##  [88] bslib_0.2.4           stringi_1.5.3         highr_0.9            
##  [91] lattice_0.20-41       Matrix_1.3-2          vctrs_0.3.7          
##  [94] pillar_1.6.0          lifecycle_1.0.0       spatstat.geom_2.1-0  
##  [97] lmtest_0.9-38         jquerylib_0.1.3       RcppAnnoy_0.0.18     
## [100] data.table_1.14.0     cowplot_1.1.1         irlba_2.3.3          
## [103] httpuv_1.5.5          patchwork_1.1.1       R6_2.5.0             
## [106] bookdown_0.21         promises_1.2.0.1      KernSmooth_2.23-18   
## [109] gridExtra_2.3         parallelly_1.24.0     codetools_0.2-18     
## [112] MASS_7.3-53.1         assertthat_0.2.1      withr_2.4.2          
## [115] sctransform_0.3.2     mgcv_1.8-35           parallel_4.0.4       
## [118] hms_1.0.0             grid_4.0.4            rpart_4.1-15         
## [121] rmarkdown_2.7         Rtsne_0.15            shiny_1.6.0          
## [124] lubridate_1.7.10</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="a-introduction-to-single-cell-sequening.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Single_cell_training.pdf", "Single_cell_training.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
