<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Single-cell transcriptomic landscape of human blood cells | Single Cell Sequencing Analysis</title>
  <meta name="description" content="Training Class of Single Cell Sequencing Analysis." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Single-cell transcriptomic landscape of human blood cells | Single Cell Sequencing Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Training Class of Single Cell Sequencing Analysis." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Single-cell transcriptomic landscape of human blood cells | Single Cell Sequencing Analysis" />
  
  <meta name="twitter:description" content="Training Class of Single Cell Sequencing Analysis." />
  

<meta name="author" content="Zhilong Jia" />


<meta name="date" content="2021-05-14" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="ref-httpsgithub-comsqjincellchat.html"/>
<link rel="next" href="references.html"/>
<script src="libs/header-attrs-2.7/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://zhilongjia.github.io/scRNA">scRNA Note</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Content</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="intro.html"><a href="intro.html#read-data-library-packages"><i class="fa fa-check"></i><b>2.1</b> Read data Library packages</a></li>
<li class="chapter" data-level="2.2" data-path="intro.html"><a href="intro.html#dimplot-cell-or-cluster"><i class="fa fa-check"></i><b>2.2</b> DimPlot : cell or cluster</a></li>
<li class="chapter" data-level="2.3" data-path="intro.html"><a href="intro.html#单细胞可视化"><i class="fa fa-check"></i><b>2.3</b> 单细胞可视化</a></li>
<li class="chapter" data-level="2.4" data-path="intro.html"><a href="intro.html#dimplot1"><i class="fa fa-check"></i><b>2.4</b> dimplot1</a></li>
<li class="chapter" data-level="2.5" data-path="intro.html"><a href="intro.html#比例"><i class="fa fa-check"></i><b>2.5</b> 比例</a></li>
<li class="chapter" data-level="2.6" data-path="intro.html"><a href="intro.html#数据转化"><i class="fa fa-check"></i><b>2.6</b> 数据转化</a></li>
<li class="chapter" data-level="2.7" data-path="intro.html"><a href="intro.html#sessioninfo"><i class="fa fa-check"></i><b>2.7</b> sessionInfo</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="a-introduction-to-single-cell-sequening.html"><a href="a-introduction-to-single-cell-sequening.html"><i class="fa fa-check"></i><b>3</b> A introduction to Single Cell Sequening</a>
<ul>
<li class="chapter" data-level="3.1" data-path="a-introduction-to-single-cell-sequening.html"><a href="a-introduction-to-single-cell-sequening.html#data-format-of-seurat"><i class="fa fa-check"></i><b>3.1</b> data format of Seurat</a></li>
<li class="chapter" data-level="3.2" data-path="a-introduction-to-single-cell-sequening.html"><a href="a-introduction-to-single-cell-sequening.html#data-access"><i class="fa fa-check"></i><b>3.2</b> Data Access</a></li>
<li class="chapter" data-level="3.3" data-path="a-introduction-to-single-cell-sequening.html"><a href="a-introduction-to-single-cell-sequening.html#methods"><i class="fa fa-check"></i><b>3.3</b> Methods</a></li>
<li class="chapter" data-level="3.4" data-path="a-introduction-to-single-cell-sequening.html"><a href="a-introduction-to-single-cell-sequening.html#test-read-h5-file."><i class="fa fa-check"></i><b>3.4</b> test read h5 file.</a></li>
<li class="chapter" data-level="3.5" data-path="a-introduction-to-single-cell-sequening.html"><a href="a-introduction-to-single-cell-sequening.html#reproduce-umap-fig-1a"><i class="fa fa-check"></i><b>3.5</b> reproduce UMAP fig 1a</a></li>
<li class="chapter" data-level="3.6" data-path="a-introduction-to-single-cell-sequening.html"><a href="a-introduction-to-single-cell-sequening.html#sessioninfo-1"><i class="fa fa-check"></i><b>3.6</b> sessionInfo</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="pipeline.html"><a href="pipeline.html"><i class="fa fa-check"></i><b>4</b> Pipeline</a>
<ul>
<li class="chapter" data-level="4.1" data-path="pipeline.html"><a href="pipeline.html#数据准备"><i class="fa fa-check"></i><b>4.1</b> 数据准备</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="pipeline.html"><a href="pipeline.html#下载测试数据"><i class="fa fa-check"></i><b>4.1.1</b> 下载测试数据</a></li>
<li class="chapter" data-level="4.1.2" data-path="pipeline.html"><a href="pipeline.html#载入表达矩阵"><i class="fa fa-check"></i><b>4.1.2</b> 载入表达矩阵</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="pipeline.html"><a href="pipeline.html#数据预处理"><i class="fa fa-check"></i><b>4.2</b> 数据预处理</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="pipeline.html"><a href="pipeline.html#构建seurat数据对象"><i class="fa fa-check"></i><b>4.2.1</b> 构建Seurat数据对象</a></li>
<li class="chapter" data-level="4.2.2" data-path="pipeline.html"><a href="pipeline.html#多个单细胞矩阵合并"><i class="fa fa-check"></i><b>4.2.2</b> 多个单细胞矩阵合并</a></li>
<li class="chapter" data-level="4.2.3" data-path="pipeline.html"><a href="pipeline.html#单细胞表达矩阵质控"><i class="fa fa-check"></i><b>4.2.3</b> 单细胞表达矩阵质控</a></li>
<li class="chapter" data-level="4.2.4" data-path="pipeline.html"><a href="pipeline.html#单细胞数据标准化和均一化"><i class="fa fa-check"></i><b>4.2.4</b> 单细胞数据标准化和均一化</a></li>
<li class="chapter" data-level="4.2.5" data-path="pipeline.html"><a href="pipeline.html#数据降维"><i class="fa fa-check"></i><b>4.2.5</b> 数据降维</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="pipeline.html"><a href="pipeline.html#单细胞聚类"><i class="fa fa-check"></i><b>4.3</b> 单细胞聚类</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="pipeline.html"><a href="pipeline.html#聚类分簇"><i class="fa fa-check"></i><b>4.3.1</b> 聚类分簇</a></li>
<li class="chapter" data-level="4.3.2" data-path="pipeline.html"><a href="pipeline.html#单细胞聚类可视化"><i class="fa fa-check"></i><b>4.3.2</b> 单细胞聚类可视化</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="pipeline.html"><a href="pipeline.html#单细胞聚类簇类型鉴定differential-expression-analysis"><i class="fa fa-check"></i><b>4.4</b> 单细胞聚类簇类型鉴定(Differential expression analysis)</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="pipeline.html"><a href="pipeline.html#细胞簇差异基因分析"><i class="fa fa-check"></i><b>4.4.1</b> 细胞簇差异基因分析</a></li>
<li class="chapter" data-level="4.4.2" data-path="pipeline.html"><a href="pipeline.html#marker基因展示"><i class="fa fa-check"></i><b>4.4.2</b> marker基因展示</a></li>
<li class="chapter" data-level="4.4.3" data-path="pipeline.html"><a href="pipeline.html#细胞类型标签注释"><i class="fa fa-check"></i><b>4.4.3</b> 细胞类型标签注释</a></li>
<li class="chapter" data-level="4.4.4" data-path="pipeline.html"><a href="pipeline.html#细胞类型比例可视化"><i class="fa fa-check"></i><b>4.4.4</b> 细胞类型比例可视化</a></li>
<li class="chapter" data-level="4.4.5" data-path="pipeline.html"><a href="pipeline.html#细胞类型marker基因展示"><i class="fa fa-check"></i><b>4.4.5</b> 细胞类型marker基因展示</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="pipeline.html"><a href="pipeline.html#sessioninfo-2"><i class="fa fa-check"></i><b>4.5</b> sessionInfo</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="富集分析.html"><a href="富集分析.html"><i class="fa fa-check"></i><b>5</b> 富集分析</a>
<ul>
<li class="chapter" data-level="5.1" data-path="富集分析.html"><a href="富集分析.html#获取差异基因"><i class="fa fa-check"></i><b>5.1</b> 获取差异基因</a></li>
<li class="chapter" data-level="5.2" data-path="富集分析.html"><a href="富集分析.html#go富集分析"><i class="fa fa-check"></i><b>5.2</b> GO富集分析</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="富集分析.html"><a href="富集分析.html#go富集分析-1"><i class="fa fa-check"></i><b>5.2.1</b> GO富集分析</a></li>
<li class="chapter" data-level="5.2.2" data-path="富集分析.html"><a href="富集分析.html#go富集结果可视化"><i class="fa fa-check"></i><b>5.2.2</b> GO富集结果可视化</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="富集分析.html"><a href="富集分析.html#kegg富集分析"><i class="fa fa-check"></i><b>5.3</b> KEGG富集分析</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="富集分析.html"><a href="富集分析.html#kegg富集分析代码"><i class="fa fa-check"></i><b>5.3.1</b> KEGG富集分析代码</a></li>
<li class="chapter" data-level="5.3.2" data-path="富集分析.html"><a href="富集分析.html#kegg结果可视化"><i class="fa fa-check"></i><b>5.3.2</b> KEGG结果可视化</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="富集分析.html"><a href="富集分析.html#gsea富集分析"><i class="fa fa-check"></i><b>5.4</b> GSEA富集分析</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="富集分析.html"><a href="富集分析.html#gsea富集"><i class="fa fa-check"></i><b>5.4.1</b> GSEA富集</a></li>
<li class="chapter" data-level="5.4.2" data-path="富集分析.html"><a href="富集分析.html#gsea结果可视化"><i class="fa fa-check"></i><b>5.4.2</b> GSEA结果可视化</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="富集分析.html"><a href="富集分析.html#gsva富集分析"><i class="fa fa-check"></i><b>5.5</b> GSVA富集分析</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="富集分析.html"><a href="富集分析.html#gsva富集示例"><i class="fa fa-check"></i><b>5.5.1</b> GSVA富集示例</a></li>
<li class="chapter" data-level="5.5.2" data-path="富集分析.html"><a href="富集分析.html#gsva结果可视化"><i class="fa fa-check"></i><b>5.5.2</b> GSVA结果可视化</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="富集分析.html"><a href="富集分析.html#sessioninfo-3"><i class="fa fa-check"></i><b>5.6</b> sessionInfo</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="cellchat.html"><a href="cellchat.html"><i class="fa fa-check"></i><b>6</b> CellChat</a>
<ul>
<li class="chapter" data-level="6.1" data-path="cellchat.html"><a href="cellchat.html#简介"><i class="fa fa-check"></i><b>6.1</b> 简介</a></li>
<li class="chapter" data-level="6.2" data-path="cellchat.html"><a href="cellchat.html#数据加载与预处理"><i class="fa fa-check"></i><b>6.2</b> 数据加载与预处理</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="cellchat.html"><a href="cellchat.html#加载分析包数据"><i class="fa fa-check"></i><b>6.2.1</b> 加载分析包,数据</a></li>
<li class="chapter" data-level="6.2.2" data-path="cellchat.html"><a href="cellchat.html#构建cellchat对象"><i class="fa fa-check"></i><b>6.2.2</b> 构建CellChat对象</a></li>
<li class="chapter" data-level="6.2.3" data-path="cellchat.html"><a href="cellchat.html#设置受配体互作数据库"><i class="fa fa-check"></i><b>6.2.3</b> 设置受配体互作数据库</a></li>
<li class="chapter" data-level="6.2.4" data-path="cellchat.html"><a href="cellchat.html#预处理表达数据"><i class="fa fa-check"></i><b>6.2.4</b> 预处理表达数据</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="cellchat.html"><a href="cellchat.html#计算和推断细胞间通讯网络"><i class="fa fa-check"></i><b>6.3</b> 计算和推断细胞间通讯网络</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="cellchat.html"><a href="cellchat.html#细胞通讯互作可能性的计算"><i class="fa fa-check"></i><b>6.3.1</b> 细胞通讯互作可能性的计算</a></li>
<li class="chapter" data-level="6.3.2" data-path="cellchat.html"><a href="cellchat.html#推断通路水平的互作网络"><i class="fa fa-check"></i><b>6.3.2</b> 推断通路水平的互作网络</a></li>
<li class="chapter" data-level="6.3.3" data-path="cellchat.html"><a href="cellchat.html#汇总及展示细胞通讯网络整体结果"><i class="fa fa-check"></i><b>6.3.3</b> 汇总及展示细胞通讯网络整体结果</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="cellchat.html"><a href="cellchat.html#多种形式的互作通讯网络可视化"><i class="fa fa-check"></i><b>6.4</b> 多种形式的互作通讯网络可视化</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="cellchat.html"><a href="cellchat.html#信号通路可视化"><i class="fa fa-check"></i><b>6.4.1</b> 信号通路可视化</a></li>
<li class="chapter" data-level="6.4.2" data-path="cellchat.html"><a href="cellchat.html#信号通路水平展示"><i class="fa fa-check"></i><b>6.4.2</b> 信号通路水平展示</a></li>
<li class="chapter" data-level="6.4.3" data-path="cellchat.html"><a href="cellchat.html#快速探索推演的网络"><i class="fa fa-check"></i><b>6.4.3</b> 快速探索推演的网络</a></li>
<li class="chapter" data-level="6.4.4" data-path="cellchat.html"><a href="cellchat.html#通讯互作配受体对水平展示"><i class="fa fa-check"></i><b>6.4.4</b> 通讯互作配受体对水平展示</a></li>
<li class="chapter" data-level="6.4.5" data-path="cellchat.html"><a href="cellchat.html#多个信号通路或互作对的可视化"><i class="fa fa-check"></i><b>6.4.5</b> 多个信号通路或互作对的可视化</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="cellchat.html"><a href="cellchat.html#系统性分析细胞通讯网络"><i class="fa fa-check"></i><b>6.5</b> 系统性分析细胞通讯网络</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="cellchat.html"><a href="cellchat.html#识别信号角色和主要贡献信号"><i class="fa fa-check"></i><b>6.5.1</b> 识别信号角色和主要贡献信号</a></li>
<li class="chapter" data-level="6.5.2" data-path="cellchat.html"><a href="cellchat.html#通路信号角色识别及可视化"><i class="fa fa-check"></i><b>6.5.2</b> 通路信号角色识别及可视化</a></li>
<li class="chapter" data-level="6.5.3" data-path="cellchat.html"><a href="cellchat.html#特定细胞类型中输入输出信号贡献程度"><i class="fa fa-check"></i><b>6.5.3</b> 特定细胞类型中输入输出信号贡献程度</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="cellchat.html"><a href="cellchat.html#全局信号模式识别"><i class="fa fa-check"></i><b>6.6</b> 全局信号模式识别</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="cellchat.html"><a href="cellchat.html#确定信号模式数量"><i class="fa fa-check"></i><b>6.6.1</b> 确定信号模式数量</a></li>
<li class="chapter" data-level="6.6.2" data-path="cellchat.html"><a href="cellchat.html#识别信号模式"><i class="fa fa-check"></i><b>6.6.2</b> 识别信号模式</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="cellchat.html"><a href="cellchat.html#manifold-and-classification-learning-analysis-of-signaling-networks"><i class="fa fa-check"></i><b>6.7</b> Manifold and classification learning analysis of signaling networks</a></li>
<li class="chapter" data-level="6.8" data-path="cellchat.html"><a href="cellchat.html#sessioninfo-4"><i class="fa fa-check"></i><b>6.8</b> sessionInfo</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="ref-httpsgithub-comsqjincellchat.html"><a href="ref-httpsgithub-comsqjincellchat.html"><i class="fa fa-check"></i><b>7</b> ref: <span>https://github.com/sqjin/CellChat</span></a></li>
<li class="chapter" data-level="8" data-path="single-cell-transcriptomic-landscape-of-human-blood-cells.html"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html"><i class="fa fa-check"></i><b>8</b> Single-cell transcriptomic landscape of human blood cells</a>
<ul>
<li class="chapter" data-level="8.1" data-path="single-cell-transcriptomic-landscape-of-human-blood-cells.html"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#read-scrna-data"><i class="fa fa-check"></i><b>8.1</b> Read scRNA data</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="single-cell-transcriptomic-landscape-of-human-blood-cells.html"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#read-hspc-data"><i class="fa fa-check"></i><b>8.1.1</b> read HSPC data</a></li>
<li class="chapter" data-level="8.1.2" data-path="single-cell-transcriptomic-landscape-of-human-blood-cells.html"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#read-blood-cells-data"><i class="fa fa-check"></i><b>8.1.2</b> read Blood Cells data</a></li>
<li class="chapter" data-level="8.1.3" data-path="single-cell-transcriptomic-landscape-of-human-blood-cells.html"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#merge-two-mats-to-createseuratobject"><i class="fa fa-check"></i><b>8.1.3</b> Merge two mats to CreateSeuratObject</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="single-cell-transcriptomic-landscape-of-human-blood-cells.html"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#figure-1b"><i class="fa fa-check"></i><b>8.2</b> Figure 1B</a></li>
<li class="chapter" data-level="8.3" data-path="single-cell-transcriptomic-landscape-of-human-blood-cells.html"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#figure-1c"><i class="fa fa-check"></i><b>8.3</b> Figure 1C</a></li>
<li class="chapter" data-level="8.4" data-path="single-cell-transcriptomic-landscape-of-human-blood-cells.html"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#figure-1d"><i class="fa fa-check"></i><b>8.4</b> Figure 1D</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="single-cell-transcriptomic-landscape-of-human-blood-cells.html"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#lognormalize"><i class="fa fa-check"></i><b>8.4.1</b> LogNormalize</a></li>
<li class="chapter" data-level="8.4.2" data-path="single-cell-transcriptomic-landscape-of-human-blood-cells.html"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#筛选高变基因-1"><i class="fa fa-check"></i><b>8.4.2</b> 筛选高变基因</a></li>
<li class="chapter" data-level="8.4.3" data-path="single-cell-transcriptomic-landscape-of-human-blood-cells.html"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#筛选高变基因-and-pca降维"><i class="fa fa-check"></i><b>8.4.3</b> 筛选高变基因 and PCA降维</a></li>
<li class="chapter" data-level="8.4.4" data-path="single-cell-transcriptomic-landscape-of-human-blood-cells.html"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#聚类分簇-1"><i class="fa fa-check"></i><b>8.4.4</b> 聚类分簇</a></li>
<li class="chapter" data-level="8.4.5" data-path="single-cell-transcriptomic-landscape-of-human-blood-cells.html"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#单细胞聚类可视化-1"><i class="fa fa-check"></i><b>8.4.5</b> 单细胞聚类可视化</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="single-cell-transcriptomic-landscape-of-human-blood-cells.html"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#figure-1e"><i class="fa fa-check"></i><b>8.5</b> Figure 1E</a>
<ul>
<li class="chapter" data-level="8.5.1" data-path="single-cell-transcriptomic-landscape-of-human-blood-cells.html"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#细胞簇差异基因分析-1"><i class="fa fa-check"></i><b>8.5.1</b> 细胞簇差异基因分析</a></li>
<li class="chapter" data-level="8.5.2" data-path="single-cell-transcriptomic-landscape-of-human-blood-cells.html"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#marker基因展示-1"><i class="fa fa-check"></i><b>8.5.2</b> marker基因展示</a></li>
</ul></li>
<li class="chapter" data-level="8.6" data-path="single-cell-transcriptomic-landscape-of-human-blood-cells.html"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#sessioninfo-5"><i class="fa fa-check"></i><b>8.6</b> sessionInfo</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>9</b> References</a>
<ul>
<li class="chapter" data-level="9.1" data-path="references.html"><a href="references.html#python"><i class="fa fa-check"></i><b>9.1</b> Python</a></li>
<li class="chapter" data-level="9.2" data-path="references.html"><a href="references.html#analysis-of-single-cell-rna-seq-data-分析流程"><i class="fa fa-check"></i><b>9.2</b> Analysis of single cell RNA-seq data 分析流程</a></li>
<li class="chapter" data-level="9.3" data-path="references.html"><a href="references.html#assigning-cell-types-with-singler-细胞类型注释"><i class="fa fa-check"></i><b>9.3</b> Assigning cell types with SingleR # 细胞类型注释</a></li>
<li class="chapter" data-level="9.4" data-path="references.html"><a href="references.html#museum-of-spatial-transcriptomics-空转简史"><i class="fa fa-check"></i><b>9.4</b> Museum of Spatial Transcriptomics # 空转简史</a></li>
<li class="chapter" data-level="9.5" data-path="references.html"><a href="references.html#atac-分析-archr"><i class="fa fa-check"></i><b>9.5</b> ATAC 分析 Archr</a></li>
<li class="chapter" data-level="9.6" data-path="references.html"><a href="references.html#monocl-and-velocyto"><i class="fa fa-check"></i><b>9.6</b> monocl and velocyto</a></li>
<li class="chapter" data-level="9.7" data-path="references.html"><a href="references.html#integraton"><i class="fa fa-check"></i><b>9.7</b> integraton</a></li>
<li class="chapter" data-level="9.8" data-path="references.html"><a href="references.html#gene-set-enrichment"><i class="fa fa-check"></i><b>9.8</b> Gene set enrichment</a></li>
<li class="chapter" data-level="9.9" data-path="references.html"><a href="references.html#转录因子"><i class="fa fa-check"></i><b>9.9</b> 转录因子</a></li>
<li class="chapter" data-level="9.10" data-path="references.html"><a href="references.html#reproducible-research"><i class="fa fa-check"></i><b>9.10</b> Reproducible research</a></li>
<li class="chapter" data-level="9.11" data-path="references.html"><a href="references.html#基因调控"><i class="fa fa-check"></i><b>9.11</b> 基因调控</a></li>
<li class="chapter" data-level="9.12" data-path="references.html"><a href="references.html#cell-annotation"><i class="fa fa-check"></i><b>9.12</b> cell annotation</a></li>
<li class="chapter" data-level="9.13" data-path="references.html"><a href="references.html#空间分析"><i class="fa fa-check"></i><b>9.13</b> 空间分析</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Single Cell Sequencing Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="single-cell-transcriptomic-landscape-of-human-blood-cells" class="section level1" number="8">
<h1><span class="header-section-number">Chapter 8</span> Single-cell transcriptomic landscape of human blood cells</h1>
<p>Reproducible Research of <a href="https://academic.oup.com/nsr/article/8/3/nwaa180/5896476">Single-cell transcriptomic landscape of human blood cells</a></p>
<p>Download data from GEO</p>
<ul>
<li><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149938">Single-cell Transcriptomic Landscape of Human Blood Cells</a>
<ul>
<li>GSM4201332 BM1</li>
<li>GSM4201333 BM2</li>
<li>GSM4201334 BM3</li>
</ul></li>
<li><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=HSPC_GSE137864">Single-cell transcriptome landscape of adult HSPC</a>
<ul>
<li>GSM4793029 B cell population</li>
<li>GSM4793030 NK cell population</li>
<li>GSM4793031 T cell population</li>
<li>GSM4793032 Monocyte population</li>
<li>GSM4793033 Neutrophil population</li>
<li>GSM4793034 Erythrocyte population</li>
</ul></li>
</ul>
<div id="read-scrna-data" class="section level2" number="8.1">
<h2><span class="header-section-number">8.1</span> Read scRNA data</h2>
<p>Data are located in <code>./ABC/data/GSE137864_HSPC/</code>.</p>
<div id="read-hspc-data" class="section level3" number="8.1.1">
<h3><span class="header-section-number">8.1.1</span> read HSPC data</h3>
<div class="sourceCode" id="cb379"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb379-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb379-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb379-2"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb379-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb379-3"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb379-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Matrix)</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;Matrix&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:S4Vectors&#39;:
## 
##     expand</code></pre>
<pre><code>## The following objects are masked from &#39;package:tidyr&#39;:
## 
##     expand, pack, unpack</code></pre>
<div class="sourceCode" id="cb383"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb383-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb383-1" aria-hidden="true" tabindex="-1"></a><span class="do">## HSPC</span></span>
<span id="cb383-2"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb383-2" aria-hidden="true" tabindex="-1"></a>dir_10x <span class="ot">&lt;-</span> <span class="st">&quot;./ABC/data/GSE137864_HSPC/&quot;</span></span>
<span id="cb383-3"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb383-3" aria-hidden="true" tabindex="-1"></a><span class="co"># change names as Read10X needed.</span></span>
<span id="cb383-4"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb383-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dir</span>(dir_10x)</span></code></pre></div>
<pre><code>## [1] &quot;barcodes.tsv.gz&quot;                &quot;features.tsv.gz&quot;               
## [3] &quot;GPL20301.soft&quot;                  &quot;GSE137864_series_matrix.txt.gz&quot;
## [5] &quot;matrix.mtx.gz&quot;</code></pre>
<div class="sourceCode" id="cb385"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb385-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb385-1" aria-hidden="true" tabindex="-1"></a>HSPC_raw <span class="ot">&lt;-</span> <span class="fu">Read10X</span>(dir_10x)</span>
<span id="cb385-2"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb385-2" aria-hidden="true" tabindex="-1"></a>HSPC_meta <span class="ot">&lt;-</span> GEOquery<span class="sc">::</span><span class="fu">getGEO</span>(<span class="st">&quot;GSE137864&quot;</span>, <span class="at">destdir=</span><span class="st">&quot;./ABC/data/GSE137864_HSPC/&quot;</span>)</span></code></pre></div>
<pre><code>## Setting options(&#39;download.file.method.GEOquery&#39;=&#39;auto&#39;)</code></pre>
<pre><code>## Setting options(&#39;GEOquery.inmemory.gpl&#39;=FALSE)</code></pre>
<pre><code>## Found 1 file(s)</code></pre>
<pre><code>## GSE137864_series_matrix.txt.gz</code></pre>
<pre><code>## Using locally cached version: ./ABC/data/GSE137864_HSPC//GSE137864_series_matrix.txt.gz</code></pre>
<pre><code>## 
## ── Column specification ────────────────────────────────────────────────────────
## cols(
##   ID_REF = col_character(),
##   GSM4201332 = col_character(),
##   GSM4201333 = col_character(),
##   GSM4201334 = col_character()
## )</code></pre>
<pre><code>## Using locally cached version of GPL20301 found here:
## ./ABC/data/GSE137864_HSPC//GPL20301.soft</code></pre>
<div class="sourceCode" id="cb393"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb393-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb393-1" aria-hidden="true" tabindex="-1"></a>HSPC_meta <span class="ot">&lt;-</span> Biobase<span class="sc">::</span><span class="fu">pData</span>(HSPC_meta<span class="sc">$</span>GSE137864_series_matrix.txt.gz)</span>
<span id="cb393-2"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb393-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb393-3"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb393-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb393-4"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb393-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(HSPC_raw) <span class="sc">%&gt;%</span>  head</span></code></pre></div>
<pre><code>## [1] &quot;HSC_BM1_L1_bar1&quot; &quot;HSC_BM1_L1_bar2&quot; &quot;HSC_BM1_L1_bar3&quot; &quot;HSC_BM1_L1_bar5&quot;
## [5] &quot;HSC_BM1_L1_bar6&quot; &quot;HSC_BM1_L1_bar7&quot;</code></pre>
</div>
<div id="read-blood-cells-data" class="section level3" number="8.1.2">
<h3><span class="header-section-number">8.1.2</span> read Blood Cells data</h3>
<div class="sourceCode" id="cb395"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb395-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb395-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Blood Cells</span></span>
<span id="cb395-2"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb395-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb395-3"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb395-3" aria-hidden="true" tabindex="-1"></a>BC_raw <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;./ABC/data/HumanBloodCells/GSE149938_umi_matrix.csv&quot;</span>, <span class="at">check.names=</span>F)</span>
<span id="cb395-4"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb395-4" aria-hidden="true" tabindex="-1"></a>BC_mat <span class="ot">&lt;-</span> <span class="fu">Matrix</span>( <span class="fu">t</span>(<span class="fu">as.matrix</span>(BC_raw)) )</span>
<span id="cb395-5"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb395-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb395-6"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb395-6" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(BC_raw)</span></code></pre></div>
<pre><code>## [1]  7643 19813</code></pre>
<div class="sourceCode" id="cb397"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb397-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb397-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(BC_mat) <span class="sc">%&gt;%</span>  head</span></code></pre></div>
<pre><code>## [1] &quot;BNK_spBM1_L1_bar25&quot; &quot;BNK_spBM1_L1_bar26&quot; &quot;BNK_spBM1_L1_bar27&quot;
## [4] &quot;BNK_spBM1_L1_bar28&quot; &quot;BNK_spBM1_L1_bar29&quot; &quot;BNK_spBM1_L1_bar30&quot;</code></pre>
<div class="sourceCode" id="cb399"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb399-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb399-1" aria-hidden="true" tabindex="-1"></a>BC_meta <span class="ot">&lt;-</span> GEOquery<span class="sc">::</span><span class="fu">getGEO</span>(<span class="st">&quot;GSE149938&quot;</span>, <span class="at">destdir=</span><span class="st">&quot;./ABC/data/HumanBloodCells/&quot;</span>)</span></code></pre></div>
<pre><code>## Found 1 file(s)</code></pre>
<pre><code>## GSE149938_series_matrix.txt.gz</code></pre>
<pre><code>## Using locally cached version: ./ABC/data/HumanBloodCells//GSE149938_series_matrix.txt.gz</code></pre>
<pre><code>## 
## ── Column specification ────────────────────────────────────────────────────────
## cols(
##   ID_REF = col_character(),
##   GSM4793029 = col_character(),
##   GSM4793030 = col_character(),
##   GSM4793031 = col_character(),
##   GSM4793032 = col_character(),
##   GSM4793033 = col_character(),
##   GSM4793034 = col_character()
## )</code></pre>
<pre><code>## Using locally cached version of GPL20301 found here:
## ./ABC/data/HumanBloodCells//GPL20301.soft</code></pre>
<div class="sourceCode" id="cb405"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb405-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb405-1" aria-hidden="true" tabindex="-1"></a>BC_meta <span class="ot">&lt;-</span> Biobase<span class="sc">::</span><span class="fu">pData</span>(BC_meta<span class="sc">$</span>GSE149938_series_matrix.txt.gz)</span></code></pre></div>
</div>
<div id="merge-two-mats-to-createseuratobject" class="section level3" number="8.1.3">
<h3><span class="header-section-number">8.1.3</span> Merge two mats to CreateSeuratObject</h3>
<div class="sourceCode" id="cb406"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb406-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb406-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge two mats</span></span>
<span id="cb406-2"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb406-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(HSPC_raw)</span></code></pre></div>
<pre><code>## [1] 19813  1813</code></pre>
<div class="sourceCode" id="cb408"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb408-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb408-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(BC_mat)</span></code></pre></div>
<pre><code>## [1] 19813  7643</code></pre>
<div class="sourceCode" id="cb410"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb410-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb410-1" aria-hidden="true" tabindex="-1"></a>ABC_HSPC_mat <span class="ot">&lt;-</span> <span class="fu">cbind</span>(HSPC_raw, BC_mat)</span>
<span id="cb410-2"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb410-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(ABC_HSPC_mat)</span></code></pre></div>
<pre><code>## [1] 19813  9456</code></pre>
<div class="sourceCode" id="cb412"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb412-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb412-1" aria-hidden="true" tabindex="-1"></a><span class="co"># distinguish protein-coding genes and lncRNAs</span></span>
<span id="cb412-2"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb412-2" aria-hidden="true" tabindex="-1"></a><span class="co"># gtf: ftp://ftp.ensembl.org/pub/release-84/gtf/homo_sapiens/</span></span>
<span id="cb412-3"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb412-3" aria-hidden="true" tabindex="-1"></a><span class="co"># add genesymols annotation</span></span>
<span id="cb412-4"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb412-4" aria-hidden="true" tabindex="-1"></a>gtf_fn <span class="ot">&lt;-</span> <span class="st">&quot;~/data/drug_seq/rawdata/genome/Homo_sapiens.GRCh38.102.gtf&quot;</span></span>
<span id="cb412-5"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb412-5" aria-hidden="true" tabindex="-1"></a>gtf <span class="ot">&lt;-</span> rtracklayer<span class="sc">::</span><span class="fu">import</span>(gtf_fn)</span>
<span id="cb412-6"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb412-6" aria-hidden="true" tabindex="-1"></a>gtf_df <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">as.data.frame</span>(gtf), type<span class="sc">==</span><span class="st">&quot;gene&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb412-7"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb412-7" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id, gene_name, gene_biotype)</span>
<span id="cb412-8"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb412-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb412-9"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb412-9" aria-hidden="true" tabindex="-1"></a>ABC_HSPC_gene <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">rownames</span>(ABC_HSPC_mat)) </span>
<span id="cb412-10"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb412-10" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(ABC_HSPC_gene) <span class="ot">&lt;-</span> <span class="st">&quot;gene_name&quot;</span></span>
<span id="cb412-11"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb412-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb412-12"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb412-12" aria-hidden="true" tabindex="-1"></a>gene_types <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(ABC_HSPC_gene, gtf_df)</span></code></pre></div>
<pre><code>## Joining, by = &quot;gene_name&quot;</code></pre>
<div class="sourceCode" id="cb414"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb414-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb414-1" aria-hidden="true" tabindex="-1"></a>protein_coding_genes <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(gene_types, gene_biotype<span class="sc">==</span><span class="st">&quot;protein_coding&quot;</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_name) <span class="sc">%&gt;%</span> <span class="fu">unlist</span>(<span class="at">use.names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb414-2"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb414-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb414-3"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb414-3" aria-hidden="true" tabindex="-1"></a><span class="co"># should use NNONCODE website (http://www.noncode.org/datadownload/NONCODEv6_human_hg38_lncRNA.gtf.gz)</span></span>
<span id="cb414-4"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb414-4" aria-hidden="true" tabindex="-1"></a>lncRNAs <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(gene_types, gene_biotype<span class="sc">==</span><span class="st">&quot;lncRNA&quot;</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_name) <span class="sc">%&gt;%</span> <span class="fu">unlist</span>(<span class="at">use.names =</span> <span class="cn">FALSE</span>) </span>
<span id="cb414-5"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb414-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb414-6"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb414-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 7551 single cells</span></span>
<span id="cb414-7"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb414-7" aria-hidden="true" tabindex="-1"></a>ABC <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">count =</span> ABC_HSPC_mat[protein_coding_genes,], <span class="at">project =</span> <span class="st">&#39;ABC&#39;</span>, <span class="at">min.cells =</span> <span class="dv">5</span>, <span class="at">min.features =</span> <span class="dv">1000</span> )</span>
<span id="cb414-8"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb414-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb414-9"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb414-9" aria-hidden="true" tabindex="-1"></a>ABC</span></code></pre></div>
<pre><code>## An object of class Seurat 
## 17057 features across 9364 samples within 1 assay 
## Active assay: RNA (17057 features, 0 variable features)</code></pre>
<div id="add-celltype-info." class="section level4" number="8.1.3.1">
<h4><span class="header-section-number">8.1.3.1</span> add celltype info.</h4>
<div class="figure">
<img src="figure/hematopoiesis.png" alt="" />
<p class="caption">hematopoiesis</p>
</div>
<div class="figure">
<img src="figure/celltype.png" alt="" />
<p class="caption">celltype</p>
</div>
<p><img src="figure/tcell.png" alt="tcell" />
<img src="figure/bcell.png" alt="bcell" /></p>
<p>Ref: <code>细胞类型鉴定及单细胞数据库-2021.04.24.pdf</code></p>
<div class="sourceCode" id="cb416"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb416-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb416-1" aria-hidden="true" tabindex="-1"></a><span class="co"># cell types</span></span>
<span id="cb416-2"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb416-2" aria-hidden="true" tabindex="-1"></a>cell_types <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">&quot;./ABC/data/blood_ceel_types.xlsx&quot;</span>, <span class="at">sheet=</span><span class="st">&quot;Sheet1&quot;</span>)</span>
<span id="cb416-3"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb416-3" aria-hidden="true" tabindex="-1"></a>ABC_meta <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(ABC<span class="sc">@</span>meta.data,  cell_types)</span></code></pre></div>
<pre><code>## Joining, by = &quot;orig.ident&quot;</code></pre>
<div class="sourceCode" id="cb418"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb418-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-1" aria-hidden="true" tabindex="-1"></a>ABC_meta<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(ABC_meta<span class="sc">$</span>group, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;HSPC&quot;</span>, <span class="st">&quot;Bcells&quot;</span>, <span class="st">&quot;NK&quot;</span>, </span>
<span id="cb418-2"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-2" aria-hidden="true" tabindex="-1"></a>                                                     <span class="st">&quot;Tcells&quot;</span>, <span class="st">&quot;Monocytes&quot;</span>, <span class="st">&quot;Neutrophils&quot;</span> , <span class="st">&quot;Erythrocytes&quot;</span> ) )</span>
<span id="cb418-3"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb418-4"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-4" aria-hidden="true" tabindex="-1"></a>ranked_celltypes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;HSC&quot;</span>, <span class="st">&quot;BNK&quot;</span>, <span class="st">&quot;LMPP&quot;</span>, <span class="st">&quot;MLP&quot;</span>, <span class="st">&quot;MPP&quot;</span>, <span class="st">&quot;CMP&quot;</span>, <span class="st">&quot;MEP&quot;</span>, <span class="st">&quot;GMP&quot;</span>, <span class="st">&quot;memB&quot;</span>, <span class="st">&quot;preB&quot;</span>, <span class="st">&quot;proB&quot;</span>, <span class="st">&quot;immB&quot;</span>, <span class="st">&quot;regB&quot;</span>, </span>
<span id="cb418-5"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-5" aria-hidden="true" tabindex="-1"></a>                      <span class="st">&quot;naiB&quot;</span>, <span class="st">&quot;plasma&quot;</span>, <span class="st">&quot;CLP&quot;</span>, <span class="st">&quot;NKP&quot;</span>, <span class="st">&quot;kineNK&quot;</span>, <span class="st">&quot;toxiNK&quot;</span>, <span class="st">&quot;CD4T&quot;</span>, <span class="st">&quot;CD8T&quot;</span>, <span class="st">&quot;claM&quot;</span>, <span class="st">&quot;cMOP&quot;</span>, <span class="st">&quot;hMDP&quot;</span>, </span>
<span id="cb418-6"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-6" aria-hidden="true" tabindex="-1"></a>                      <span class="st">&quot;interM&quot;</span>, <span class="st">&quot;nonM&quot;</span>, <span class="st">&quot;preM&quot;</span>, <span class="st">&quot;metaN&quot;</span>, <span class="st">&quot;matureN&quot;</span>, <span class="st">&quot;proN&quot;</span>, <span class="st">&quot;myeN&quot;</span>, <span class="st">&quot;ery&quot;</span>)</span>
<span id="cb418-7"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb418-8"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-8" aria-hidden="true" tabindex="-1"></a>ABC_meta<span class="sc">$</span>orig.ident <span class="ot">&lt;-</span> <span class="fu">factor</span>(ABC_meta<span class="sc">$</span>orig.ident, <span class="at">levels=</span>ranked_celltypes )</span>
<span id="cb418-9"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb418-10"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-10" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ABC_meta) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(ABC<span class="sc">@</span>meta.data)</span>
<span id="cb418-11"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb418-12"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-12" aria-hidden="true" tabindex="-1"></a>ABC<span class="sc">@</span>meta.data <span class="ot">&lt;-</span> ABC_meta</span>
<span id="cb418-13"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-13" aria-hidden="true" tabindex="-1"></a><span class="co"># HSPC_types &lt;- unique(gsub(&quot;_.*&quot;,&quot;&quot;,colnames(HSPC_raw)))</span></span>
<span id="cb418-14"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-14" aria-hidden="true" tabindex="-1"></a><span class="co"># HSPC_types &lt;- c(&quot;HSC&quot;, &quot;BNK&quot;,  &quot;LMPP&quot;, &quot;MLP&quot;,  &quot;MPP&quot;,  &quot;CMP&quot;,  &quot;MEP&quot;,  &quot;GMP&quot; )</span></span>
<span id="cb418-15"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-15" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb418-16"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-16" aria-hidden="true" tabindex="-1"></a><span class="co"># ABC_meta[which(ABC@meta.data$orig.ident %in% HSPC_types ), &quot;group&quot;] &lt;- &quot;HSPC&quot;</span></span>
<span id="cb418-17"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-17" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb418-18"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-18" aria-hidden="true" tabindex="-1"></a><span class="co"># PB_types &lt;- c(&quot;memB&quot;, &quot;regB&quot;)</span></span>
<span id="cb418-19"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-19" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb418-20"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-20" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb418-21"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Neutrophils_types &lt;- c(&quot;metaN&quot;, &quot;myeN&quot;, &quot;matureN&quot;, &quot;proN&quot;, &quot;myeN&quot; )</span></span>
<span id="cb418-22"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-22" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb418-23"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-23" aria-hidden="true" tabindex="-1"></a><span class="co"># NK_cells_types &lt;- c(&quot;CLP&quot;, &quot;NKP&quot;, &quot;kineNK&quot;, &quot;toxiNK&quot;)</span></span>
<span id="cb418-24"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-24" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb418-25"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Bcells_types &lt;- c(&quot;preB&quot;, &quot;proB&quot;, &quot;immB&quot;, &quot;regB&quot;, &quot;naiB&quot;, &quot;plasma&quot;)</span></span>
<span id="cb418-26"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Tcells_types &lt;- c(&quot;CD4T&quot;, &quot;CD8T&quot;)</span></span>
<span id="cb418-27"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-27" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb418-28"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Erythrocytes_types &lt;- c(&quot;ery&quot;)</span></span>
<span id="cb418-29"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb418-30"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb418-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Monocytes_types &lt;- c(&quot;claM&quot;, &quot;cMOP&quot;, &quot;hMDP&quot;, &quot;interM&quot;, &quot;nonM&quot;, &quot;preM&quot;)</span></span></code></pre></div>
</div>
<div id="线粒体" class="section level4" number="8.1.3.2">
<h4><span class="header-section-number">8.1.3.2</span> 线粒体</h4>
<div class="sourceCode" id="cb419"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb419-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb419-1" aria-hidden="true" tabindex="-1"></a>ABC <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(ABC, <span class="at">pattern =</span> <span class="st">&quot;^MT-&quot;</span>, <span class="at">col.name =</span> <span class="st">&quot;percent.mt&quot;</span>) </span>
<span id="cb419-2"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb419-2" aria-hidden="true" tabindex="-1"></a><span class="co"># ABC[[&quot;percent.mt&quot;]] &lt;- PercentageFeatureSet(ABC, pattern = &quot;^MT-&quot;)</span></span>
<span id="cb419-3"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb419-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ABC<span class="sc">@</span>meta.data)</span></code></pre></div>
<pre><code>##                 orig.ident nCount_RNA nFeature_RNA group cellname percent.mt
## HSC_BM1_L1_bar1        HSC     153232         4697  HSPC      HSC          0
## HSC_BM1_L1_bar2        HSC      61300         3807  HSPC      HSC          0
## HSC_BM1_L1_bar3        HSC      61831         4034  HSPC      HSC          0
## HSC_BM1_L1_bar5        HSC      30637         4156  HSPC      HSC          0
## HSC_BM1_L1_bar6        HSC      37540         4439  HSPC      HSC          0
## HSC_BM1_L1_bar7        HSC      30843         4367  HSPC      HSC          0</code></pre>
</div>
</div>
</div>
<div id="figure-1b" class="section level2" number="8.2">
<h2><span class="header-section-number">8.2</span> Figure 1B</h2>
<div class="sourceCode" id="cb421"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb421-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb421-1" aria-hidden="true" tabindex="-1"></a><span class="co"># VlnPlot(ABC, features=c(&#39;nFeature_RNA&#39;), group.by = &quot;group&quot;)</span></span>
<span id="cb421-2"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb421-2" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(ABC, <span class="at">features=</span><span class="fu">c</span>(<span class="st">&#39;nFeature_RNA&#39;</span>), <span class="at">group.by =</span> <span class="st">&quot;orig.ident&quot;</span>)</span></code></pre></div>
<p><img src="Single_cell_training_files/figure-html/VlnPlot_gene-1.png" width="1440" /></p>
</div>
<div id="figure-1c" class="section level2" number="8.3">
<h2><span class="header-section-number">8.3</span> Figure 1C</h2>
<div class="sourceCode" id="cb422"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb422-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb422-1" aria-hidden="true" tabindex="-1"></a><span class="co">#boxplot all genes</span></span>
<span id="cb422-2"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb422-2" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(ABC<span class="sc">@</span>meta.data<span class="sc">$</span>nFeature_RNA)</span></code></pre></div>
<p><img src="Single_cell_training_files/figure-html/boxplot_gene-1.png" width="672" /></p>
</div>
<div id="figure-1d" class="section level2" number="8.4">
<h2><span class="header-section-number">8.4</span> Figure 1D</h2>
<div id="lognormalize" class="section level3" number="8.4.1">
<h3><span class="header-section-number">8.4.1</span> LogNormalize</h3>
<div class="sourceCode" id="cb423"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb423-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb423-1" aria-hidden="true" tabindex="-1"></a><span class="do">## ref: https://satijalab.org/seurat/articles/sctransform_vignette.html</span></span>
<span id="cb423-2"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb423-2" aria-hidden="true" tabindex="-1"></a><span class="do">## SCTransform replaces NormalizeData, ScaleData, and FindVariableFeatures.</span></span>
<span id="cb423-3"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb423-3" aria-hidden="true" tabindex="-1"></a><span class="do">## During normalization, we can also remove confounding sources of variation, </span></span>
<span id="cb423-4"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb423-4" aria-hidden="true" tabindex="-1"></a><span class="do">## for example, mitochondrial mapping percentage</span></span>
<span id="cb423-5"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb423-5" aria-hidden="true" tabindex="-1"></a>ABC <span class="ot">&lt;-</span> <span class="fu">SCTransform</span>(ABC, <span class="at">vars.to.regress =</span> <span class="st">&quot;percent.mt&quot;</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb423-6"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb423-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb423-7"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb423-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb423-8"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb423-8" aria-hidden="true" tabindex="-1"></a><span class="co"># ABC &lt;- NormalizeData(object=ABC, </span></span>
<span id="cb423-9"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb423-9" aria-hidden="true" tabindex="-1"></a><span class="co">#                      normalization.method= &#39;LogNormalize&#39;, </span></span>
<span id="cb423-10"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb423-10" aria-hidden="true" tabindex="-1"></a><span class="co">#                      scale.factor=10000 )</span></span></code></pre></div>
</div>
<div id="筛选高变基因-1" class="section level3" number="8.4.2">
<h3><span class="header-section-number">8.4.2</span> 筛选高变基因</h3>
<div class="sourceCode" id="cb424"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb424-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb424-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ABC &lt;- FindVariableFeatures(ABC,</span></span>
<span id="cb424-2"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb424-2" aria-hidden="true" tabindex="-1"></a><span class="co">#                             selection.method = &quot;vst&quot;, </span></span>
<span id="cb424-3"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb424-3" aria-hidden="true" tabindex="-1"></a><span class="co">#                             nfeatures = 2000)</span></span>
<span id="cb424-4"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb424-4" aria-hidden="true" tabindex="-1"></a><span class="co"># head(VariableFeatures(ABC))</span></span>
<span id="cb424-5"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb424-5" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb424-6"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb424-6" aria-hidden="true" tabindex="-1"></a><span class="co"># ABC &lt;- ScaleData(ABC)</span></span></code></pre></div>
</div>
<div id="筛选高变基因-and-pca降维" class="section level3" number="8.4.3">
<h3><span class="header-section-number">8.4.3</span> 筛选高变基因 and PCA降维</h3>
<div class="sourceCode" id="cb425"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb425-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb425-1" aria-hidden="true" tabindex="-1"></a><span class="co"># hvg &lt;- VariableFeatures(object = ABC)</span></span>
<span id="cb425-2"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb425-2" aria-hidden="true" tabindex="-1"></a><span class="co"># ABC &lt;- RunPCA(ABC, features = hvg )</span></span>
<span id="cb425-3"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb425-3" aria-hidden="true" tabindex="-1"></a>ABC <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(ABC )</span></code></pre></div>
<pre><code>## PC_ 1 
## Positive:  S100A9, S100A8, LYZ, MNDA, FCN1, S100A12, SERPINA1, TYROBP, LTF, S100A11 
##     CTSS, COTL1, FGR, FCER1G, IFITM2, C5AR1, CSF3R, CYBB, CYP4F3, G0S2 
##     NAMPT, SRGN, FPR1, MMP9, LCN2, FCGR3B, MYO1F, PLBD1, NCF2, CAMP 
## Negative:  HBB, HBA2, HBA1, AHSP, CA1, HBD, HBM, BLVRB, GYPA, ALAS2 
##     PRDX2, GYPB, SLC4A1, CA2, HEMGN, HMBS, RHAG, SELENBP1, RPS27, MGST3 
##     DNTT, IGLL1, VPREB3, MYL4, SNCA, UROD, SLC25A39, EIF1AY, VPREB1, TFRC 
## PC_ 2 
## Positive:  HBA1, HBB, HBA2, AHSP, CA1, HBD, HBM, GYPA, BLVRB, ALAS2 
##     SLC4A1, GYPB, CA2, SLC25A37, PRDX2, HMBS, HEMGN, S100A9, S100A8, RHAG 
##     CD36, SELENBP1, MGST3, DEFA3, LYZ, LGALS3, S100A6, MYL4, MNDA, SNCA 
## Negative:  CD79B, CD79A, VPREB3, CD52, MS4A1, DNTT, TCL1A, CD74, CD69, HLA-DRA 
##     IGLL1, VPREB1, RPS27, CXCR4, MZB1, PAX5, SOX4, TMSB4X, B2M, BIRC3 
##     TMSB10, BANK1, CD9, SPIB, NEIL1, HLA-DRB1, ARPP21, PCDH9, HLA-DRB5, JUN 
## PC_ 3 
## Positive:  TCL1A, MS4A1, CD79B, CD79A, VPREB3, PAX5, CD69, HBA2, SPIB, HBB 
##     HBA1, BANK1, CD52, CA1, AHSP, FCRL1, RALGPS2, PCDH9, CXCR4, NEIL1 
##     HLA-DRA, BIRC3, CD74, HBD, IKZF3, HBM, CD24, CD83, AFF3, CD22 
## Negative:  MPO, ELANE, SPINK2, PRTN3, AZU1, SRGN, RNASE2, NKG7, IL32, AIF1 
##     AVP, CST7, HOPX, LYZ, PRSS57, ANXA1, MT2A, CTSW, CRHBP, CLEC11A 
##     LGALS1, CALR, SMIM24, CCL5, CD3D, HCST, CTSG, CYTL1, EREG, TSC22D1 
## PC_ 4 
## Positive:  LTF, LCN2, PGLYRP1, CYP4F3, CAMP, S100P, CD24, MMP9, CRISP3, DEFA3 
##     S100A12, DNTT, MMP8, ANXA3, BPI, ALOX5AP, PADI4, IGLL1, CEACAM8, VPREB1 
##     S100A8, CD177, FCGR3B, ITGAM, RETN, GCA, ARG1, CHI3L1, VPREB3, MMP25 
## Negative:  PSAP, CTSS, SERPINA1, FGL2, CST3, FCGR3A, MS4A7, COTL1, MPEG1, VCAN 
##     TYROBP, MAFB, FCER1G, SAMHD1, SAT1, KLF4, FOS, FOSB, TMEM176B, S100A10 
##     TNFRSF1B, POU2F2, TGFBI, NFKBIZ, CD52, ZFP36, CD300E, LST1, AHNAK, CD14 
## PC_ 5 
## Positive:  IL32, CCL5, NKG7, CD3D, GZMA, PRF1, GZMH, GNLY, CD52, GZMB 
##     CD3G, FGFBP2, CST7, KLRD1, CD2, CD3E, CTSW, LCK, CD8B, B2M 
##     CD8A, SYNE2, CD247, MS4A1, HCST, IFITM1, DUSP2, CCL4, PYHIN1, IFITM2 
## Negative:  DNTT, IGLL1, VPREB1, VPREB3, CD79B, MPO, ARPP21, CD9, LYZ, SOX4 
##     AKAP12, VCAN, MZB1, STMN1, RNASE2, PRTN3, PTPRE, LCN6, ELANE, MME 
##     CST3, AZU1, FCN1, PSAP, NEIL1, SERPINA1, FOS, FOSB, KLF4, CMTM8</code></pre>
<div class="sourceCode" id="cb427"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb427-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb427-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ElbowPlot</span>(ABC)</span></code></pre></div>
<p><img src="Single_cell_training_files/figure-html/unnamed-chunk-26-1.png" width="672" /></p>
</div>
<div id="聚类分簇-1" class="section level3" number="8.4.4">
<h3><span class="header-section-number">8.4.4</span> 聚类分簇</h3>
<div class="sourceCode" id="cb428"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb428-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb428-1" aria-hidden="true" tabindex="-1"></a>ABC <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(ABC, <span class="at">reduction =</span> <span class="st">&quot;pca&quot;</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>)</span></code></pre></div>
<pre><code>## Computing nearest neighbor graph</code></pre>
<pre><code>## Computing SNN</code></pre>
<div class="sourceCode" id="cb431"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb431-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb431-1" aria-hidden="true" tabindex="-1"></a>ABC <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(ABC, <span class="at">resolution =</span> <span class="fl">0.5</span>)</span></code></pre></div>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 9364
## Number of edges: 287258
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.9396
## Number of communities: 22
## Elapsed time: 0 seconds</code></pre>
<div class="sourceCode" id="cb433"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb433-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb433-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ABC<span class="sc">@</span>meta.data)</span></code></pre></div>
<pre><code>##                 orig.ident nCount_RNA nFeature_RNA group cellname percent.mt
## HSC_BM1_L1_bar1        HSC     153232         4697  HSPC      HSC          0
## HSC_BM1_L1_bar2        HSC      61300         3807  HSPC      HSC          0
## HSC_BM1_L1_bar3        HSC      61831         4034  HSPC      HSC          0
## HSC_BM1_L1_bar5        HSC      30637         4156  HSPC      HSC          0
## HSC_BM1_L1_bar6        HSC      37540         4439  HSPC      HSC          0
## HSC_BM1_L1_bar7        HSC      30843         4367  HSPC      HSC          0
##                 nCount_SCT nFeature_SCT SCT_snn_res.0.5 seurat_clusters
## HSC_BM1_L1_bar1      37956         3769               6               6
## HSC_BM1_L1_bar2      38263         3790               6               6
## HSC_BM1_L1_bar3      38005         4027               5               5
## HSC_BM1_L1_bar5      37563         4156               2               2
## HSC_BM1_L1_bar6      37705         4439               2               2
## HSC_BM1_L1_bar7      37553         4367               6               6</code></pre>
<div class="sourceCode" id="cb435"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb435-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb435-1" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(ABC)</span></code></pre></div>
<pre><code>##  [1] &quot;0&quot;  &quot;1&quot;  &quot;2&quot;  &quot;3&quot;  &quot;4&quot;  &quot;5&quot;  &quot;6&quot;  &quot;7&quot;  &quot;8&quot;  &quot;9&quot;  &quot;10&quot; &quot;11&quot; &quot;12&quot; &quot;13&quot; &quot;14&quot;
## [16] &quot;15&quot; &quot;16&quot; &quot;17&quot; &quot;18&quot; &quot;19&quot; &quot;20&quot; &quot;21&quot;</code></pre>
</div>
<div id="单细胞聚类可视化-1" class="section level3" number="8.4.5">
<h3><span class="header-section-number">8.4.5</span> 单细胞聚类可视化</h3>
<div class="sourceCode" id="cb437"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb437-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb437-1" aria-hidden="true" tabindex="-1"></a>ABC <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(ABC, </span>
<span id="cb437-2"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb437-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">reduction =</span> <span class="st">&quot;pca&quot;</span>, </span>
<span id="cb437-3"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb437-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>,</span>
<span id="cb437-4"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb437-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">verbose=</span><span class="cn">TRUE</span> )</span></code></pre></div>
<pre><code>## 23:44:22 UMAP embedding parameters a = 0.9922 b = 1.112</code></pre>
<pre><code>## 23:44:22 Read 9364 rows and found 30 numeric columns</code></pre>
<pre><code>## 23:44:22 Using Annoy for neighbor search, n_neighbors = 30</code></pre>
<pre><code>## 23:44:22 Building Annoy index with metric = cosine, n_trees = 50</code></pre>
<pre><code>## 0%   10   20   30   40   50   60   70   80   90   100%</code></pre>
<pre><code>## [----|----|----|----|----|----|----|----|----|----|</code></pre>
<pre><code>## **************************************************|
## 23:44:24 Writing NN index file to temp file /tmp/RtmpxBTcqh/file7e9e29fc3d1c
## 23:44:24 Searching Annoy index using 4 threads, search_k = 3000
## 23:44:24 Annoy recall = 100%
## 23:44:26 Commencing smooth kNN distance calibration using 4 threads
## 23:44:29 Initializing from normalized Laplacian + noise
## 23:44:30 Commencing optimization for 500 epochs, with 360614 positive edges
## 23:44:40 Optimization finished</code></pre>
<div class="sourceCode" id="cb445"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb445-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb445-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(ABC, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>, <span class="at">label=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Single_cell_training_files/figure-html/unnamed-chunk-28-1.png" width="672" /></p>
</div>
</div>
<div id="figure-1e" class="section level2" number="8.5">
<h2><span class="header-section-number">8.5</span> Figure 1E</h2>
<div id="细胞簇差异基因分析-1" class="section level3" number="8.5.1">
<h3><span class="header-section-number">8.5.1</span> 细胞簇差异基因分析</h3>
<div class="sourceCode" id="cb446"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb446-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb446-1" aria-hidden="true" tabindex="-1"></a>all_markers <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(ABC, </span>
<span id="cb446-2"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb446-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">only.pos =</span> <span class="cn">TRUE</span>, </span>
<span id="cb446-3"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb446-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">min.pct=</span> <span class="fl">0.1</span>, </span>
<span id="cb446-4"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb446-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">logfc.threshold=</span> <span class="fl">0.25</span>)</span>
<span id="cb446-5"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb446-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(all_markers)</span></code></pre></div>
<pre><code>##               p_val avg_log2FC pct.1 pct.2     p_val_adj cluster  gene
## AVP    0.000000e+00  1.2878495 0.809 0.196  0.000000e+00       0   AVP
## ABCB1 5.930955e-279  1.4981573 0.631 0.136 1.011643e-274       0 ABCB1
## ROS1  6.380195e-273  0.6759969 0.389 0.046 1.088270e-268       0  ROS1
## RPL31 1.800806e-266  0.9301858 1.000 0.999 3.071634e-262       0 RPL31
## CRHBP 7.394411e-266  1.7369954 0.704 0.177 1.261265e-261       0 CRHBP
## RPS24 4.585922e-262  1.0209993 1.000 0.996 7.822207e-258       0 RPS24</code></pre>
<div class="sourceCode" id="cb448"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb448-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb448-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb448-2"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb448-2" aria-hidden="true" tabindex="-1"></a>top_markers <span class="ot">&lt;-</span> all_markers <span class="sc">%&gt;%</span> <span class="fu">group_by</span>( cluster) <span class="sc">%&gt;%</span> <span class="fu">top_n</span>( <span class="at">n =</span> <span class="dv">5</span>, <span class="at">wt=</span> avg_log2FC )</span>
<span id="cb448-3"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb448-3" aria-hidden="true" tabindex="-1"></a>top_markers</span></code></pre></div>
<pre><code>## # A tibble: 110 x 7
## # Groups:   cluster [22]
##        p_val avg_log2FC pct.1 pct.2 p_val_adj cluster gene   
##        &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;fct&gt;   &lt;chr&gt;  
##  1 1.77e-237       2.24 0.741 0.247 3.01e-233 0       SLC2A5 
##  2 1.37e-236       2.30 0.776 0.297 2.34e-232 0       HOXA9  
##  3 9.82e-235       2.35 0.768 0.275 1.68e-230 0       HOPX   
##  4 3.44e-191       3.77 0.367 0.065 5.87e-187 0       PRSS2  
##  5 4.84e-107       2.27 0.904 0.667 8.26e-103 0       MT2A   
##  6 0               5.76 0.999 0.073 0         1       MS4A1  
##  7 0               4.30 0.939 0.138 0         1       BANK1  
##  8 0               4.04 0.692 0.029 0         1       FCER2  
##  9 0               3.86 0.926 0.273 0         1       RALGPS2
## 10 0               3.76 0.94  0.23  0         1       CD83   
## # … with 100 more rows</code></pre>
<div class="sourceCode" id="cb450"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb450-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb450-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DoHeatmap</span>(ABC, <span class="at">features =</span> top_markers<span class="sc">$</span>gene) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span></code></pre></div>
<p><img src="Single_cell_training_files/figure-html/all_markers_ABC-1.png" width="672" /></p>
<div class="sourceCode" id="cb451"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb451-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb451-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ABC, <span class="at">file=</span><span class="st">&quot;./ABC/ABC.RDS&quot;</span>)</span></code></pre></div>
</div>
<div id="marker基因展示-1" class="section level3" number="8.5.2">
<h3><span class="header-section-number">8.5.2</span> marker基因展示</h3>
<div class="sourceCode" id="cb452"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb452-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb452-1" aria-hidden="true" tabindex="-1"></a><span class="co"># marker_genes = c(&quot;AVP&quot;, &quot;CCR7&quot;, &quot;CD79E&quot;, &quot;GATA1&quot;, &quot;GZMH&quot;, &quot;SPI1&quot;)</span></span>
<span id="cb452-2"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb452-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb452-3"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb452-3" aria-hidden="true" tabindex="-1"></a><span class="do">## cluster 0</span></span>
<span id="cb452-4"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb452-4" aria-hidden="true" tabindex="-1"></a>marker_genes <span class="ot">&lt;-</span> top_markers<span class="sc">$</span>gene[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]</span>
<span id="cb452-5"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb452-5" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(ABC, <span class="at">features =</span>marker_genes, <span class="at">pt.size=</span> <span class="dv">0</span>)</span></code></pre></div>
<p><img src="Single_cell_training_files/figure-html/Marker_ABC_cluster0-1.png" width="1920" /></p>
<div class="sourceCode" id="cb453"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb453-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb453-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(ABC, <span class="at">features =</span> marker_genes )</span></code></pre></div>
<p><img src="Single_cell_training_files/figure-html/Marker_ABC_cluster0-2.png" width="1920" /></p>
</div>
</div>
<div id="sessioninfo-5" class="section level2" number="8.6">
<h2><span class="header-section-number">8.6</span> sessionInfo</h2>
<div class="sourceCode" id="cb454"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb454-1"><a href="single-cell-transcriptomic-landscape-of-human-blood-cells.html#cb454-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.4 (2021-02-15)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.10
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
##  [1] grid      parallel  stats4    stats     graphics  grDevices utils    
##  [8] datasets  methods   base     
## 
## other attached packages:
##  [1] Matrix_1.3-2            doParallel_1.0.16       iterators_1.0.13       
##  [4] foreach_1.5.1           ggalluvial_0.12.3       NMF_0.23.0             
##  [7] cluster_2.1.2           rngtools_1.5            pkgmaker_0.32.2        
## [10] registry_0.5-1          CellChat_1.0.0          igraph_1.2.6           
## [13] pheatmap_1.0.12         GSVA_1.38.2             enrichplot_1.10.2      
## [16] GSEABase_1.52.1         annotate_1.68.0         XML_3.99-0.6           
## [19] Rgraphviz_2.34.0        topGO_2.42.0            SparseM_1.81           
## [22] GO.db_3.12.1            graph_1.68.0            org.Hs.eg.db_3.12.0    
## [25] AnnotationDbi_1.52.0    IRanges_2.24.1          S4Vectors_0.28.1       
## [28] Biobase_2.50.0          BiocGenerics_0.36.1     clusterProfiler_3.18.1 
## [31] forcats_0.5.1           stringr_1.4.0           dplyr_1.0.5            
## [34] purrr_0.3.4             readr_1.4.0             tidyr_1.1.3            
## [37] tibble_3.1.1            ggplot2_3.3.3           tidyverse_1.3.1        
## [40] pbmc3k.SeuratData_3.1.4 SeuratData_0.2.1        SeuratObject_4.0.0     
## [43] Seurat_4.0.1           
## 
## loaded via a namespace (and not attached):
##   [1] rappdirs_0.3.3              rtracklayer_1.50.0         
##   [3] scattermore_0.7             KEGG.db_3.2.4              
##   [5] coda_0.19-4                 bit64_4.0.5                
##   [7] knitr_1.32                  irlba_2.3.3                
##   [9] DelayedArray_0.16.3         data.table_1.14.0          
##  [11] rpart_4.1-15                GEOquery_2.58.0            
##  [13] RCurl_1.98-1.3              generics_0.1.0             
##  [15] cowplot_1.1.1               RSQLite_2.2.5              
##  [17] shadowtext_0.0.7            RANN_2.6.1                 
##  [19] future_1.21.0               bit_4.0.4                  
##  [21] spatstat.data_2.1-0         xml2_1.3.2                 
##  [23] lubridate_1.7.10            httpuv_1.5.5               
##  [25] SummarizedExperiment_1.20.0 assertthat_0.2.1           
##  [27] viridis_0.6.0               xfun_0.22                  
##  [29] hms_1.0.0                   jquerylib_0.1.3            
##  [31] evaluate_0.14               promises_1.2.0.1           
##  [33] fansi_0.4.2                 dbplyr_2.1.1               
##  [35] readxl_1.3.1                DBI_1.1.1                  
##  [37] htmlwidgets_1.5.3           spatstat.geom_2.1-0        
##  [39] ellipsis_0.3.1              RSpectra_0.16-0            
##  [41] ggnewscale_0.4.5            backports_1.2.1            
##  [43] bookdown_0.21               gridBase_0.4-7             
##  [45] deldir_0.2-10               MatrixGenerics_1.2.1       
##  [47] vctrs_0.3.7                 Cairo_1.5-12.2             
##  [49] ROCR_1.0-11                 abind_1.4-5                
##  [51] cachem_1.0.4                withr_2.4.2                
##  [53] ggforce_0.3.3               sctransform_0.3.2          
##  [55] GenomicAlignments_1.26.0    sna_2.6                    
##  [57] goftest_1.2-2               svglite_2.0.0              
##  [59] DOSE_3.16.0                 lazyeval_0.2.2             
##  [61] crayon_1.4.1                hdf5r_1.3.3                
##  [63] pkgconfig_2.0.3             labeling_0.4.2             
##  [65] tweenr_1.0.2                GenomeInfoDb_1.26.7        
##  [67] nlme_3.1-152                rlang_0.4.10               
##  [69] globals_0.14.0              lifecycle_1.0.0            
##  [71] miniUI_0.1.1.1              downloader_0.4             
##  [73] modelr_0.1.8                cellranger_1.1.0           
##  [75] polyclip_1.10-0             matrixStats_0.58.0         
##  [77] lmtest_0.9-38               zoo_1.8-9                  
##  [79] reprex_2.0.0                GlobalOptions_0.1.2        
##  [81] ggridges_0.5.3              rjson_0.2.20               
##  [83] png_0.1-7                   viridisLite_0.4.0          
##  [85] bitops_1.0-6                KernSmooth_2.23-18         
##  [87] Biostrings_2.58.0           blob_1.2.1                 
##  [89] shape_1.4.5                 qvalue_2.22.0              
##  [91] parallelly_1.24.0           scales_1.1.1               
##  [93] memoise_2.0.0               magrittr_2.0.1             
##  [95] plyr_1.8.6                  ica_1.0-2                  
##  [97] zlibbioc_1.36.0             compiler_4.0.4             
##  [99] scatterpie_0.1.5            RColorBrewer_1.1-2         
## [101] clue_0.3-59                 fitdistrplus_1.1-3         
## [103] Rsamtools_2.6.0             cli_2.4.0                  
## [105] XVector_0.30.0              listenv_0.8.0              
## [107] patchwork_1.1.1             pbapply_1.4-3              
## [109] MASS_7.3-53.1               mgcv_1.8-35                
## [111] tidyselect_1.1.0            stringi_1.5.3              
## [113] gg.gap_1.3                  highr_0.9                  
## [115] yaml_2.2.1                  GOSemSim_2.16.1            
## [117] ggrepel_0.9.1               sass_0.3.1                 
## [119] fastmatch_1.1-0             tools_4.0.4                
## [121] future.apply_1.7.0          circlize_0.4.12            
## [123] rstudioapi_0.13             gridExtra_2.3              
## [125] farver_2.1.0                Rtsne_0.15                 
## [127] ggraph_2.0.5                digest_0.6.27              
## [129] rvcheck_0.1.8               BiocManager_1.30.12        
## [131] FNN_1.1.3                   shiny_1.6.0                
## [133] Rcpp_1.0.6                  GenomicRanges_1.42.0       
## [135] broom_0.7.6                 later_1.1.0.1              
## [137] RcppAnnoy_0.0.18            httr_1.4.2                 
## [139] ComplexHeatmap_2.6.2        colorspace_2.0-0           
## [141] rvest_1.0.0                 fs_1.5.0                   
## [143] tensor_1.5                  reticulate_1.18            
## [145] splines_4.0.4               uwot_0.1.10                
## [147] spatstat.utils_2.1-0        graphlayouts_0.7.1         
## [149] systemfonts_1.0.1           plotly_4.9.3               
## [151] xtable_1.8-4                jsonlite_1.7.2             
## [153] rle_0.9.2                   tidygraph_1.2.0            
## [155] R6_2.5.0                    pillar_1.6.0               
## [157] htmltools_0.5.1.1           mime_0.10                  
## [159] glue_1.4.2                  fastmap_1.1.0              
## [161] BiocParallel_1.24.1         codetools_0.2-18           
## [163] fgsea_1.16.0                utf8_1.2.1                 
## [165] lattice_0.20-41             bslib_0.2.4                
## [167] spatstat.sparse_2.0-0       network_1.16.1             
## [169] curl_4.3                    leiden_0.3.7               
## [171] survival_3.2-10             limma_3.46.0               
## [173] statnet.common_4.4.1        rmarkdown_2.7              
## [175] munsell_0.5.0               GetoptLong_1.0.5           
## [177] DO.db_2.9                   GenomeInfoDbData_1.2.4     
## [179] haven_2.4.0                 reshape2_1.4.4             
## [181] gtable_0.3.0                spatstat.core_2.1-2</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="ref-httpsgithub-comsqjincellchat.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Single_cell_training.pdf", "Single_cell_training.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
