<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 富集分析 | Single Cell Sequencing Analysis</title>
  <meta name="description" content="Training Class of Single Cell Sequencing Analysis." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 富集分析 | Single Cell Sequencing Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Training Class of Single Cell Sequencing Analysis." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 富集分析 | Single Cell Sequencing Analysis" />
  
  <meta name="twitter:description" content="Training Class of Single Cell Sequencing Analysis." />
  

<meta name="author" content="Zhilong Jia" />


<meta name="date" content="2021-04-21" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="pipeline.html"/>
<link rel="next" href="references.html"/>
<script src="libs/header-attrs-2.7/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Single Cell Sequencing Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Content</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="intro.html"><a href="intro.html#read-data-library-packages"><i class="fa fa-check"></i><b>2.1</b> Read data Library packages</a></li>
<li class="chapter" data-level="2.2" data-path="intro.html"><a href="intro.html#dimplot-cell-or-cluster"><i class="fa fa-check"></i><b>2.2</b> DimPlot : cell or cluster</a></li>
<li class="chapter" data-level="2.3" data-path="intro.html"><a href="intro.html#单细胞可视化"><i class="fa fa-check"></i><b>2.3</b> 单细胞可视化</a></li>
<li class="chapter" data-level="2.4" data-path="intro.html"><a href="intro.html#dimplot1"><i class="fa fa-check"></i><b>2.4</b> dimplot1</a></li>
<li class="chapter" data-level="2.5" data-path="intro.html"><a href="intro.html#比例"><i class="fa fa-check"></i><b>2.5</b> 比例</a></li>
<li class="chapter" data-level="2.6" data-path="intro.html"><a href="intro.html#数据转化"><i class="fa fa-check"></i><b>2.6</b> 数据转化</a></li>
<li class="chapter" data-level="2.7" data-path="intro.html"><a href="intro.html#sessioninfo"><i class="fa fa-check"></i><b>2.7</b> sessionInfo</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="a-introduction-to-single-cell-sequening.html"><a href="a-introduction-to-single-cell-sequening.html"><i class="fa fa-check"></i><b>3</b> A introduction to Single Cell Sequening</a>
<ul>
<li class="chapter" data-level="3.1" data-path="a-introduction-to-single-cell-sequening.html"><a href="a-introduction-to-single-cell-sequening.html#data-format-of-seurat"><i class="fa fa-check"></i><b>3.1</b> data format of Seurat</a></li>
<li class="chapter" data-level="3.2" data-path="a-introduction-to-single-cell-sequening.html"><a href="a-introduction-to-single-cell-sequening.html#data-access"><i class="fa fa-check"></i><b>3.2</b> Data Access</a></li>
<li class="chapter" data-level="3.3" data-path="a-introduction-to-single-cell-sequening.html"><a href="a-introduction-to-single-cell-sequening.html#methods"><i class="fa fa-check"></i><b>3.3</b> Methods</a></li>
<li class="chapter" data-level="3.4" data-path="a-introduction-to-single-cell-sequening.html"><a href="a-introduction-to-single-cell-sequening.html#test-read-h5-file."><i class="fa fa-check"></i><b>3.4</b> test read h5 file.</a></li>
<li class="chapter" data-level="3.5" data-path="a-introduction-to-single-cell-sequening.html"><a href="a-introduction-to-single-cell-sequening.html#reproduce-umap-fig-1a"><i class="fa fa-check"></i><b>3.5</b> reproduce UMAP fig 1a</a></li>
<li class="chapter" data-level="3.6" data-path="a-introduction-to-single-cell-sequening.html"><a href="a-introduction-to-single-cell-sequening.html#sessioninfo-1"><i class="fa fa-check"></i><b>3.6</b> sessionInfo</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="pipeline.html"><a href="pipeline.html"><i class="fa fa-check"></i><b>4</b> Pipeline</a>
<ul>
<li class="chapter" data-level="4.1" data-path="pipeline.html"><a href="pipeline.html#数据准备"><i class="fa fa-check"></i><b>4.1</b> 数据准备</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="pipeline.html"><a href="pipeline.html#下载测试数据"><i class="fa fa-check"></i><b>4.1.1</b> 下载测试数据</a></li>
<li class="chapter" data-level="4.1.2" data-path="pipeline.html"><a href="pipeline.html#载入表达矩阵"><i class="fa fa-check"></i><b>4.1.2</b> 载入表达矩阵</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="pipeline.html"><a href="pipeline.html#数据预处理"><i class="fa fa-check"></i><b>4.2</b> 数据预处理</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="pipeline.html"><a href="pipeline.html#构建seurat数据对象"><i class="fa fa-check"></i><b>4.2.1</b> 构建Seurat数据对象</a></li>
<li class="chapter" data-level="4.2.2" data-path="pipeline.html"><a href="pipeline.html#多个单细胞矩阵合并"><i class="fa fa-check"></i><b>4.2.2</b> 多个单细胞矩阵合并</a></li>
<li class="chapter" data-level="4.2.3" data-path="pipeline.html"><a href="pipeline.html#单细胞表达矩阵质控"><i class="fa fa-check"></i><b>4.2.3</b> 单细胞表达矩阵质控</a></li>
<li class="chapter" data-level="4.2.4" data-path="pipeline.html"><a href="pipeline.html#单细胞数据标准化和均一化"><i class="fa fa-check"></i><b>4.2.4</b> 单细胞数据标准化和均一化</a></li>
<li class="chapter" data-level="4.2.5" data-path="pipeline.html"><a href="pipeline.html#数据降维"><i class="fa fa-check"></i><b>4.2.5</b> 数据降维</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="pipeline.html"><a href="pipeline.html#单细胞聚类"><i class="fa fa-check"></i><b>4.3</b> 单细胞聚类</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="pipeline.html"><a href="pipeline.html#聚类分簇"><i class="fa fa-check"></i><b>4.3.1</b> 聚类分簇</a></li>
<li class="chapter" data-level="4.3.2" data-path="pipeline.html"><a href="pipeline.html#单细胞聚类可视化"><i class="fa fa-check"></i><b>4.3.2</b> 单细胞聚类可视化</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="pipeline.html"><a href="pipeline.html#单细胞聚类簇类型鉴定differential-expression-analysis"><i class="fa fa-check"></i><b>4.4</b> 单细胞聚类簇类型鉴定(Differential expression analysis)</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="pipeline.html"><a href="pipeline.html#细胞簇差异基因分析"><i class="fa fa-check"></i><b>4.4.1</b> 细胞簇差异基因分析</a></li>
<li class="chapter" data-level="4.4.2" data-path="pipeline.html"><a href="pipeline.html#marker基因展示"><i class="fa fa-check"></i><b>4.4.2</b> marker基因展示</a></li>
<li class="chapter" data-level="4.4.3" data-path="pipeline.html"><a href="pipeline.html#细胞类型标签注释"><i class="fa fa-check"></i><b>4.4.3</b> 细胞类型标签注释</a></li>
<li class="chapter" data-level="4.4.4" data-path="pipeline.html"><a href="pipeline.html#细胞类型比例可视化"><i class="fa fa-check"></i><b>4.4.4</b> 细胞类型比例可视化</a></li>
<li class="chapter" data-level="4.4.5" data-path="pipeline.html"><a href="pipeline.html#细胞类型marker基因展示"><i class="fa fa-check"></i><b>4.4.5</b> 细胞类型marker基因展示</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="pipeline.html"><a href="pipeline.html#sessioninfo-2"><i class="fa fa-check"></i><b>4.5</b> sessionInfo</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="富集分析.html"><a href="富集分析.html"><i class="fa fa-check"></i><b>5</b> 富集分析</a>
<ul>
<li class="chapter" data-level="5.1" data-path="富集分析.html"><a href="富集分析.html#获取差异基因"><i class="fa fa-check"></i><b>5.1</b> 获取差异基因</a></li>
<li class="chapter" data-level="5.2" data-path="富集分析.html"><a href="富集分析.html#go富集分析"><i class="fa fa-check"></i><b>5.2</b> GO富集分析</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="富集分析.html"><a href="富集分析.html#go富集分析-1"><i class="fa fa-check"></i><b>5.2.1</b> GO富集分析</a></li>
<li class="chapter" data-level="5.2.2" data-path="富集分析.html"><a href="富集分析.html#go富集结果可视化"><i class="fa fa-check"></i><b>5.2.2</b> GO富集结果可视化</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="富集分析.html"><a href="富集分析.html#kegg富集分析"><i class="fa fa-check"></i><b>5.3</b> KEGG富集分析</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="富集分析.html"><a href="富集分析.html#kegg富集分析代码"><i class="fa fa-check"></i><b>5.3.1</b> KEGG富集分析代码</a></li>
<li class="chapter" data-level="5.3.2" data-path="富集分析.html"><a href="富集分析.html#kegg结果可视化"><i class="fa fa-check"></i><b>5.3.2</b> KEGG结果可视化</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="富集分析.html"><a href="富集分析.html#gsea富集分析"><i class="fa fa-check"></i><b>5.4</b> GSEA富集分析</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="富集分析.html"><a href="富集分析.html#gsea结果可视化"><i class="fa fa-check"></i><b>5.4.1</b> GSEA结果可视化</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="富集分析.html"><a href="富集分析.html#gsva富集分析"><i class="fa fa-check"></i><b>5.5</b> GSVA富集分析</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="富集分析.html"><a href="富集分析.html#gsva结果可视化"><i class="fa fa-check"></i><b>5.5.1</b> GSVA结果可视化</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="富集分析.html"><a href="富集分析.html#sessioninfo-3"><i class="fa fa-check"></i><b>5.6</b> sessionInfo</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>6</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Single Cell Sequencing Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="富集分析" class="section level1" number="5">
<h1><span class="header-section-number">Chapter 5</span> 富集分析</h1>
<p>Enrichment Using clusterprofiler based on <code>enrich.html</code></p>
<div class="figure">
<img src="figure/clusterProfiler-diagram.png" alt="" />
<p class="caption">clusterProfiler-diagram</p>
</div>
<p>ref: <a href="http://yulab-smu.top/clusterProfiler-book/index.html" class="uri">http://yulab-smu.top/clusterProfiler-book/index.html</a></p>
<div id="获取差异基因" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> 获取差异基因</h2>
<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb162-1"><a href="富集分析.html#cb162-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb162-2"><a href="富集分析.html#cb162-2" aria-hidden="true" tabindex="-1"></a>pbmc <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/PRO_seurat.RDS&quot;</span>)</span>
<span id="cb162-3"><a href="富集分析.html#cb162-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb162-4"><a href="富集分析.html#cb162-4" aria-hidden="true" tabindex="-1"></a>pbmc<span class="sc">@</span>meta.data<span class="sc">$</span>celltype <span class="ot">&lt;-</span> pbmc<span class="sc">@</span>active.ident</span>
<span id="cb162-5"><a href="富集分析.html#cb162-5" aria-hidden="true" tabindex="-1"></a><span class="co">#获取细胞间的差异基因</span></span>
<span id="cb162-6"><a href="富集分析.html#cb162-6" aria-hidden="true" tabindex="-1"></a>cells1 <span class="ot">&lt;-</span> <span class="fu">subset</span>(pbmc<span class="sc">@</span>meta.data, celltype <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;CD14+ Mono&quot;</span>))  <span class="sc">%&gt;%</span> <span class="fu">rownames</span>()</span>
<span id="cb162-7"><a href="富集分析.html#cb162-7" aria-hidden="true" tabindex="-1"></a>cells2 <span class="ot">&lt;-</span> <span class="fu">subset</span>(pbmc<span class="sc">@</span>meta.data, celltype <span class="sc">%in%</span>  <span class="fu">c</span>(<span class="st">&quot;FCGR3A+ Mono&quot;</span>))  <span class="sc">%&gt;%</span> <span class="fu">rownames</span>()</span>
<span id="cb162-8"><a href="富集分析.html#cb162-8" aria-hidden="true" tabindex="-1"></a>deg <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(pbmc, <span class="at">ident.1 =</span> cells1, <span class="at">ident.2 =</span> cells2)</span>
<span id="cb162-9"><a href="富集分析.html#cb162-9" aria-hidden="true" tabindex="-1"></a>deg <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene =</span> <span class="fu">rownames</span>(deg), deg)</span>
<span id="cb162-10"><a href="富集分析.html#cb162-10" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(deg)</span></code></pre></div>
<pre><code>##          gene         p_val avg_log2FC pct.1 pct.2     p_val_adj
## FCGR3A FCGR3A 1.097516e-116  -4.287857 0.046 0.981 3.593049e-112
## CFD       CFD 2.601448e-114  -3.500834 0.026 0.944 8.516621e-110
## IFITM3 IFITM3 1.629039e-113  -3.880574 0.055 0.981 5.333147e-109
## FCER1G FCER1G 1.997252e-112  -4.859236 0.092 1.000 6.538605e-108
## TYROBP TYROBP 4.882748e-106  -4.782237 0.138 1.000 1.598514e-101
## CD68     CD68 1.556965e-105  -3.043349 0.044 0.919 5.097192e-101</code></pre>
<div class="sourceCode" id="cb164"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb164-1"><a href="富集分析.html#cb164-1" aria-hidden="true" tabindex="-1"></a>deg1 <span class="ot">&lt;-</span> deg</span>
<span id="cb164-2"><a href="富集分析.html#cb164-2" aria-hidden="true" tabindex="-1"></a>logFC_t<span class="ot">=</span><span class="fl">0.5</span></span>
<span id="cb164-3"><a href="富集分析.html#cb164-3" aria-hidden="true" tabindex="-1"></a>P.Value_t <span class="ot">=</span> <span class="fl">0.05</span></span>
<span id="cb164-4"><a href="富集分析.html#cb164-4" aria-hidden="true" tabindex="-1"></a>k1 <span class="ot">=</span> (deg1<span class="sc">$</span>p_val_adj <span class="sc">&lt;</span> P.Value_t)<span class="sc">&amp;</span>(deg1<span class="sc">$</span>avg_log2FC <span class="sc">&lt;</span> <span class="sc">-</span>logFC_t)</span>
<span id="cb164-5"><a href="富集分析.html#cb164-5" aria-hidden="true" tabindex="-1"></a>k2 <span class="ot">=</span> (deg1<span class="sc">$</span>p_val_adj <span class="sc">&lt;</span> P.Value_t)<span class="sc">&amp;</span>(deg1<span class="sc">$</span>avg_log2FC <span class="sc">&gt;</span> logFC_t)</span>
<span id="cb164-6"><a href="富集分析.html#cb164-6" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(k1)</span></code></pre></div>
<pre><code>## k1
## FALSE  TRUE 
##  1376   540</code></pre>
<div class="sourceCode" id="cb166"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb166-1"><a href="富集分析.html#cb166-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(k2)</span></code></pre></div>
<pre><code>## k2
## FALSE  TRUE 
##  1712   204</code></pre>
<div class="sourceCode" id="cb168"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb168-1"><a href="富集分析.html#cb168-1" aria-hidden="true" tabindex="-1"></a><span class="co">#分组 上调/下调</span></span>
<span id="cb168-2"><a href="富集分析.html#cb168-2" aria-hidden="true" tabindex="-1"></a>change <span class="ot">=</span> <span class="fu">ifelse</span>(k1,<span class="st">&quot;down&quot;</span>,<span class="fu">ifelse</span>(k2,<span class="st">&quot;up&quot;</span>,<span class="st">&quot;stable&quot;</span>))</span>
<span id="cb168-3"><a href="富集分析.html#cb168-3" aria-hidden="true" tabindex="-1"></a>deg1<span class="sc">$</span>change <span class="ot">&lt;-</span> change</span>
<span id="cb168-4"><a href="富集分析.html#cb168-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(deg1)</span></code></pre></div>
<pre><code>##          gene         p_val avg_log2FC pct.1 pct.2     p_val_adj change
## FCGR3A FCGR3A 1.097516e-116  -4.287857 0.046 0.981 3.593049e-112   down
## CFD       CFD 2.601448e-114  -3.500834 0.026 0.944 8.516621e-110   down
## IFITM3 IFITM3 1.629039e-113  -3.880574 0.055 0.981 5.333147e-109   down
## FCER1G FCER1G 1.997252e-112  -4.859236 0.092 1.000 6.538605e-108   down
## TYROBP TYROBP 4.882748e-106  -4.782237 0.138 1.000 1.598514e-101   down
## CD68     CD68 1.556965e-105  -3.043349 0.044 0.919 5.097192e-101   down</code></pre>
<div class="sourceCode" id="cb170"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb170-1"><a href="富集分析.html#cb170-1" aria-hidden="true" tabindex="-1"></a><span class="co">#基因名称转换</span></span>
<span id="cb170-2"><a href="富集分析.html#cb170-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;clusterProfiler&quot;</span>)</span></code></pre></div>
<pre><code>## </code></pre>
<pre><code>## clusterProfiler v3.18.1  For help: https://guangchuangyu.github.io/software/clusterProfiler
## 
## If you use clusterProfiler in published research, please cite:
## Guangchuang Yu, Li-Gen Wang, Yanyan Han, Qing-Yu He. clusterProfiler: an R package for comparing biological themes among gene clusters. OMICS: A Journal of Integrative Biology. 2012, 16(5):284-287.</code></pre>
<pre><code>## 
## Attaching package: &#39;clusterProfiler&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:purrr&#39;:
## 
##     simplify</code></pre>
<pre><code>## The following object is masked from &#39;package:stats&#39;:
## 
##     filter</code></pre>
<div class="sourceCode" id="cb176"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb176-1"><a href="富集分析.html#cb176-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;org.Hs.eg.db&quot;</span>)</span></code></pre></div>
<pre><code>## Loading required package: AnnotationDbi</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## 
## Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:parallel&#39;:
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     combine, intersect, setdiff, union</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     &#39;browseVignettes()&#39;. To cite Bioconductor, see
##     &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## 
## Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:clusterProfiler&#39;:
## 
##     rename</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     first, rename</code></pre>
<pre><code>## The following object is masked from &#39;package:tidyr&#39;:
## 
##     expand</code></pre>
<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     expand.grid</code></pre>
<pre><code>## 
## Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:clusterProfiler&#39;:
## 
##     slice</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     collapse, desc, slice</code></pre>
<pre><code>## The following object is masked from &#39;package:purrr&#39;:
## 
##     reduce</code></pre>
<pre><code>## 
## Attaching package: &#39;AnnotationDbi&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:clusterProfiler&#39;:
## 
##     select</code></pre>
<pre><code>## The following object is masked from &#39;package:dplyr&#39;:
## 
##     select</code></pre>
<pre><code>## </code></pre>
<div class="sourceCode" id="cb203"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb203-1"><a href="富集分析.html#cb203-1" aria-hidden="true" tabindex="-1"></a>s2e <span class="ot">&lt;-</span> <span class="fu">bitr</span>(deg1<span class="sc">$</span>gene, </span>
<span id="cb203-2"><a href="富集分析.html#cb203-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">fromType =</span> <span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb203-3"><a href="富集分析.html#cb203-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">toType =</span> <span class="st">&quot;ENTREZID&quot;</span>,</span>
<span id="cb203-4"><a href="富集分析.html#cb203-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">OrgDb =</span> org.Hs.eg.db)<span class="co">#人类 转换成ENTREZID</span></span></code></pre></div>
<pre><code>## &#39;select()&#39; returned 1:1 mapping between keys and columns</code></pre>
<pre><code>## Warning in bitr(deg1$gene, fromType = &quot;SYMBOL&quot;, toType = &quot;ENTREZID&quot;, OrgDb =
## org.Hs.eg.db): 6.68% of input gene IDs are fail to map...</code></pre>
<div class="sourceCode" id="cb206"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb206-1"><a href="富集分析.html#cb206-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb206-2"><a href="富集分析.html#cb206-2" aria-hidden="true" tabindex="-1"></a>deg1 <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(deg1,s2e,<span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;gene&quot;</span><span class="ot">=</span><span class="st">&quot;SYMBOL&quot;</span>))</span>
<span id="cb206-3"><a href="富集分析.html#cb206-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(deg1)</span></code></pre></div>
<pre><code>##     gene         p_val avg_log2FC pct.1 pct.2     p_val_adj change ENTREZID
## 1 FCGR3A 1.097516e-116  -4.287857 0.046 0.981 3.593049e-112   down     2214
## 2    CFD 2.601448e-114  -3.500834 0.026 0.944 8.516621e-110   down     1675
## 3 IFITM3 1.629039e-113  -3.880574 0.055 0.981 5.333147e-109   down    10410
## 4 FCER1G 1.997252e-112  -4.859236 0.092 1.000 6.538605e-108   down     2207
## 5 TYROBP 4.882748e-106  -4.782237 0.138 1.000 1.598514e-101   down     7305
## 6   CD68 1.556965e-105  -3.043349 0.044 0.919 5.097192e-101   down      968</code></pre>
<div class="sourceCode" id="cb208"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb208-1"><a href="富集分析.html#cb208-1" aria-hidden="true" tabindex="-1"></a><span class="co">#GO富集分析差异基因列表[Symbol]</span></span>
<span id="cb208-2"><a href="富集分析.html#cb208-2" aria-hidden="true" tabindex="-1"></a>gene_up <span class="ot">=</span> deg1[deg1<span class="sc">$</span>change <span class="sc">==</span> <span class="st">&#39;up&#39;</span>,<span class="st">&#39;gene&#39;</span>] </span>
<span id="cb208-3"><a href="富集分析.html#cb208-3" aria-hidden="true" tabindex="-1"></a>gene_down <span class="ot">=</span> deg1[deg1<span class="sc">$</span>change <span class="sc">==</span> <span class="st">&#39;down&#39;</span>,<span class="st">&#39;gene&#39;</span>] </span>
<span id="cb208-4"><a href="富集分析.html#cb208-4" aria-hidden="true" tabindex="-1"></a>gene_diff <span class="ot">=</span> <span class="fu">c</span>(gene_up,gene_down)</span>
<span id="cb208-5"><a href="富集分析.html#cb208-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb208-6"><a href="富集分析.html#cb208-6" aria-hidden="true" tabindex="-1"></a><span class="co">#KEGG富集分析差异基因列表[ENTREZID]</span></span>
<span id="cb208-7"><a href="富集分析.html#cb208-7" aria-hidden="true" tabindex="-1"></a>gene_all <span class="ot">=</span> deg1[,<span class="st">&#39;ENTREZID&#39;</span>]</span>
<span id="cb208-8"><a href="富集分析.html#cb208-8" aria-hidden="true" tabindex="-1"></a>gene_up_KEGG <span class="ot">=</span> deg1[deg1<span class="sc">$</span>change <span class="sc">==</span> <span class="st">&#39;up&#39;</span>,<span class="st">&#39;ENTREZID&#39;</span>]</span>
<span id="cb208-9"><a href="富集分析.html#cb208-9" aria-hidden="true" tabindex="-1"></a>gene_down_KEGG <span class="ot">=</span> deg1[deg1<span class="sc">$</span>change <span class="sc">==</span> <span class="st">&#39;down&#39;</span>,<span class="st">&#39;ENTREZID&#39;</span>]</span>
<span id="cb208-10"><a href="富集分析.html#cb208-10" aria-hidden="true" tabindex="-1"></a>gene_diff_KEGG <span class="ot">=</span> <span class="fu">c</span>(gene_up_KEGG,gene_down_KEGG)</span></code></pre></div>
</div>
<div id="go富集分析" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> GO富集分析</h2>
<div id="go富集分析-1" class="section level3" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> GO富集分析</h3>
<div class="sourceCode" id="cb209"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb209-1"><a href="富集分析.html#cb209-1" aria-hidden="true" tabindex="-1"></a><span class="co">#细胞组分</span></span>
<span id="cb209-2"><a href="富集分析.html#cb209-2" aria-hidden="true" tabindex="-1"></a>ego_CC <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene          =</span> gene_up,</span>
<span id="cb209-3"><a href="富集分析.html#cb209-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">keyType       =</span> <span class="st">&#39;SYMBOL&#39;</span>, <span class="co">#基因ID的类型</span></span>
<span id="cb209-4"><a href="富集分析.html#cb209-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">OrgDb         =</span> org.Hs.eg.db,  <span class="co">#包含人注释信息的数据库</span></span>
<span id="cb209-5"><a href="富集分析.html#cb209-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">ont           =</span> <span class="st">&quot;CC&quot;</span>,</span>
<span id="cb209-6"><a href="富集分析.html#cb209-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>, <span class="co">#指定多重假设检验矫正的方法</span></span>
<span id="cb209-7"><a href="富集分析.html#cb209-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">pvalueCutoff  =</span> <span class="fl">0.01</span>,</span>
<span id="cb209-8"><a href="富集分析.html#cb209-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">qvalueCutoff  =</span> <span class="fl">0.05</span>)</span>
<span id="cb209-9"><a href="富集分析.html#cb209-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">summary</span>(ego_CC))</span></code></pre></div>
<pre><code>## Warning in summary(ego_CC): summary method to convert the object to data.frame
## is deprecated, please use as.data.frame instead.</code></pre>
<pre><code>##                    ID                       Description GeneRatio   BgRatio
## GO:0022626 GO:0022626                cytosolic ribosome    66/192 110/19559
## GO:0044391 GO:0044391                 ribosomal subunit    66/192 187/19559
## GO:0005840 GO:0005840                          ribosome    66/192 270/19559
## GO:0022625 GO:0022625 cytosolic large ribosomal subunit    39/192  58/19559
## GO:0015934 GO:0015934           large ribosomal subunit    39/192 116/19559
## GO:0022627 GO:0022627 cytosolic small ribosomal subunit    27/192  47/19559
##                   pvalue      p.adjust        qvalue
## GO:0022626 8.168314e-108 1.895049e-105 1.693850e-105
## GO:0044391  1.478001e-87  1.714481e-85  1.532453e-85
## GO:0005840  2.456292e-75  1.899533e-73  1.697858e-73
## GO:0022625  6.530221e-66  3.787528e-64  3.385404e-64
## GO:0015934  5.056823e-50  2.346366e-48  2.097251e-48
## GO:0022627  7.513437e-43  2.905196e-41  2.596749e-41
##                                                                                                                                                                                                                                                                                                                                                                                                               geneID
## GO:0022626 RPS18/RPS27/RPS3/RPL13A/RPL3/RPL13/RPS27A/RPS25/RPS12/RPL23A/RPSA/RPLP2/RPS15A/RPS6/RPL31/RPL9/RPL32/RPS29/RPLP0/RPS8/RPL30/RPL11/RPL27A/RPS7/RPS14/RPS10/RPL35A/RPLP1/RPL5/RPL19/RPS3A/RPS20/RPS4X/RPS5/RPL10A/RPL36/RPS28/RPS2/RPS13/RPL10/RPL14/RPL17/RPL15/RPS16/RPL21/RPL6/RPS23/RPS15/RPL7A/RPL18/RPL18A/RPL35/RPS26/RPS21/RPL37A/RPL4/RPL29/RPL27/RPL7/RPL24/RPL37/RPS4Y1/RPL38/RPL22/RPL36A/RPL34
## GO:0044391 RPS18/RPS27/RPS3/RPL13A/RPL3/RPL13/RPS27A/RPS25/RPS12/RPL23A/RPSA/RPLP2/RPS15A/RPS6/RPL31/RPL9/RPL32/RPS29/RPLP0/RPS8/RPL30/RPL11/RPL27A/RPS7/RPS14/RPS10/RPL35A/RPLP1/RPL5/RPL19/RPS3A/RPS20/RPS4X/RPS5/RPL10A/RPL36/RPS28/RPS2/RPS13/RPL10/RPL14/RPL17/RPL15/RPS16/RPL21/RPL6/RPS23/RPS15/RPL7A/RPL18/RPL18A/RPL35/RPS26/RPS21/RPL37A/RPL4/RPL29/RPL27/RPL7/RPL24/RPL37/RPS4Y1/RPL38/RPL22/RPL36A/RPL34
## GO:0005840 RPS18/RPS27/RPS3/RPL13A/RPL3/RPL13/RPS27A/RPS25/RPS12/RPL23A/RPSA/RPLP2/RPS15A/RPS6/RPL31/RPL9/RPL32/RPS29/RPLP0/RPS8/RPL30/RPL11/RPL27A/RPS7/RPS14/RPS10/RPL35A/RPLP1/RPL5/RPL19/RPS3A/RPS20/RPS4X/RPS5/RPL10A/RPL36/RPS28/RPS2/RPS13/RPL10/RPL14/RPL17/RPL15/RPS16/RPL21/RPL6/RPS23/RPS15/RPL7A/RPL18/RPL18A/RPL35/RPS26/RPS21/RPL37A/RPL4/RPL29/RPL27/RPL7/RPL24/RPL37/RPS4Y1/RPL38/RPL22/RPL36A/RPL34
## GO:0022625                                                                                                                                                               RPL13A/RPL3/RPL13/RPL23A/RPLP2/RPL31/RPL9/RPL32/RPLP0/RPL30/RPL11/RPL27A/RPL35A/RPLP1/RPL5/RPL19/RPL10A/RPL36/RPL10/RPL14/RPL17/RPL15/RPL21/RPL6/RPL7A/RPL18/RPL18A/RPL35/RPL37A/RPL4/RPL29/RPL27/RPL7/RPL24/RPL37/RPL38/RPL22/RPL36A/RPL34
## GO:0015934                                                                                                                                                               RPL13A/RPL3/RPL13/RPL23A/RPLP2/RPL31/RPL9/RPL32/RPLP0/RPL30/RPL11/RPL27A/RPL35A/RPLP1/RPL5/RPL19/RPL10A/RPL36/RPL10/RPL14/RPL17/RPL15/RPL21/RPL6/RPL7A/RPL18/RPL18A/RPL35/RPL37A/RPL4/RPL29/RPL27/RPL7/RPL24/RPL37/RPL38/RPL22/RPL36A/RPL34
## GO:0022627                                                                                                                                                                                                                                             RPS18/RPS27/RPS3/RPS27A/RPS25/RPS12/RPSA/RPS15A/RPS6/RPS29/RPS8/RPS7/RPS14/RPS10/RPS3A/RPS20/RPS4X/RPS5/RPS28/RPS2/RPS13/RPS16/RPS23/RPS15/RPS26/RPS21/RPS4Y1
##            Count
## GO:0022626    66
## GO:0044391    66
## GO:0005840    66
## GO:0022625    39
## GO:0015934    39
## GO:0022627    27</code></pre>
<div class="sourceCode" id="cb212"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb212-1"><a href="富集分析.html#cb212-1" aria-hidden="true" tabindex="-1"></a><span class="co">#生物过程</span></span>
<span id="cb212-2"><a href="富集分析.html#cb212-2" aria-hidden="true" tabindex="-1"></a>ego_BP <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene          =</span> gene_up,</span>
<span id="cb212-3"><a href="富集分析.html#cb212-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">OrgDb          =</span> org.Hs.eg.db,</span>
<span id="cb212-4"><a href="富集分析.html#cb212-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">keyType       =</span> <span class="st">&#39;SYMBOL&#39;</span>,</span>
<span id="cb212-5"><a href="富集分析.html#cb212-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">ont           =</span> <span class="st">&quot;BP&quot;</span>,</span>
<span id="cb212-6"><a href="富集分析.html#cb212-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb212-7"><a href="富集分析.html#cb212-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">pvalueCutoff  =</span> <span class="fl">0.01</span>,</span>
<span id="cb212-8"><a href="富集分析.html#cb212-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">qvalueCutoff  =</span> <span class="fl">0.05</span>)</span>
<span id="cb212-9"><a href="富集分析.html#cb212-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-10"><a href="富集分析.html#cb212-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-11"><a href="富集分析.html#cb212-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-12"><a href="富集分析.html#cb212-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-13"><a href="富集分析.html#cb212-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-14"><a href="富集分析.html#cb212-14" aria-hidden="true" tabindex="-1"></a><span class="co">#分子功能</span></span>
<span id="cb212-15"><a href="富集分析.html#cb212-15" aria-hidden="true" tabindex="-1"></a>ego_MF <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene          =</span> gene_up,</span>
<span id="cb212-16"><a href="富集分析.html#cb212-16" aria-hidden="true" tabindex="-1"></a>                <span class="at">OrgDb         =</span> org.Hs.eg.db,</span>
<span id="cb212-17"><a href="富集分析.html#cb212-17" aria-hidden="true" tabindex="-1"></a>                 <span class="at">keyType       =</span> <span class="st">&#39;SYMBOL&#39;</span>,</span>
<span id="cb212-18"><a href="富集分析.html#cb212-18" aria-hidden="true" tabindex="-1"></a>                <span class="at">ont           =</span> <span class="st">&quot;MF&quot;</span>,</span>
<span id="cb212-19"><a href="富集分析.html#cb212-19" aria-hidden="true" tabindex="-1"></a>                <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb212-20"><a href="富集分析.html#cb212-20" aria-hidden="true" tabindex="-1"></a>                <span class="at">pvalueCutoff  =</span> <span class="fl">0.01</span>,</span>
<span id="cb212-21"><a href="富集分析.html#cb212-21" aria-hidden="true" tabindex="-1"></a>                <span class="at">qvalueCutoff  =</span> <span class="fl">0.05</span>)</span>
<span id="cb212-22"><a href="富集分析.html#cb212-22" aria-hidden="true" tabindex="-1"></a><span class="co"># save(ego_CC,ego_BP,ego_MF,file = &quot;GO.Rdata&quot;)</span></span>
<span id="cb212-23"><a href="富集分析.html#cb212-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-24"><a href="富集分析.html#cb212-24" aria-hidden="true" tabindex="-1"></a><span class="co">#细胞组分、分子功能、生物学过程</span></span>
<span id="cb212-25"><a href="富集分析.html#cb212-25" aria-hidden="true" tabindex="-1"></a>go <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> gene_up, <span class="at">OrgDb =</span> <span class="st">&quot;org.Hs.eg.db&quot;</span>, <span class="at">keyType       =</span> <span class="st">&#39;SYMBOL&#39;</span>,<span class="at">ont=</span><span class="st">&quot;all&quot;</span>)</span>
<span id="cb212-26"><a href="富集分析.html#cb212-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-27"><a href="富集分析.html#cb212-27" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(go)</span></code></pre></div>
<pre><code>##            ONTOLOGY         ID
## GO:0006614       BP GO:0006614
## GO:0006613       BP GO:0006613
## GO:0000184       BP GO:0000184
## GO:0045047       BP GO:0045047
## GO:0072599       BP GO:0072599
## GO:0070972       BP GO:0070972
##                                                                    Description
## GO:0006614         SRP-dependent cotranslational protein targeting to membrane
## GO:0006613                       cotranslational protein targeting to membrane
## GO:0000184 nuclear-transcribed mRNA catabolic process, nonsense-mediated decay
## GO:0045047                                             protein targeting to ER
## GO:0072599      establishment of protein localization to endoplasmic reticulum
## GO:0070972                       protein localization to endoplasmic reticulum
##            GeneRatio   BgRatio        pvalue      p.adjust        qvalue
## GO:0006614    66/186 105/18866 6.243599e-110 1.688894e-106 1.520809e-106
## GO:0006613    66/186 109/18866 2.744310e-108 3.711680e-105 3.342281e-105
## GO:0000184    66/186 120/18866 3.104015e-104 2.099090e-101 1.890182e-101
## GO:0045047    66/186 120/18866 3.104015e-104 2.099090e-101 1.890182e-101
## GO:0072599    66/186 124/18866 6.691966e-103 3.620354e-100 3.260044e-100
## GO:0070972    66/186 152/18866  4.734663e-95  2.134544e-92  1.922107e-92
##                                                                                                                                                                                                                                                                                                                                                                                                               geneID
## GO:0006614 RPS18/RPS27/RPS3/RPL13A/RPL3/RPL13/RPS27A/RPS25/RPS12/RPL23A/RPSA/RPLP2/RPS15A/RPS6/RPL31/RPL9/RPL32/RPS29/RPLP0/RPS8/RPL30/RPL11/RPL27A/RPS7/RPS14/RPS10/RPL35A/RPLP1/RPL5/RPL19/RPS3A/RPS20/RPS4X/RPS5/RPL10A/RPL36/RPS28/RPS2/RPS13/RPL10/RPL14/RPL17/RPL15/RPS16/RPL21/RPL6/RPS23/RPS15/RPL7A/RPL18/RPL18A/RPL35/RPS26/RPS21/RPL37A/RPL4/RPL29/RPL27/RPL7/RPL24/RPL37/RPS4Y1/RPL38/RPL22/RPL36A/RPL34
## GO:0006613 RPS18/RPS27/RPS3/RPL13A/RPL3/RPL13/RPS27A/RPS25/RPS12/RPL23A/RPSA/RPLP2/RPS15A/RPS6/RPL31/RPL9/RPL32/RPS29/RPLP0/RPS8/RPL30/RPL11/RPL27A/RPS7/RPS14/RPS10/RPL35A/RPLP1/RPL5/RPL19/RPS3A/RPS20/RPS4X/RPS5/RPL10A/RPL36/RPS28/RPS2/RPS13/RPL10/RPL14/RPL17/RPL15/RPS16/RPL21/RPL6/RPS23/RPS15/RPL7A/RPL18/RPL18A/RPL35/RPS26/RPS21/RPL37A/RPL4/RPL29/RPL27/RPL7/RPL24/RPL37/RPS4Y1/RPL38/RPL22/RPL36A/RPL34
## GO:0000184 RPS18/RPS27/RPS3/RPL13A/RPL3/RPL13/RPS27A/RPS25/RPS12/RPL23A/RPSA/RPLP2/RPS15A/RPS6/RPL31/RPL9/RPL32/RPS29/RPLP0/RPS8/RPL30/RPL11/RPL27A/RPS7/RPS14/RPS10/RPL35A/RPLP1/RPL5/RPL19/RPS3A/RPS20/RPS4X/RPS5/RPL10A/RPL36/RPS28/RPS2/RPS13/RPL10/RPL14/RPL17/RPL15/RPS16/RPL21/RPL6/RPS23/RPS15/RPL7A/RPL18/RPL18A/RPL35/RPS26/RPS21/RPL37A/RPL4/RPL29/RPL27/RPL7/RPL24/RPL37/RPS4Y1/RPL38/RPL22/RPL36A/RPL34
## GO:0045047 RPS18/RPS27/RPS3/RPL13A/RPL3/RPL13/RPS27A/RPS25/RPS12/RPL23A/RPSA/RPLP2/RPS15A/RPS6/RPL31/RPL9/RPL32/RPS29/RPLP0/RPS8/RPL30/RPL11/RPL27A/RPS7/RPS14/RPS10/RPL35A/RPLP1/RPL5/RPL19/RPS3A/RPS20/RPS4X/RPS5/RPL10A/RPL36/RPS28/RPS2/RPS13/RPL10/RPL14/RPL17/RPL15/RPS16/RPL21/RPL6/RPS23/RPS15/RPL7A/RPL18/RPL18A/RPL35/RPS26/RPS21/RPL37A/RPL4/RPL29/RPL27/RPL7/RPL24/RPL37/RPS4Y1/RPL38/RPL22/RPL36A/RPL34
## GO:0072599 RPS18/RPS27/RPS3/RPL13A/RPL3/RPL13/RPS27A/RPS25/RPS12/RPL23A/RPSA/RPLP2/RPS15A/RPS6/RPL31/RPL9/RPL32/RPS29/RPLP0/RPS8/RPL30/RPL11/RPL27A/RPS7/RPS14/RPS10/RPL35A/RPLP1/RPL5/RPL19/RPS3A/RPS20/RPS4X/RPS5/RPL10A/RPL36/RPS28/RPS2/RPS13/RPL10/RPL14/RPL17/RPL15/RPS16/RPL21/RPL6/RPS23/RPS15/RPL7A/RPL18/RPL18A/RPL35/RPS26/RPS21/RPL37A/RPL4/RPL29/RPL27/RPL7/RPL24/RPL37/RPS4Y1/RPL38/RPL22/RPL36A/RPL34
## GO:0070972 RPS18/RPS27/RPS3/RPL13A/RPL3/RPL13/RPS27A/RPS25/RPS12/RPL23A/RPSA/RPLP2/RPS15A/RPS6/RPL31/RPL9/RPL32/RPS29/RPLP0/RPS8/RPL30/RPL11/RPL27A/RPS7/RPS14/RPS10/RPL35A/RPLP1/RPL5/RPL19/RPS3A/RPS20/RPS4X/RPS5/RPL10A/RPL36/RPS28/RPS2/RPS13/RPL10/RPL14/RPL17/RPL15/RPS16/RPL21/RPL6/RPS23/RPS15/RPL7A/RPL18/RPL18A/RPL35/RPS26/RPS21/RPL37A/RPL4/RPL29/RPL27/RPL7/RPL24/RPL37/RPS4Y1/RPL38/RPL22/RPL36A/RPL34
##            Count
## GO:0006614    66
## GO:0006613    66
## GO:0000184    66
## GO:0045047    66
## GO:0072599    66
## GO:0070972    66</code></pre>
</div>
<div id="go富集结果可视化" class="section level3" number="5.2.2">
<h3><span class="header-section-number">5.2.2</span> GO富集结果可视化</h3>
<div class="sourceCode" id="cb214"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb214-1"><a href="富集分析.html#cb214-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dotplot</span>(ego_CC, <span class="at">showCategory=</span><span class="dv">30</span>)</span></code></pre></div>
<pre><code>## wrong orderBy parameter; set to default `orderBy = &quot;x&quot;`</code></pre>
<p><img src="Single_cell_training_files/figure-html/dotplot_GO-1.png" width="672" /></p>
<div class="sourceCode" id="cb216"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb216-1"><a href="富集分析.html#cb216-1" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(ego_CC)</span></code></pre></div>
<p><img src="Single_cell_training_files/figure-html/dotplot_GO-2.png" width="672" /></p>
<div class="sourceCode" id="cb217"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb217-1"><a href="富集分析.html#cb217-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb217-2"><a href="富集分析.html#cb217-2" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(go, <span class="at">split=</span><span class="st">&quot;ONTOLOGY&quot;</span>) <span class="sc">+</span><span class="fu">facet_grid</span>(ONTOLOGY<span class="sc">~</span>., <span class="at">scale=</span><span class="st">&quot;free&quot;</span>)</span></code></pre></div>
<pre><code>## wrong orderBy parameter; set to default `orderBy = &quot;x&quot;`</code></pre>
<div class="sourceCode" id="cb219"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb219-1"><a href="富集分析.html#cb219-1" aria-hidden="true" tabindex="-1"></a>p</span></code></pre></div>
<p><img src="Single_cell_training_files/figure-html/dotplot_GO-3.png" width="672" />
### DAG有向无环图</p>
<div class="sourceCode" id="cb220"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb220-1"><a href="富集分析.html#cb220-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(topGO)</span></code></pre></div>
<pre><code>## Loading required package: graph</code></pre>
<pre><code>## 
## Attaching package: &#39;graph&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:stringr&#39;:
## 
##     boundary</code></pre>
<pre><code>## Loading required package: GO.db</code></pre>
<pre><code>## Loading required package: SparseM</code></pre>
<pre><code>## 
## Attaching package: &#39;SparseM&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     backsolve</code></pre>
<pre><code>## 
## groupGOTerms:    GOBPTerm, GOMFTerm, GOCCTerm environments built.</code></pre>
<pre><code>## 
## Attaching package: &#39;topGO&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:IRanges&#39;:
## 
##     members</code></pre>
<div class="sourceCode" id="cb231"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb231-1"><a href="富集分析.html#cb231-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotGOgraph</span>(ego_MF)  <span class="co">#矩形代表富集到的top10个GO terms, 颜色从黄色过滤到红色，对应p值从大到小。</span></span></code></pre></div>
<pre><code>## 
## groupGOTerms:    GOBPTerm, GOMFTerm, GOCCTerm environments built.</code></pre>
<pre><code>## 
## Building most specific GOs .....</code></pre>
<pre><code>##  ( 541 GO terms found. )</code></pre>
<pre><code>## 
## Build GO DAG topology ..........</code></pre>
<pre><code>##  ( 541 GO terms and 676 relations. )</code></pre>
<pre><code>## 
## Annotating nodes ...............</code></pre>
<pre><code>##  ( 18352 genes annotated to the GO terms. )</code></pre>
<pre><code>## Loading required package: Rgraphviz</code></pre>
<pre><code>## Loading required package: grid</code></pre>
<pre><code>## 
## Attaching package: &#39;grid&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:topGO&#39;:
## 
##     depth</code></pre>
<pre><code>## 
## Attaching package: &#39;Rgraphviz&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:IRanges&#39;:
## 
##     from, to</code></pre>
<pre><code>## The following objects are masked from &#39;package:S4Vectors&#39;:
## 
##     from, to</code></pre>
<p><img src="Single_cell_training_files/figure-html/cnetplot-1.png" width="672" /></p>
<pre><code>## $dag
## A graphNEL graph with directed edges
## Number of Nodes = 23 
## Number of Edges = 23 
## 
## $complete.dag
## [1] &quot;A graph with 23 nodes.&quot;</code></pre>
<div class="sourceCode" id="cb247"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb247-1"><a href="富集分析.html#cb247-1" aria-hidden="true" tabindex="-1"></a><span class="co">#igraph布局的DAG </span></span>
<span id="cb247-2"><a href="富集分析.html#cb247-2" aria-hidden="true" tabindex="-1"></a><span class="co"># goplot(go)</span></span>
<span id="cb247-3"><a href="富集分析.html#cb247-3" aria-hidden="true" tabindex="-1"></a><span class="co">#GO terms关系网络图（通过差异基因关联） </span></span>
<span id="cb247-4"><a href="富集分析.html#cb247-4" aria-hidden="true" tabindex="-1"></a><span class="co"># emapplot(go, showCategory = 30)</span></span>
<span id="cb247-5"><a href="富集分析.html#cb247-5" aria-hidden="true" tabindex="-1"></a><span class="co">#GO term与差异基因关系网络图 </span></span>
<span id="cb247-6"><a href="富集分析.html#cb247-6" aria-hidden="true" tabindex="-1"></a><span class="fu">cnetplot</span>(go, <span class="at">showCategory =</span> <span class="dv">5</span>)</span></code></pre></div>
<p><img src="Single_cell_training_files/figure-html/cnetplot-2.png" width="672" /></p>
</div>
</div>
<div id="kegg富集分析" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> KEGG富集分析</h2>
<div id="kegg富集分析代码" class="section level3" number="5.3.1">
<h3><span class="header-section-number">5.3.1</span> KEGG富集分析代码</h3>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="富集分析.html#cb248-1" aria-hidden="true" tabindex="-1"></a><span class="co">#上调基因富集</span></span>
<span id="cb248-2"><a href="富集分析.html#cb248-2" aria-hidden="true" tabindex="-1"></a>kk.up <span class="ot">&lt;-</span> <span class="fu">enrichKEGG</span>(<span class="at">gene         =</span> gene_up_KEGG, <span class="co">#注意这里只能用 entrzeid</span></span>
<span id="cb248-3"><a href="富集分析.html#cb248-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">organism     =</span> <span class="st">&#39;hsa&#39;</span>,</span>
<span id="cb248-4"><a href="富集分析.html#cb248-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">universe     =</span> gene_all, <span class="do">##背景基因集，可省</span></span>
<span id="cb248-5"><a href="富集分析.html#cb248-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">pvalueCutoff =</span> <span class="fl">0.9</span>, <span class="do">##指定 p 值阈值，不显著的值将不显示在结果中</span></span>
<span id="cb248-6"><a href="富集分析.html#cb248-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">qvalueCutoff =</span> <span class="fl">0.9</span>)</span></code></pre></div>
<pre><code>## Reading KEGG annotation online:
## 
## Reading KEGG annotation online:</code></pre>
<div class="sourceCode" id="cb250"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb250-1"><a href="富集分析.html#cb250-1" aria-hidden="true" tabindex="-1"></a><span class="co">#下调基因富集</span></span>
<span id="cb250-2"><a href="富集分析.html#cb250-2" aria-hidden="true" tabindex="-1"></a>kk.down <span class="ot">&lt;-</span> <span class="fu">enrichKEGG</span>(<span class="at">gene         =</span>  gene_down_KEGG,</span>
<span id="cb250-3"><a href="富集分析.html#cb250-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">organism     =</span> <span class="st">&#39;hsa&#39;</span>,</span>
<span id="cb250-4"><a href="富集分析.html#cb250-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">universe     =</span> gene_all,</span>
<span id="cb250-5"><a href="富集分析.html#cb250-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">pvalueCutoff =</span> <span class="fl">0.9</span>,</span>
<span id="cb250-6"><a href="富集分析.html#cb250-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">qvalueCutoff =</span><span class="fl">0.9</span>)</span>
<span id="cb250-7"><a href="富集分析.html#cb250-7" aria-hidden="true" tabindex="-1"></a>kk.diff <span class="ot">&lt;-</span> <span class="fu">enrichKEGG</span>(<span class="at">gene         =</span> gene_diff_KEGG,</span>
<span id="cb250-8"><a href="富集分析.html#cb250-8" aria-hidden="true" tabindex="-1"></a>                        <span class="at">organism     =</span> <span class="st">&#39;hsa&#39;</span>,</span>
<span id="cb250-9"><a href="富集分析.html#cb250-9" aria-hidden="true" tabindex="-1"></a>                        <span class="at">pvalueCutoff =</span> <span class="fl">0.9</span>)</span>
<span id="cb250-10"><a href="富集分析.html#cb250-10" aria-hidden="true" tabindex="-1"></a><span class="co"># save(kk.diff,kk.down,kk.up,file = &quot;GSE4107kegg.Rdata&quot;)</span></span>
<span id="cb250-11"><a href="富集分析.html#cb250-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb250-12"><a href="富集分析.html#cb250-12" aria-hidden="true" tabindex="-1"></a><span class="co">#从富集结果中提取结果数据框</span></span>
<span id="cb250-13"><a href="富集分析.html#cb250-13" aria-hidden="true" tabindex="-1"></a>ekegg <span class="ot">&lt;-</span> <span class="fu">setReadable</span>(kk.up, <span class="at">OrgDb =</span> org.Hs.eg.db, <span class="at">keyType=</span><span class="st">&quot;ENTREZID&quot;</span>)</span>
<span id="cb250-14"><a href="富集分析.html#cb250-14" aria-hidden="true" tabindex="-1"></a>kegg_diff_dt <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(ekegg)</span>
<span id="cb250-15"><a href="富集分析.html#cb250-15" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(kegg_diff_dt)</span></code></pre></div>
<pre><code>##                ID                       Description GeneRatio BgRatio
## hsa03010 hsa03010                          Ribosome    66/137  84/998
## hsa05171 hsa05171    Coronavirus disease - COVID-19    67/137  95/998
## hsa05340 hsa05340          Primary immunodeficiency     7/137  13/998
## hsa04659 hsa04659         Th17 cell differentiation    12/137  33/998
## hsa04660 hsa04660 T cell receptor signaling pathway    10/137  26/998
## hsa04658 hsa04658  Th1 and Th2 cell differentiation    10/137  27/998
##                pvalue     p.adjust       qvalue
## hsa03010 1.396518e-47 2.206498e-45 2.205028e-45
## hsa05171 4.811804e-43 3.801325e-41 3.798793e-41
## hsa05340 6.645242e-04 2.990295e-02 2.988303e-02
## hsa04659 7.570368e-04 2.990295e-02 2.988303e-02
## hsa04660 1.298228e-03 4.102401e-02 4.099668e-02
## hsa04658 1.822463e-03 4.799151e-02 4.795954e-02
##                                                                                                                                                                                                                                                                                                                                                                                                                 geneID
## hsa03010     RPS18/RPS27/RPS3/RPL13A/RPL3/RPL13/RPS27A/RPS25/RPS12/RPL23A/RPSA/RPLP2/RPS15A/RPS6/RPL31/RPL9/RPL32/RPS29/RPLP0/RPS8/RPL30/RPL11/RPL27A/RPS7/RPS14/RPS10/RPL35A/RPLP1/RPL5/RPL19/RPS3A/RPS20/RPS4X/RPS5/RPL10A/RPL36/RPS28/RPS2/RPS13/RPL10/RPL14/RPL17/RPL15/RPS16/RPL21/RPL6/RPS23/RPS15/RPL7A/RPL18/RPL18A/RPL35/RPS26/RPS21/RPL37A/RPL4/RPL29/RPL27/RPL7/RPL24/RPL37/RPS4Y1/RPL38/RPL22/RPL36A/RPL34
## hsa05171 RPS18/RPS27/RPS3/RPL13A/RPL3/RPL13/RPS27A/RPS25/RPS12/RPL23A/RPSA/RPLP2/RPS15A/RPS6/RPL31/RPL9/RPL32/RPS29/RPLP0/RPS8/RPL30/RPL11/RPL27A/RPS7/RPS14/RPS10/RPL35A/RPLP1/RPL5/RPL19/RPS3A/RPS20/RPS4X/RPS5/RPL10A/RPL36/RPS28/RPS2/JUN/RPS13/RPL10/RPL14/RPL17/RPL15/RPS16/RPL21/RPL6/RPS23/RPS15/RPL7A/RPL18/RPL18A/RPL35/RPS26/RPS21/RPL37A/RPL4/RPL29/RPL27/RPL7/RPL24/RPL37/RPS4Y1/RPL38/RPL22/RPL36A/RPL34
## hsa05340                                                                                                                                                                                                                                                                                                                                                                         CD3D/CD3E/IL7R/LCK/IL2RG/CD40LG/ZAP70
## hsa04659                                                                                                                                                                                                                                                                                                                                                CD3D/CD3E/JUN/LCK/IL2RG/CD3G/LAT/CD247/RORA/ZAP70/GATA3/IL27RA
## hsa04660                                                                                                                                                                                                                                                                                                                                                           CD3D/CD3E/JUN/LCK/CD3G/LAT/CD247/CD40LG/ZAP70/GRAP2
## hsa04658                                                                                                                                                                                                                                                                                                                                                            CD3D/CD3E/JUN/LCK/IL2RG/CD3G/LAT/CD247/ZAP70/GATA3
##          Count
## hsa03010    66
## hsa05171    67
## hsa05340     7
## hsa04659    12
## hsa04660    10
## hsa04658    10</code></pre>
</div>
<div id="kegg结果可视化" class="section level3" number="5.3.2">
<h3><span class="header-section-number">5.3.2</span> KEGG结果可视化</h3>
<div class="sourceCode" id="cb252"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb252-1"><a href="富集分析.html#cb252-1" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">barplot</span>(ekegg, <span class="at">showCategory=</span><span class="dv">10</span>)</span>
<span id="cb252-2"><a href="富集分析.html#cb252-2" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(ekegg, <span class="at">showCategory=</span><span class="dv">10</span>)</span></code></pre></div>
<pre><code>## wrong orderBy parameter; set to default `orderBy = &quot;x&quot;`</code></pre>
<div class="sourceCode" id="cb254"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb254-1"><a href="富集分析.html#cb254-1" aria-hidden="true" tabindex="-1"></a>plotc <span class="ot">=</span> p1<span class="sc">/</span>p2</span>
<span id="cb254-2"><a href="富集分析.html#cb254-2" aria-hidden="true" tabindex="-1"></a>plotc</span></code></pre></div>
<p><img src="Single_cell_training_files/figure-html/barplot_KEGG-1.png" width="672" /></p>
<div class="sourceCode" id="cb255"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb255-1"><a href="富集分析.html#cb255-1" aria-hidden="true" tabindex="-1"></a>up_kegg <span class="ot">&lt;-</span> kk.up<span class="sc">@</span>result <span class="sc">%&gt;%</span></span>
<span id="cb255-2"><a href="富集分析.html#cb255-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(pvalue<span class="sc">&lt;</span><span class="fl">0.01</span>) <span class="sc">%&gt;%</span></span>
<span id="cb255-3"><a href="富集分析.html#cb255-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">group=</span><span class="dv">1</span>)</span>
<span id="cb255-4"><a href="富集分析.html#cb255-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb255-5"><a href="富集分析.html#cb255-5" aria-hidden="true" tabindex="-1"></a>down_kegg <span class="ot">&lt;-</span> kk.down<span class="sc">@</span>result <span class="sc">%&gt;%</span></span>
<span id="cb255-6"><a href="富集分析.html#cb255-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(pvalue<span class="sc">&lt;</span><span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="co">#筛选行</span></span>
<span id="cb255-7"><a href="富集分析.html#cb255-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">group=</span><span class="sc">-</span><span class="dv">1</span>) <span class="co">#新增列</span></span></code></pre></div>
</div>
</div>
<div id="gsea富集分析" class="section level2" number="5.4">
<h2><span class="header-section-number">5.4</span> GSEA富集分析</h2>
<div class="sourceCode" id="cb256"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb256-1"><a href="富集分析.html#cb256-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb256-2"><a href="富集分析.html#cb256-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GSEABase)</span></code></pre></div>
<pre><code>## Loading required package: annotate</code></pre>
<pre><code>## Loading required package: XML</code></pre>
<pre><code>## 
## Attaching package: &#39;XML&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:graph&#39;:
## 
##     addNode</code></pre>
<pre><code>## 
## Attaching package: &#39;annotate&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:Rgraphviz&#39;:
## 
##     toFile</code></pre>
<pre><code>## 
## Attaching package: &#39;GSEABase&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:topGO&#39;:
## 
##     ontology, phenotype</code></pre>
<div class="sourceCode" id="cb265"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb265-1"><a href="富集分析.html#cb265-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb265-2"><a href="富集分析.html#cb265-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb265-3"><a href="富集分析.html#cb265-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb265-4"><a href="富集分析.html#cb265-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-5"><a href="富集分析.html#cb265-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(enrichplot)</span>
<span id="cb265-6"><a href="富集分析.html#cb265-6" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb265-7"><a href="富集分析.html#cb265-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-8"><a href="富集分析.html#cb265-8" aria-hidden="true" tabindex="-1"></a>geneList <span class="ot">=</span> deg1<span class="sc">$</span>avg_log2FC </span>
<span id="cb265-9"><a href="富集分析.html#cb265-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(geneList) <span class="ot">=</span> deg1<span class="sc">$</span>gene</span>
<span id="cb265-10"><a href="富集分析.html#cb265-10" aria-hidden="true" tabindex="-1"></a>geneList <span class="ot">=</span> <span class="fu">sort</span>(geneList,<span class="at">decreasing =</span> T)</span>
<span id="cb265-11"><a href="富集分析.html#cb265-11" aria-hidden="true" tabindex="-1"></a>geneList[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span></code></pre></div>
<pre><code>##     IL32     CD3D      LTB  PTPRCAP     IL7R     LDHB     CD3E      CD2 
## 3.622858 3.271881 2.933899 2.696588 2.682289 2.678308 2.573389 2.346615 
##      LCK      JUN 
## 2.073253 2.013264</code></pre>
<div class="sourceCode" id="cb267"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb267-1"><a href="富集分析.html#cb267-1" aria-hidden="true" tabindex="-1"></a>geneset <span class="ot">&lt;-</span> <span class="fu">read.gmt</span>(<span class="st">&quot;./data/h.all.v7.4.symbols.gmt&quot;</span>)  </span>
<span id="cb267-2"><a href="富集分析.html#cb267-2" aria-hidden="true" tabindex="-1"></a>geneset<span class="sc">$</span>term <span class="ot">=</span> <span class="fu">str_remove</span>(geneset<span class="sc">$</span>term,<span class="st">&quot;HALLMARK_&quot;</span>)</span>
<span id="cb267-3"><a href="富集分析.html#cb267-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(geneset)</span></code></pre></div>
<pre><code>##                      term    gene
## 1 TNFA_SIGNALING_VIA_NFKB    JUNB
## 2 TNFA_SIGNALING_VIA_NFKB   CXCL2
## 3 TNFA_SIGNALING_VIA_NFKB    ATF3
## 4 TNFA_SIGNALING_VIA_NFKB  NFKBIA
## 5 TNFA_SIGNALING_VIA_NFKB TNFAIP3
## 6 TNFA_SIGNALING_VIA_NFKB   PTGS2</code></pre>
<div class="sourceCode" id="cb269"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb269-1"><a href="富集分析.html#cb269-1" aria-hidden="true" tabindex="-1"></a>egmt <span class="ot">&lt;-</span> <span class="fu">GSEA</span>(geneList, <span class="at">TERM2GENE=</span>geneset,<span class="at">verbose=</span>F,<span class="at">pvalueCutoff =</span> <span class="fl">0.5</span>)</span>
<span id="cb269-2"><a href="富集分析.html#cb269-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb269-3"><a href="富集分析.html#cb269-3" aria-hidden="true" tabindex="-1"></a><span class="co">#结果转化</span></span>
<span id="cb269-4"><a href="富集分析.html#cb269-4" aria-hidden="true" tabindex="-1"></a>y<span class="ot">=</span><span class="fu">data.frame</span>(egmt)</span>
<span id="cb269-5"><a href="富集分析.html#cb269-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(y)</span></code></pre></div>
<pre><code>##                                              ID             Description setSize
## COMPLEMENT                           COMPLEMENT              COMPLEMENT      63
## MYC_TARGETS_V1                   MYC_TARGETS_V1          MYC_TARGETS_V1      78
## COAGULATION                         COAGULATION             COAGULATION      18
## ALLOGRAFT_REJECTION         ALLOGRAFT_REJECTION     ALLOGRAFT_REJECTION      74
## XENOBIOTIC_METABOLISM     XENOBIOTIC_METABOLISM   XENOBIOTIC_METABOLISM      27
## CHOLESTEROL_HOMEOSTASIS CHOLESTEROL_HOMEOSTASIS CHOLESTEROL_HOMEOSTASIS      15
##                         enrichmentScore       NES       pvalue     p.adjust
## COMPLEMENT                   -0.5517855 -2.208229 3.762355e-06 0.0001504942
## MYC_TARGETS_V1                0.3890339  2.100255 3.143138e-05 0.0006286276
## COAGULATION                  -0.6891525 -2.091602 3.128641e-04 0.0041715210
## ALLOGRAFT_REJECTION           0.3042984  1.651046 6.952398e-03 0.0695239831
## XENOBIOTIC_METABOLISM        -0.4935005 -1.660739 2.555053e-02 0.2000333389
## CHOLESTEROL_HOMEOSTASIS      -0.5592873 -1.613931 3.328452e-02 0.2000333389
##                              qvalues rank                   leading_edge
## COMPLEMENT              0.0001267320  280 tags=44%, list=16%, signal=39%
## MYC_TARGETS_V1          0.0005293706  672 tags=69%, list=38%, signal=45%
## COAGULATION             0.0035128598  241 tags=56%, list=13%, signal=49%
## ALLOGRAFT_REJECTION     0.0585465121   76  tags=22%, list=4%, signal=22%
## XENOBIOTIC_METABOLISM   0.1684491275  638 tags=70%, list=36%, signal=46%
## CHOLESTEROL_HOMEOSTASIS 0.1684491275  137  tags=40%, list=8%, signal=37%
##                                                                                                                                                                                                                                                                                                                                       core_enrichment
## COMPLEMENT                                                                                                                                                                                ATOX1/GNAI2/PFN1/LTA4H/CTSH/DUSP6/CD55/CTSD/PLEK/CTSB/GCA/C1QA/ANXA5/CTSL/RHOG/PLAUR/CTSC/CASP1/LGALS3/CDA/LYN/TIMP1/CEBPB/S100A9/SERPINA1/CTSS/FCN1/FCER1G
## MYC_TARGETS_V1          RPLP0/NPM1/RPS3/RPS6/RPS5/RPS10/EEF1B2/SRSF7/MYC/RPL22/BUB3/RPL14/RPL34/SNRPD2/RPL18/RPL6/RPS2/CNBP/ODC1/SRM/SRSF2/YWHAQ/PSMD14/CCT2/DDX18/FBL/HNRNPR/SERBP1/HSPE1/HPRT1/HDDC2/APEX1/HNRNPA1/C1QBP/HSPD1/RAN/PRDX3/HNRNPA3/VDAC3/ACP1/DUT/HNRNPC/G3BP1/RSL1D1/LSM2/RANBP1/SRSF3/UBA2/TCP1/PRPF31/HNRNPA2B1/SNRPA/HNRNPU/IFRD1
## COAGULATION                                                                                                                                                                                                                                                                                   LTA4H/CTSH/DUSP6/PLEK/CTSB/RAC1/C1QA/TIMP1/SERPINA1/CFD
## ALLOGRAFT_REJECTION                                                                                                                                                                                                                                                 CD3D/LTB/CD3E/CD2/LCK/CD7/CCL5/TRAT1/CD3G/NPM1/CD247/IL2RG/CD40LG/RPL9/RPS3A/SIT1
## XENOBIOTIC_METABOLISM                                                                                                                                                                                                                        COMT/IRF8/PGD/ID2/GCH1/TNFRSF1A/ALDH2/CBR1/KYNU/BLVRB/FBP1/DDAH2/GSTO1/CAT/TMEM176B/MT2A/NINJ1/HMOX1/CDA
## CHOLESTEROL_HOMEOSTASIS                                                                                                                                                                                                                                                                                        GUSB/ANXA5/CXCL16/PLAUR/LGALS3/S100A11</code></pre>
<div id="gsea结果可视化" class="section level3" number="5.4.1">
<h3><span class="header-section-number">5.4.1</span> GSEA结果可视化</h3>
<div class="sourceCode" id="cb271"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb271-1"><a href="富集分析.html#cb271-1" aria-hidden="true" tabindex="-1"></a><span class="co">#气泡图，展示geneset被激活还是抑制</span></span>
<span id="cb271-2"><a href="富集分析.html#cb271-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dotplot</span>(egmt,<span class="at">split=</span><span class="st">&quot;.sign&quot;</span>)<span class="sc">+</span><span class="fu">facet_grid</span>(<span class="sc">~</span>.sign)</span></code></pre></div>
<pre><code>## wrong orderBy parameter; set to default `orderBy = &quot;x&quot;`</code></pre>
<p><img src="Single_cell_training_files/figure-html/gseaplot2-1.png" width="672" /></p>
<div class="sourceCode" id="cb273"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb273-1"><a href="富集分析.html#cb273-1" aria-hidden="true" tabindex="-1"></a><span class="co">#经典gseaplot</span></span>
<span id="cb273-2"><a href="富集分析.html#cb273-2" aria-hidden="true" tabindex="-1"></a><span class="fu">gseaplot2</span>(egmt, <span class="at">geneSetID =</span> <span class="dv">1</span>, <span class="at">title =</span> egmt<span class="sc">$</span>Description[<span class="dv">1</span>])</span></code></pre></div>
<p><img src="Single_cell_training_files/figure-html/gseaplot2-2.png" width="672" /></p>
</div>
</div>
<div id="gsva富集分析" class="section level2" number="5.5">
<h2><span class="header-section-number">5.5</span> GSVA富集分析</h2>
<p>GSVA(gene set variation analysis)，通过将基因在不同样品间的表达矩阵转化成基因集在样品间的表达矩阵，从而来评估不同的代谢通路在不同样品间是否富集</p>
<p>富集过程主要包含一下四步： 1）评估基因i在样品j中是高表达还是低表达：累积密度函数的核估计，得到每个样本的表达水平统计 2）对每个样本的表达水平统计进行排序和标准化 3）计算每个基因集的类KS随机游走统计量 4）将类KS随机游走统计量转化为ES：最大偏离量（双峰），最大正负偏离量之差（近似正态分布）
### GSVA富集示例</p>
<div class="sourceCode" id="cb274"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb274-1"><a href="富集分析.html#cb274-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GSVA)</span>
<span id="cb274-2"><a href="富集分析.html#cb274-2" aria-hidden="true" tabindex="-1"></a>expr <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pbmc<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>data)</span>
<span id="cb274-3"><a href="富集分析.html#cb274-3" aria-hidden="true" tabindex="-1"></a>expr[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span></code></pre></div>
<pre><code>##               PBMC_AAACATACAACCAC-1 PBMC_AAACATTGAGCTAC-1 PBMC_AAACATTGATCAGC-1
## MIR1302-10                        0                     0                     0
## FAM138A                           0                     0                     0
## OR4F5                             0                     0                     0
## RP11-34P13.7                      0                     0                     0
## RP11-34P13.8                      0                     0                     0
## AL627309.1                        0                     0                     0
## RP11-34P13.14                     0                     0                     0
## RP11-34P13.9                      0                     0                     0
## AP006222.2                        0                     0                     0
## RP4-669L17.10                     0                     0                     0
##               PBMC_AAACCGTGCTTCCG-1 PBMC_AAACCGTGTATGCG-1 PBMC_AAACGCACTGGTAC-1
## MIR1302-10                        0                     0                     0
## FAM138A                           0                     0                     0
## OR4F5                             0                     0                     0
## RP11-34P13.7                      0                     0                     0
## RP11-34P13.8                      0                     0                     0
## AL627309.1                        0                     0                     0
## RP11-34P13.14                     0                     0                     0
## RP11-34P13.9                      0                     0                     0
## AP006222.2                        0                     0                     0
## RP4-669L17.10                     0                     0                     0
##               PBMC_AAACGCTGACCAGT-1 PBMC_AAACGCTGGTTCTT-1 PBMC_AAACGCTGTAGCCA-1
## MIR1302-10                        0                     0                     0
## FAM138A                           0                     0                     0
## OR4F5                             0                     0                     0
## RP11-34P13.7                      0                     0                     0
## RP11-34P13.8                      0                     0                     0
## AL627309.1                        0                     0                     0
## RP11-34P13.14                     0                     0                     0
## RP11-34P13.9                      0                     0                     0
## AP006222.2                        0                     0                     0
## RP4-669L17.10                     0                     0                     0
##               PBMC_AAACGCTGTTTCTG-1
## MIR1302-10                        0
## FAM138A                           0
## OR4F5                             0
## RP11-34P13.7                      0
## RP11-34P13.8                      0
## AL627309.1                        0
## RP11-34P13.14                     0
## RP11-34P13.9                      0
## AP006222.2                        0
## RP4-669L17.10                     0</code></pre>
<div class="sourceCode" id="cb276"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb276-1"><a href="富集分析.html#cb276-1" aria-hidden="true" tabindex="-1"></a>expr<span class="ot">=</span><span class="fu">as.matrix</span>(expr)</span>
<span id="cb276-2"><a href="富集分析.html#cb276-2" aria-hidden="true" tabindex="-1"></a>kegg_list <span class="ot">=</span> <span class="fu">split</span>(geneset<span class="sc">$</span>gene, geneset<span class="sc">$</span>term)</span>
<span id="cb276-3"><a href="富集分析.html#cb276-3" aria-hidden="true" tabindex="-1"></a>kegg_list[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span></code></pre></div>
<pre><code>## $ADIPOGENESIS
##   [1] &quot;FABP4&quot;    &quot;ADIPOQ&quot;   &quot;PPARG&quot;    &quot;LIPE&quot;     &quot;DGAT1&quot;    &quot;LPL&quot;     
##   [7] &quot;CPT2&quot;     &quot;CD36&quot;     &quot;GPAM&quot;     &quot;ADIPOR2&quot;  &quot;ACAA2&quot;    &quot;ETFB&quot;    
##  [13] &quot;ACOX1&quot;    &quot;ACADM&quot;    &quot;HADH&quot;     &quot;IDH1&quot;     &quot;SORBS1&quot;   &quot;ACADS&quot;   
##  [19] &quot;UCK1&quot;     &quot;SCP2&quot;     &quot;DECR1&quot;    &quot;CDKN2C&quot;   &quot;TALDO1&quot;   &quot;TST&quot;     
##  [25] &quot;MCCC1&quot;    &quot;PGM1&quot;     &quot;REEP5&quot;    &quot;BCL2L13&quot;  &quot;SLC25A10&quot; &quot;ME1&quot;     
##  [31] &quot;PHYH&quot;     &quot;PIM3&quot;     &quot;YWHAG&quot;    &quot;NDUFAB1&quot;  &quot;GPD2&quot;     &quot;ADIG&quot;    
##  [37] &quot;ECHS1&quot;    &quot;QDPR&quot;     &quot;CS&quot;       &quot;ECH1&quot;     &quot;SLC25A1&quot;  &quot;ACADL&quot;   
##  [43] &quot;TOB1&quot;     &quot;GRPEL1&quot;   &quot;CRAT&quot;     &quot;GBE1&quot;     &quot;CAVIN2&quot;   &quot;SCARB1&quot;  
##  [49] &quot;PEMT&quot;     &quot;CHCHD10&quot;  &quot;AK2&quot;      &quot;APOE&quot;     &quot;UQCR10&quot;   &quot;TANK&quot;    
##  [55] &quot;ANGPTL4&quot;  &quot;ACO2&quot;     &quot;FAH&quot;      &quot;ACLY&quot;     &quot;IFNGR1&quot;   &quot;SLC5A6&quot;  
##  [61] &quot;JAGN1&quot;    &quot;EPHX2&quot;    &quot;IDH3G&quot;    &quot;GPX3&quot;     &quot;ELMOD3&quot;   &quot;ORM1&quot;    
##  [67] &quot;RETSAT&quot;   &quot;ESRRA&quot;    &quot;HIBCH&quot;    &quot;SUCLG1&quot;   &quot;STAT5A&quot;   &quot;ITGA7&quot;   
##  [73] &quot;MRAP&quot;     &quot;PLIN2&quot;    &quot;CYC1&quot;     &quot;ALDH2&quot;    &quot;RNF11&quot;    &quot;ALDOA&quot;   
##  [79] &quot;SULT1A1&quot;  &quot;DDT&quot;      &quot;SDHB&quot;     &quot;CD151&quot;    &quot;SLC27A1&quot;  &quot;BCKDHA&quot;  
##  [85] &quot;C3&quot;       &quot;LEP&quot;      &quot;ADCY6&quot;    &quot;ELOVL6&quot;   &quot;LTC4S&quot;    &quot;SPARCL1&quot; 
##  [91] &quot;RMDN3&quot;    &quot;MTCH2&quot;    &quot;SOWAHC&quot;   &quot;SLC1A5&quot;   &quot;CMPK1&quot;    &quot;REEP6&quot;   
##  [97] &quot;NDUFA5&quot;   &quot;FZD4&quot;     &quot;DRAM2&quot;    &quot;MGST3&quot;    &quot;ATP1B3&quot;   &quot;RETN&quot;    
## [103] &quot;STOM&quot;     &quot;ESYT1&quot;    &quot;GHITM&quot;    &quot;DNAJC15&quot;  &quot;GADD45A&quot;  &quot;VEGFB&quot;   
## [109] &quot;PFKL&quot;     &quot;COQ3&quot;     &quot;NABP1&quot;    &quot;CYP4B1&quot;   &quot;PPM1B&quot;    &quot;ARAF&quot;    
## [115] &quot;CAVIN1&quot;   &quot;COL4A1&quot;   &quot;IMMT&quot;     &quot;DHRS7&quot;    &quot;COL15A1&quot;  &quot;NMT1&quot;    
## [121] &quot;COQ5&quot;     &quot;LAMA4&quot;    &quot;AGPAT3&quot;   &quot;BAZ2A&quot;    &quot;IDH3A&quot;    &quot;LIFR&quot;    
## [127] &quot;PREB&quot;     &quot;PTGER3&quot;   &quot;GPHN&quot;     &quot;PFKFB3&quot;   &quot;GPX4&quot;     &quot;SSPN&quot;    
## [133] &quot;SQOR&quot;     &quot;MTARC2&quot;   &quot;DLD&quot;      &quot;ITIH5&quot;    &quot;CD302&quot;    &quot;ATL2&quot;    
## [139] &quot;GPAT4&quot;    &quot;LPCAT3&quot;   &quot;TKT&quot;      &quot;UQCRC1&quot;   &quot;CAT&quot;      &quot;OMD&quot;     
## [145] &quot;DLAT&quot;     &quot;MRPL15&quot;   &quot;RIOK3&quot;    &quot;RTN3&quot;     &quot;CHUK&quot;     &quot;G3BP2&quot;   
## [151] &quot;SDHC&quot;     &quot;SAMM50&quot;   &quot;ARL4A&quot;    &quot;SNCG&quot;     &quot;PDCD4&quot;    &quot;COQ9&quot;    
## [157] &quot;APLP2&quot;    &quot;SOD1&quot;     &quot;PTCD3&quot;    &quot;PHLDB1&quot;   &quot;ENPP2&quot;    &quot;HSPB8&quot;   
## [163] &quot;AIFM1&quot;    &quot;CCNG2&quot;    &quot;PPP1R15B&quot; &quot;MDH2&quot;     &quot;ABCA1&quot;    &quot;COX7B&quot;   
## [169] &quot;MYLK&quot;     &quot;COX8A&quot;    &quot;DHRS7B&quot;   &quot;MIGA2&quot;    &quot;MGLL&quot;     &quot;ITSN1&quot;   
## [175] &quot;DHCR7&quot;    &quot;RREB1&quot;    &quot;CMBL&quot;     &quot;UBC&quot;      &quot;ATP5PO&quot;   &quot;PRDX3&quot;   
## [181] &quot;DBT&quot;      &quot;NDUFS3&quot;   &quot;NKIRAS1&quot;  &quot;RAB34&quot;    &quot;CIDEA&quot;    &quot;UQCRQ&quot;   
## [187] &quot;PEX14&quot;    &quot;BCL6&quot;     &quot;COX6A1&quot;   &quot;DNAJB9&quot;   &quot;MAP4K3&quot;   &quot;ANGPT1&quot;  
## [193] &quot;UBQLN1&quot;   &quot;NDUFB7&quot;   &quot;SLC19A1&quot;  &quot;ABCB8&quot;    &quot;SLC66A3&quot;  &quot;POR&quot;     
## [199] &quot;UCP2&quot;     &quot;UQCR11&quot;  
## 
## $ALLOGRAFT_REJECTION
##   [1] &quot;PTPRC&quot;    &quot;IL12B&quot;    &quot;TGFB1&quot;    &quot;IL12A&quot;    &quot;CD3E&quot;     &quot;CD3D&quot;    
##   [7] &quot;CD28&quot;     &quot;LYN&quot;      &quot;HCLS1&quot;    &quot;IL18&quot;     &quot;CRTAM&quot;    &quot;IFNG&quot;    
##  [13] &quot;CD3G&quot;     &quot;CD86&quot;     &quot;IL10&quot;     &quot;UBE2N&quot;    &quot;BCL10&quot;    &quot;CD4&quot;     
##  [19] &quot;LCK&quot;      &quot;NCK1&quot;     &quot;C2&quot;       &quot;HLA-A&quot;    &quot;ITGB2&quot;    &quot;HLA-DQA1&quot;
##  [25] &quot;CD1D&quot;     &quot;CD80&quot;     &quot;HLA-DRA&quot;  &quot;THY1&quot;     &quot;TLR1&quot;     &quot;HLA-G&quot;   
##  [31] &quot;HLA-DMB&quot;  &quot;IL7&quot;      &quot;IL4&quot;      &quot;TNF&quot;      &quot;CD247&quot;    &quot;IL2&quot;     
##  [37] &quot;HLA-DMA&quot;  &quot;STAT1&quot;    &quot;IRF4&quot;     &quot;SRGN&quot;     &quot;INHBA&quot;    &quot;TLR3&quot;    
##  [43] &quot;ZAP70&quot;    &quot;CD74&quot;     &quot;CD40&quot;     &quot;TRAF2&quot;    &quot;B2M&quot;      &quot;BCL3&quot;    
##  [49] &quot;LTB&quot;      &quot;IFNGR1&quot;   &quot;CCR5&quot;     &quot;CD40LG&quot;   &quot;HLA-DOA&quot;  &quot;GLMN&quot;    
##  [55] &quot;IL6&quot;      &quot;HLA-E&quot;    &quot;CD2&quot;      &quot;CCL5&quot;     &quot;FAS&quot;      &quot;FASLG&quot;   
##  [61] &quot;TLR6&quot;     &quot;PF4&quot;      &quot;TGFB2&quot;    &quot;CD79A&quot;    &quot;INHBB&quot;    &quot;ELANE&quot;   
##  [67] &quot;SPI1&quot;     &quot;MAP3K7&quot;   &quot;IL15&quot;     &quot;CTSS&quot;     &quot;CD47&quot;     &quot;PRF1&quot;    
##  [73] &quot;IL12RB1&quot;  &quot;LCP2&quot;     &quot;SOCS1&quot;    &quot;CDKN2A&quot;   &quot;STAT4&quot;    &quot;CD7&quot;     
##  [79] &quot;HLA-DOB&quot;  &quot;CD8A&quot;     &quot;ICAM1&quot;    &quot;CCL4&quot;     &quot;GZMB&quot;     &quot;CSF1&quot;    
##  [85] &quot;IL11&quot;     &quot;STAB1&quot;    &quot;IL2RA&quot;    &quot;NLRP3&quot;    &quot;CCND3&quot;    &quot;EIF3A&quot;   
##  [91] &quot;SIT1&quot;     &quot;IFNAR2&quot;   &quot;HDAC9&quot;    &quot;CARTPT&quot;   &quot;TRAT1&quot;    &quot;CCL22&quot;   
##  [97] &quot;APBB1&quot;    &quot;FYB1&quot;     &quot;IL1B&quot;     &quot;TIMP1&quot;    &quot;RPS19&quot;    &quot;JAK2&quot;    
## [103] &quot;KRT1&quot;     &quot;WARS1&quot;    &quot;IFNGR2&quot;   &quot;CCR2&quot;     &quot;EREG&quot;     &quot;MMP9&quot;    
## [109] &quot;EGFR&quot;     &quot;IL16&quot;     &quot;CFP&quot;      &quot;WAS&quot;      &quot;ITGAL&quot;    &quot;KLRD1&quot;   
## [115] &quot;RARS1&quot;    &quot;TLR2&quot;     &quot;CCND2&quot;    &quot;IL2RG&quot;    &quot;ETS1&quot;     &quot;ITK&quot;     
## [121] &quot;NCR1&quot;     &quot;MAP4K1&quot;   &quot;CCL19&quot;    &quot;PSMB10&quot;   &quot;RPL39&quot;    &quot;EIF3J&quot;   
## [127] &quot;ABCE1&quot;    &quot;CD8B&quot;     &quot;F2&quot;       &quot;ELF4&quot;     &quot;LY86&quot;     &quot;FCGR2B&quot;  
## [133] &quot;GBP2&quot;     &quot;PRKCG&quot;    &quot;RPS9&quot;     &quot;MTIF2&quot;    &quot;GZMA&quot;     &quot;AARS1&quot;   
## [139] &quot;CD96&quot;     &quot;CSK&quot;      &quot;HIF1A&quot;    &quot;CCL2&quot;     &quot;ICOSLG&quot;   &quot;NPM1&quot;    
## [145] &quot;IL4R&quot;     &quot;CCL11&quot;    &quot;NME1&quot;     &quot;FLNA&quot;     &quot;GPR65&quot;    &quot;ACHE&quot;    
## [151] &quot;EIF3D&quot;    &quot;IGSF6&quot;    &quot;F2R&quot;      &quot;IL13&quot;     &quot;TAP1&quot;     &quot;DARS1&quot;   
## [157] &quot;IRF7&quot;     &quot;ACVR2A&quot;   &quot;CXCR3&quot;    &quot;PRKCB&quot;    &quot;CXCL9&quot;    &quot;PTPN6&quot;   
## [163] &quot;NCF4&quot;     &quot;UBE2D1&quot;   &quot;LIF&quot;      &quot;CCR1&quot;     &quot;MBL2&quot;     &quot;DEGS1&quot;   
## [169] &quot;TPD52&quot;    &quot;AKT1&quot;     &quot;RIPK2&quot;    &quot;IKBKB&quot;    &quot;GCNT1&quot;    &quot;SOCS5&quot;   
## [175] &quot;IRF8&quot;     &quot;TAP2&quot;     &quot;EIF4G3&quot;   &quot;ABI1&quot;     &quot;CCL7&quot;     &quot;IL2RB&quot;   
## [181] &quot;BRCA1&quot;    &quot;FGR&quot;      &quot;IL18RAP&quot;  &quot;MRPL3&quot;    &quot;CXCL13&quot;   &quot;CAPG&quot;    
## [187] &quot;EIF5A&quot;    &quot;RPS3A&quot;    &quot;GALNT1&quot;   &quot;ST8SIA4&quot;  &quot;CCL13&quot;    &quot;RPL3L&quot;   
## [193] &quot;LY75&quot;     &quot;TAPBP&quot;    &quot;NOS2&quot;     &quot;RPL9&quot;     &quot;BCAT1&quot;    &quot;IL9&quot;     
## [199] &quot;IL27RA&quot;   &quot;DYRK3&quot;</code></pre>
<div class="sourceCode" id="cb278"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb278-1"><a href="富集分析.html#cb278-1" aria-hidden="true" tabindex="-1"></a>kegg1 <span class="ot">&lt;-</span> <span class="fu">gsva</span>(expr, kegg_list, <span class="at">kcdf=</span><span class="st">&quot;Gaussian&quot;</span>,<span class="at">method =</span> <span class="st">&quot;gsva&quot;</span>,<span class="at">parallel.sz=</span><span class="dv">12</span>)</span></code></pre></div>
<pre><code>## Warning in .filterFeatures(expr, method): 16159 genes with constant expression
## values throuhgout the samples.</code></pre>
<pre><code>## Warning in .filterFeatures(expr, method): Since argument method!=&quot;ssgsea&quot;, genes
## with constant expression values are discarded.</code></pre>
<pre><code>## Setting parallel calculations through a MulticoreParam back-end
## with workers=12 and tasks=100.
## Estimating GSVA scores for 50 gene sets.
## Estimating ECDFs with Gaussian kernels
## Estimating ECDFs in parallel</code></pre>
<pre><code>## iteration: 123456789101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899100</code></pre>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |=                                                                     |   2%
  |                                                                            
  |===                                                                   |   4%
  |                                                                            
  |====                                                                  |   6%
  |                                                                            
  |======                                                                |   8%
  |                                                                            
  |=======                                                               |  10%
  |                                                                            
  |========                                                              |  12%
  |                                                                            
  |==========                                                            |  14%
  |                                                                            
  |===========                                                           |  16%
  |                                                                            
  |=============                                                         |  18%
  |                                                                            
  |==============                                                        |  20%
  |                                                                            
  |===============                                                       |  22%
  |                                                                            
  |=================                                                     |  24%
  |                                                                            
  |==================                                                    |  26%
  |                                                                            
  |====================                                                  |  28%
  |                                                                            
  |=====================                                                 |  30%
  |                                                                            
  |======================                                                |  32%
  |                                                                            
  |========================                                              |  34%
  |                                                                            
  |=========================                                             |  36%
  |                                                                            
  |===========================                                           |  38%
  |                                                                            
  |============================                                          |  40%
  |                                                                            
  |=============================                                         |  42%
  |                                                                            
  |===============================                                       |  44%
  |                                                                            
  |================================                                      |  46%
  |                                                                            
  |==================================                                    |  48%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |====================================                                  |  52%
  |                                                                            
  |======================================                                |  54%
  |                                                                            
  |=======================================                               |  56%
  |                                                                            
  |=========================================                             |  58%
  |                                                                            
  |==========================================                            |  60%
  |                                                                            
  |===========================================                           |  62%
  |                                                                            
  |=============================================                         |  64%
  |                                                                            
  |==============================================                        |  66%
  |                                                                            
  |================================================                      |  68%
  |                                                                            
  |=================================================                     |  70%
  |                                                                            
  |==================================================                    |  72%
  |                                                                            
  |====================================================                  |  74%
  |                                                                            
  |=====================================================                 |  76%
  |                                                                            
  |=======================================================               |  78%
  |                                                                            
  |========================================================              |  80%
  |                                                                            
  |=========================================================             |  82%
  |                                                                            
  |===========================================================           |  84%
  |                                                                            
  |============================================================          |  86%
  |                                                                            
  |==============================================================        |  88%
  |                                                                            
  |===============================================================       |  90%
  |                                                                            
  |================================================================      |  92%
  |                                                                            
  |==================================================================    |  94%
  |                                                                            
  |===================================================================   |  96%
  |                                                                            
  |===================================================================== |  98%
  |                                                                            
  |======================================================================| 100%</code></pre>
<div class="sourceCode" id="cb284"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb284-1"><a href="富集分析.html#cb284-1" aria-hidden="true" tabindex="-1"></a>kegg2 <span class="ot">&lt;-</span> kegg1</span>
<span id="cb284-2"><a href="富集分析.html#cb284-2" aria-hidden="true" tabindex="-1"></a>kegg2[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span></code></pre></div>
<pre><code>##                         PBMC_AAACATACAACCAC-1 PBMC_AAACATTGAGCTAC-1
## ADIPOGENESIS                      -0.35791459          -0.217004195
## ALLOGRAFT_REJECTION               -0.23093992          -0.056665501
## ANDROGEN_RESPONSE                 -0.24698071          -0.166052456
## ANGIOGENESIS                      -0.05551034          -0.051797361
## APICAL_JUNCTION                   -0.05746997           0.095356739
## APICAL_SURFACE                    -0.10568123          -0.243974970
## APOPTOSIS                         -0.28076796          -0.428613559
## BILE_ACID_METABOLISM              -0.01323618          -0.004775889
## CHOLESTEROL_HOMEOSTASIS           -0.29203360          -0.337901574
## COAGULATION                       -0.11210877          -0.052089648
##                         PBMC_AAACATTGATCAGC-1 PBMC_AAACCGTGCTTCCG-1
## ADIPOGENESIS                      -0.19974647           -0.20068439
## ALLOGRAFT_REJECTION               -0.20360929           -0.19464576
## ANDROGEN_RESPONSE                 -0.06225333           -0.24685381
## ANGIOGENESIS                      -0.10130833            0.27363561
## APICAL_JUNCTION                    0.02150503            0.04306834
## APICAL_SURFACE                    -0.06189907           -0.34868601
## APOPTOSIS                         -0.09531318           -0.18871131
## BILE_ACID_METABOLISM               0.09536061           -0.05813957
## CHOLESTEROL_HOMEOSTASIS           -0.23696223           -0.13187924
## COAGULATION                       -0.09289654           -0.03721555
##                         PBMC_AAACCGTGTATGCG-1 PBMC_AAACGCACTGGTAC-1
## ADIPOGENESIS                      -0.41736608           -0.23090947
## ALLOGRAFT_REJECTION               -0.24839110           -0.30312300
## ANDROGEN_RESPONSE                 -0.18624406           -0.11226897
## ANGIOGENESIS                       0.23986521            0.11545168
## APICAL_JUNCTION                    0.05554962            0.06432303
## APICAL_SURFACE                     0.06105414           -0.09894962
## APOPTOSIS                         -0.42240807           -0.32952326
## BILE_ACID_METABOLISM              -0.16223440            0.04829511
## CHOLESTEROL_HOMEOSTASIS           -0.28216451           -0.19718791
## COAGULATION                        0.05074903           -0.02322263
##                         PBMC_AAACGCTGACCAGT-1 PBMC_AAACGCTGGTTCTT-1
## ADIPOGENESIS                      -0.26101663           -0.31655487
## ALLOGRAFT_REJECTION               -0.20920469           -0.14525146
## ANDROGEN_RESPONSE                 -0.19701361           -0.13466390
## ANGIOGENESIS                       0.03184648            0.03183292
## APICAL_JUNCTION                   -0.01886515           -0.10688643
## APICAL_SURFACE                    -0.21492544           -0.21274795
## APOPTOSIS                         -0.40947077           -0.23442168
## BILE_ACID_METABOLISM              -0.10375464           -0.05968525
## CHOLESTEROL_HOMEOSTASIS           -0.37783520           -0.24226727
## COAGULATION                        0.01418585            0.05508910
##                         PBMC_AAACGCTGTAGCCA-1 PBMC_AAACGCTGTTTCTG-1
## ADIPOGENESIS                      -0.34097446          -0.282262139
## ALLOGRAFT_REJECTION               -0.38120606          -0.365907151
## ANDROGEN_RESPONSE                 -0.26639078          -0.237912755
## ANGIOGENESIS                       0.15355799           0.324644202
## APICAL_JUNCTION                    0.02317564          -0.004812126
## APICAL_SURFACE                    -0.16543791          -0.292213844
## APOPTOSIS                         -0.35512722          -0.308162836
## BILE_ACID_METABOLISM              -0.01218455          -0.092504836
## CHOLESTEROL_HOMEOSTASIS           -0.32480210          -0.201390143
## COAGULATION                        0.06547464           0.153338323</code></pre>
<div id="gsva结果可视化" class="section level3" number="5.5.1">
<h3><span class="header-section-number">5.5.1</span> GSVA结果可视化</h3>
<div class="sourceCode" id="cb286"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb286-1"><a href="富集分析.html#cb286-1" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pbmc<span class="sc">@</span>meta.data[,<span class="fu">c</span>(<span class="st">&#39;orig.ident&#39;</span>,<span class="st">&quot;celltype&quot;</span>)])</span>
<span id="cb286-2"><a href="富集分析.html#cb286-2" aria-hidden="true" tabindex="-1"></a><span class="co">#细胞按照细胞类型排序</span></span>
<span id="cb286-3"><a href="富集分析.html#cb286-3" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> meta <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(meta<span class="sc">$</span>celltype)</span>
<span id="cb286-4"><a href="富集分析.html#cb286-4" aria-hidden="true" tabindex="-1"></a>kegg2 <span class="ot">&lt;-</span> kegg2[,<span class="fu">rownames</span>(meta)]</span>
<span id="cb286-5"><a href="富集分析.html#cb286-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb286-6"><a href="富集分析.html#cb286-6" aria-hidden="true" tabindex="-1"></a><span class="co">#取各细胞类型对应的通路score的均值</span></span>
<span id="cb286-7"><a href="富集分析.html#cb286-7" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">colnames</span>(kegg2),<span class="fu">rownames</span>(meta))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb288"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb288-1"><a href="富集分析.html#cb288-1" aria-hidden="true" tabindex="-1"></a>kegg3 <span class="ot">&lt;-</span> <span class="fu">cbind</span>(meta, <span class="fu">t</span>(kegg2)) <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>()</span>
<span id="cb288-2"><a href="富集分析.html#cb288-2" aria-hidden="true" tabindex="-1"></a>kegg4 <span class="ot">&lt;-</span> tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(kegg3, <span class="at">cols=</span><span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(kegg3), <span class="st">&quot;kegg&quot;</span>, <span class="st">&quot;value&quot;</span>)</span>
<span id="cb288-3"><a href="富集分析.html#cb288-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb288-4"><a href="富集分析.html#cb288-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb288-5"><a href="富集分析.html#cb288-5" aria-hidden="true" tabindex="-1"></a>kegg5 <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">group_by</span>(kegg4, celltype, kegg) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">value2=</span><span class="fu">mean</span>(value) ) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">ungroup</span>()</span></code></pre></div>
<pre><code>## `summarise()` has grouped output by &#39;celltype&#39;. You can override using the `.groups` argument.</code></pre>
<div class="sourceCode" id="cb290"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb290-1"><a href="富集分析.html#cb290-1" aria-hidden="true" tabindex="-1"></a>kegg6 <span class="ot">&lt;-</span> tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(kegg5, <span class="at">names_from=</span><span class="st">&quot;kegg&quot;</span>, <span class="at">values_from=</span><span class="st">&quot;value2&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb290-2"><a href="富集分析.html#cb290-2" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">&quot;celltype&quot;</span>)</span>
<span id="cb290-3"><a href="富集分析.html#cb290-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb290-4"><a href="富集分析.html#cb290-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb290-5"><a href="富集分析.html#cb290-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb290-6"><a href="富集分析.html#cb290-6" aria-hidden="true" tabindex="-1"></a>G1 <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(kegg6,</span>
<span id="cb290-7"><a href="富集分析.html#cb290-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">cluster_rows =</span> F,</span>
<span id="cb290-8"><a href="富集分析.html#cb290-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">cluster_cols =</span> F,</span>
<span id="cb290-9"><a href="富集分析.html#cb290-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">show_rownames =</span> T,</span>
<span id="cb290-10"><a href="富集分析.html#cb290-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">show_colnames =</span> T,</span>
<span id="cb290-11"><a href="富集分析.html#cb290-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span><span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;white&quot;</span>,<span class="st">&quot;red&quot;</span>))(<span class="dv">100</span>),</span>
<span id="cb290-12"><a href="富集分析.html#cb290-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">cellwidth =</span> <span class="dv">10</span>, <span class="at">cellheight =</span> <span class="dv">15</span>,</span>
<span id="cb290-13"><a href="富集分析.html#cb290-13" aria-hidden="true" tabindex="-1"></a>                <span class="at">fontsize =</span> <span class="dv">10</span>)</span></code></pre></div>
<p><img src="Single_cell_training_files/figure-html/GSVA_vis-1.png" width="672" /></p>
</div>
</div>
<div id="sessioninfo-3" class="section level2" number="5.6">
<h2><span class="header-section-number">5.6</span> sessionInfo</h2>
<div class="sourceCode" id="cb291"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb291-1"><a href="富集分析.html#cb291-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.4 (2021-02-15)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.10
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
##  [1] grid      parallel  stats4    stats     graphics  grDevices utils    
##  [8] datasets  methods   base     
## 
## other attached packages:
##  [1] pheatmap_1.0.12         GSVA_1.38.2             enrichplot_1.10.2      
##  [4] GSEABase_1.52.1         annotate_1.68.0         XML_3.99-0.6           
##  [7] Rgraphviz_2.34.0        topGO_2.42.0            SparseM_1.81           
## [10] GO.db_3.12.1            graph_1.68.0            org.Hs.eg.db_3.12.0    
## [13] AnnotationDbi_1.52.0    IRanges_2.24.1          S4Vectors_0.28.1       
## [16] Biobase_2.50.0          BiocGenerics_0.36.1     clusterProfiler_3.18.1 
## [19] forcats_0.5.1           stringr_1.4.0           dplyr_1.0.5            
## [22] purrr_0.3.4             readr_1.4.0             tidyr_1.1.3            
## [25] tibble_3.1.1            ggplot2_3.3.3           tidyverse_1.3.1        
## [28] pbmc3k.SeuratData_3.1.4 SeuratData_0.2.1        SeuratObject_4.0.0     
## [31] Seurat_4.0.1           
## 
## loaded via a namespace (and not attached):
##   [1] utf8_1.2.1                  reticulate_1.18            
##   [3] tidyselect_1.1.0            RSQLite_2.2.5              
##   [5] htmlwidgets_1.5.3           BiocParallel_1.24.1        
##   [7] Rtsne_0.15                  scatterpie_0.1.5           
##   [9] munsell_0.5.0               codetools_0.2-18           
##  [11] ica_1.0-2                   future_1.21.0              
##  [13] miniUI_0.1.1.1              withr_2.4.2                
##  [15] colorspace_2.0-0            GOSemSim_2.16.1            
##  [17] highr_0.9                   knitr_1.32                 
##  [19] rstudioapi_0.13             ROCR_1.0-11                
##  [21] tensor_1.5                  DOSE_3.16.0                
##  [23] listenv_0.8.0               MatrixGenerics_1.2.1       
##  [25] labeling_0.4.2              GenomeInfoDbData_1.2.4     
##  [27] polyclip_1.10-0             bit64_4.0.5                
##  [29] farver_2.1.0                downloader_0.4             
##  [31] parallelly_1.24.0           vctrs_0.3.7                
##  [33] generics_0.1.0              xfun_0.22                  
##  [35] GenomeInfoDb_1.26.7         R6_2.5.0                   
##  [37] graphlayouts_0.7.1          hdf5r_1.3.3                
##  [39] DelayedArray_0.16.3         bitops_1.0-6               
##  [41] spatstat.utils_2.1-0        cachem_1.0.4               
##  [43] fgsea_1.16.0                assertthat_0.2.1           
##  [45] promises_1.2.0.1            scales_1.1.1               
##  [47] ggraph_2.0.5                gtable_0.3.0               
##  [49] globals_0.14.0              goftest_1.2-2              
##  [51] tidygraph_1.2.0             rlang_0.4.10               
##  [53] splines_4.0.4               lazyeval_0.2.2             
##  [55] spatstat.geom_2.1-0         broom_0.7.6                
##  [57] BiocManager_1.30.12         yaml_2.2.1                 
##  [59] reshape2_1.4.4              abind_1.4-5                
##  [61] modelr_0.1.8                backports_1.2.1            
##  [63] httpuv_1.5.5                qvalue_2.22.0              
##  [65] tools_4.0.4                 bookdown_0.21              
##  [67] ellipsis_0.3.1              spatstat.core_2.1-2        
##  [69] jquerylib_0.1.3             RColorBrewer_1.1-2         
##  [71] ggridges_0.5.3              Rcpp_1.0.6                 
##  [73] plyr_1.8.6                  zlibbioc_1.36.0            
##  [75] RCurl_1.98-1.3              rpart_4.1-15               
##  [77] deldir_0.2-10               pbapply_1.4-3              
##  [79] viridis_0.6.0               cowplot_1.1.1              
##  [81] zoo_1.8-9                   SummarizedExperiment_1.20.0
##  [83] haven_2.4.0                 ggrepel_0.9.1              
##  [85] cluster_2.1.2               fs_1.5.0                   
##  [87] magrittr_2.0.1              data.table_1.14.0          
##  [89] RSpectra_0.16-0             scattermore_0.7            
##  [91] DO.db_2.9                   lmtest_0.9-38              
##  [93] reprex_2.0.0                RANN_2.6.1                 
##  [95] ggnewscale_0.4.5            fitdistrplus_1.1-3         
##  [97] matrixStats_0.58.0          hms_1.0.0                  
##  [99] patchwork_1.1.1             mime_0.10                  
## [101] evaluate_0.14               xtable_1.8-4               
## [103] readxl_1.3.1                gridExtra_2.3              
## [105] compiler_4.0.4              shadowtext_0.0.7           
## [107] KernSmooth_2.23-18          crayon_1.4.1               
## [109] htmltools_0.5.1.1           mgcv_1.8-35                
## [111] later_1.1.0.1               lubridate_1.7.10           
## [113] DBI_1.1.1                   tweenr_1.0.2               
## [115] dbplyr_2.1.1                MASS_7.3-53.1              
## [117] rappdirs_0.3.3              Matrix_1.3-2               
## [119] cli_2.4.0                   igraph_1.2.6               
## [121] GenomicRanges_1.42.0        pkgconfig_2.0.3            
## [123] rvcheck_0.1.8               plotly_4.9.3               
## [125] spatstat.sparse_2.0-0       xml2_1.3.2                 
## [127] bslib_0.2.4                 XVector_0.30.0             
## [129] rvest_1.0.0                 digest_0.6.27              
## [131] sctransform_0.3.2           RcppAnnoy_0.0.18           
## [133] spatstat.data_2.1-0         rmarkdown_2.7              
## [135] cellranger_1.1.0            leiden_0.3.7               
## [137] fastmatch_1.1-0             uwot_0.1.10                
## [139] shiny_1.6.0                 lifecycle_1.0.0            
## [141] nlme_3.1-152                jsonlite_1.7.2             
## [143] viridisLite_0.4.0           limma_3.46.0               
## [145] fansi_0.4.2                 pillar_1.6.0               
## [147] lattice_0.20-41             fastmap_1.1.0              
## [149] httr_1.4.2                  survival_3.2-10            
## [151] glue_1.4.2                  png_0.1-7                  
## [153] bit_4.0.4                   ggforce_0.3.3              
## [155] stringi_1.5.3               sass_0.3.1                 
## [157] blob_1.2.1                  memoise_2.0.0              
## [159] irlba_2.3.3                 future.apply_1.7.0</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="pipeline.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Single_cell_training.pdf", "Single_cell_training.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
